<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • spatialLIBD</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.11" id="spatiallibd-1711">spatialLIBD 1.7.11<a class="anchor" aria-label="anchor" href="#spatiallibd-1711"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Now <code><a href="../reference/layer_stat_cor.html">layer_stat_cor()</a></code> has the <code>top_n</code>
argument which can be used for subsetting the marker genes prior to
computing the correlation as part of the spatial registration
process.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.10" id="spatiallibd-1710">spatialLIBD 1.7.10<a class="anchor" aria-label="anchor" href="#spatiallibd-1710"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added the <code><a href="../reference/add_key.html">add_key()</a></code> function to reduce code duplication
and resolve <a href="https://github.com/LieberInstitute/spatialLIBD/issues/31" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/31</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.9" id="spatiallibd-179">spatialLIBD 1.7.9<a class="anchor" aria-label="anchor" href="#spatiallibd-179"></a></h2>
<p>NEW FEATURES</p>
<ul><li>This version is now compatible with the bioc-devel version of
SpatialExperiment where spatialData() was deprecated. Details at <a href="https://github.com/LieberInstitute/spatialLIBD/pull/29/files" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/pull/29/files</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.7" id="spatiallibd-177">spatialLIBD 1.7.7<a class="anchor" aria-label="anchor" href="#spatiallibd-177"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fixed a bug where the using the left-mouse click was not working for
annotating individual spots under the “gene (interactive)” tab.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.6" id="spatiallibd-176">spatialLIBD 1.7.6<a class="anchor" aria-label="anchor" href="#spatiallibd-176"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code><a href="../reference/vis_gene_p.html">vis_gene_p()</a></code>, <code><a href="../reference/vis_clus_p.html">vis_clus_p()</a></code> and all related
functions now have an argument <code>point_size</code> which lets you
control how big the points are plotted. This can be useful for
visualization purposes.</li>
<li>The shiny app now has an input controlling the point size. If you
increase it to say <code>5</code>, then if you zoom in the
<code>clusters (interactive)</code> panel, you can see larger spots when
zooming in.</li>
<li>These features are related to <a href="https://github.com/LieberInstitute/spatialLIBD/issues/28" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/28</a>
although the spot diameter is still not the true spot diameter. However,
now you have more flexibility for visualizing the spots.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.5" id="spatiallibd-175">spatialLIBD 1.7.5<a class="anchor" aria-label="anchor" href="#spatiallibd-175"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Expanded the <em>Using spatialLIBD with 10x Genomics public
datasets</em> vignette to show how you can deploy your web application.
See <a href="https://libd.shinyapps.io/spatialLIBD_Human_Lymph_Node_10x/" class="external-link uri">https://libd.shinyapps.io/spatialLIBD_Human_Lymph_Node_10x/</a>
for the live example.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.4" id="spatiallibd-174">spatialLIBD 1.7.4<a class="anchor" aria-label="anchor" href="#spatiallibd-174"></a></h2>
<p>BUG FIXES</p>
<ul><li>
<code><a href="../reference/vis_gene.html">vis_gene()</a></code> and <code><a href="../reference/vis_grid_gene.html">vis_grid_gene()</a></code> now support
<code>geneid</code>s that are found in the <code>rownames(spe)</code>.
This makese these functions more flexible.</li>
<li>
<code><a href="../reference/vis_grid_gene.html">vis_grid_gene()</a></code> and <code><a href="../reference/vis_grid_clus.html">vis_grid_clus()</a></code> now
have the <code>sample_order</code> argument which gives you more control
in case you want to plot a subset of samples. This should also reduced
the memory required as discovered at <a href="https://github.com/LieberInstitute/spatialDLPFC/issues/45" class="external-link uri">https://github.com/LieberInstitute/spatialDLPFC/issues/45</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.3" id="spatiallibd-173">spatialLIBD 1.7.3<a class="anchor" aria-label="anchor" href="#spatiallibd-173"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added support for more than one background picture per sample. This
was done through the new argument <code>image_id</code>. Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/25" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/25</a>.</li>
<li>Added options for side by side visualization of the background image
and the clusters or gene expression values in the static versions.
Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/19" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/19</a>.</li>
<li>Allow changing the transparency level of the spots with the
<code>alpha</code> argument. Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/20" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/20</a>.</li>
<li>Add support for image manipulation with the <code>magick</code>
package. Adds functions <code><a href="../reference/img_edit.html">img_edit()</a></code>,
<code><a href="../reference/img_update.html">img_update()</a></code> and <code><a href="../reference/img_update_all.html">img_update_all()</a></code> as well as
new features on the web application. Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/21" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/21</a>.</li>
<li>Added support for more control over the gene color scale and in the
web application also added support for reversing the order of the scale.
Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/22" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/22</a>
and <a href="https://github.com/LieberInstitute/spatialLIBD/issues/23" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/23</a>.</li>
<li>Added <code>export_cluster()</code> and
<code>import_cluster()</code> to help export/import clustering results
instead of having to save large <code>spe</code> objects when exploring
different clustering methods.</li>
<li>Added <code><a href="../reference/locate_images.html">locate_images()</a></code> and <code><a href="../reference/add_images.html">add_images()</a></code> for
adding non-standard images to a <code>spe</code> object.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.2" id="spatiallibd-172">spatialLIBD 1.7.2<a class="anchor" aria-label="anchor" href="#spatiallibd-172"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fixed an issue introduced by newer versions of <code>shiny</code>.
This version of <code>spatialLIBD</code> works with <code>shiny</code>
version 1.7.1, though it’s likely backwards compatible. Resolves <a href="https://github.com/LieberInstitute/spatialLIBD/issues/24" class="external-link uri">https://github.com/LieberInstitute/spatialLIBD/issues/24</a>.</li>
<li>Fix an issue where <code>as.data.frame(colData(spe))</code> uses
<code>check.names = TRUE</code> by default and then changes the column
names unintentionally.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.1" id="spatiallibd-171">spatialLIBD 1.7.1<a class="anchor" aria-label="anchor" href="#spatiallibd-171"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added <code><a href="../reference/read10xVisiumWrapper.html">read10xVisiumWrapper()</a></code> and related functions that
make it easier to read in the SpaceRanger output files and launch a
shiny web application using <code><a href="../reference/run_app.html">run_app()</a></code>. These new functions
read in the analysis output from SpaceRanger by 10x Genomics, in
particular, the clustering and dimension reduction (projection)
results.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.19" id="spatiallibd-1319">spatialLIBD 1.3.19<a class="anchor" aria-label="anchor" href="#spatiallibd-1319"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<code>spatialLIBD</code> has been updated to work with
<code>SpatialExperiment</code> version 1.1.701 which will be released as
part of Bioconductor 3.13. This changes internal code of
<code>spatialLIBD</code> which will work with any objects created with
<code>SpatialExperiment</code> version 1.1.700.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.16" id="spatiallibd-1316">spatialLIBD 1.3.16<a class="anchor" aria-label="anchor" href="#spatiallibd-1316"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>The citation information has changed now that
<code>spatialLIBD</code> has a bioRxiv pre-print at <a href="https://www.biorxiv.org/content/10.1101/2021.04.29.440149v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/2021.04.29.440149v1</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.15" id="spatiallibd-1315">spatialLIBD 1.3.15<a class="anchor" aria-label="anchor" href="#spatiallibd-1315"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>We now use <code><a href="https://rdrr.io/pkg/plotly/man/toWebGL.html" class="external-link">plotly::toWebGL()</a></code> to make the web
application more responsive.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.14" id="spatiallibd-1314">spatialLIBD 1.3.14<a class="anchor" aria-label="anchor" href="#spatiallibd-1314"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>The documentation and help messages shown in the web application
have been revamped and improved.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.12" id="spatiallibd-1312">spatialLIBD 1.3.12<a class="anchor" aria-label="anchor" href="#spatiallibd-1312"></a></h2>
<p>NEW FEATURES</p>
<ul><li>We added a new vignette that shows how you can use
<code>spatialLIBD</code> with any 10x Genomics Visium dataset processed
with <code>spaceranger</code>. The vignette uses the publicly available
human lymph node example from the 10x Genomics website.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.3" id="spatiallibd-133">spatialLIBD 1.3.3<a class="anchor" aria-label="anchor" href="#spatiallibd-133"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Overall the package has been updated to use
<code>SpatialExperiment</code> version 1.1.427 available on Bioconductor
3.13 (bioc-devel). Several functions were re-named such as
<code>sce_image_gene_p()</code> now has a shorter name
<code><a href="../reference/vis_gene_p.html">vis_gene_p()</a></code>. This update also changes these visualization
functions to ONLY support <code>SpatialExperiment</code> objects instead
of the original modified <code>SingleCellExperiment</code> objects.</li>
<li>Updated citation information to reflect that <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link uri">https://doi.org/10.1038/s41593-020-00787-0</a> is now
public. Also added a link on the README to <a href="https://doi.org/10.6084/m9.figshare.13623902.v1" class="external-link uri">https://doi.org/10.6084/m9.figshare.13623902.v1</a> for the
manuscript high resolution images.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.7" id="spatiallibd-117">spatialLIBD 1.1.7<a class="anchor" aria-label="anchor" href="#spatiallibd-117"></a></h2>
<p>NEW FEATURES</p>
<ul><li>The functions <code>sce_image_gene_p()</code>,
<code>sce_image_gene()</code>, <code>sce_image_grid()</code>,
<code>sce_image_grid_gene()</code>, <code>sce_image_clus()</code>,
<code>sce_image_clus_p()</code>, <code><a href="../reference/geom_spatial.html">geom_spatial()</a></code> now work
with VisiumExperiment objects thanks to the new function
<code>read_image()</code> and <code>ve_image_colData()</code>. This work
was done by Brenda Pardo and Leonardo.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.5" id="spatiallibd-115">spatialLIBD 1.1.5<a class="anchor" aria-label="anchor" href="#spatiallibd-115"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code><a href="../reference/fetch_data.html">fetch_data()</a></code> takes the data from sce object and creates
a VisiumExperiment object containing these data thanks to the function
<code>sce_to_ve()</code>. VisiumExperiment object can be obtained with
<code>fetch_data("ve")</code>. This work was done by Brenda Pardo and
Leonardo.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.4" id="spatiallibd-114">spatialLIBD 1.1.4<a class="anchor" aria-label="anchor" href="#spatiallibd-114"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code><a href="../reference/fetch_data.html">fetch_data()</a></code> now uses <code><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache()</a></code> when
downloading the data from Dropbox.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.14" id="spatiallibd-09914">spatialLIBD 0.99.14<a class="anchor" aria-label="anchor" href="#spatiallibd-09914"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Added the function <code><a href="../reference/enough_ram.html">enough_ram()</a></code> which is used to
control the execution of examples. If it fails when using
<code>fetch_data("sce")</code> then <code><a href="../reference/fetch_data.html">fetch_data()</a></code> will show
a warning.</li>
<li>
<code>fetch_data(type = "sce_example")</code> is now supported and
used visibly in the vignette, eliminating the need for
<code>eval = FALSE</code> chunks. This should enable testing the
vignette code on the Bioconductor Single Package Builder on Windows (max
2.5 GB of RAM available).</li>
</ul><p>BUG FIXES</p>
<ul><li>Fixed the example in <code><a href="../reference/get_colors.html">get_colors()</a></code>.</li>
<li>Fixed <code><a href="../reference/layer_stat_cor_plot.html">layer_stat_cor_plot()</a></code> for when <code>min</code>
and/or <code>max</code> are specified.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.13" id="spatiallibd-09913">spatialLIBD 0.99.13<a class="anchor" aria-label="anchor" href="#spatiallibd-09913"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Documentation website is now available at <a href="http://LieberInstitute.github.io/spatialLIBD/" class="external-link uri">http://LieberInstitute.github.io/spatialLIBD/</a>. It gets
updated with every commit on the master branch (bioc-devel) using GitHub
Actions and pkgdown.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.12" id="spatiallibd-09912">spatialLIBD 0.99.12<a class="anchor" aria-label="anchor" href="#spatiallibd-09912"></a></h2>
<p>BUG FIXES</p>
<ul><li>Remove the spatialLIBD.Rproj file =( since BioC’s SBP is asking me
to do so <a href="http://bioconductor.org/spb_reports/spatialLIBD_buildreport_20200303135350.html" class="external-link uri">http://bioconductor.org/spb_reports/spatialLIBD_buildreport_20200303135350.html</a>
</li>
<li>Use <code><a href="https://rdrr.io/r/base/system2.html" class="external-link">system2()</a></code> instead of <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code>.</li>
<li>Move the <code><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed()</a></code> call outside of
<code><a href="../reference/layer_boxplot.html">layer_boxplot()</a></code> as noted by Martin Morgan <a href="https://github.com/Bioconductor/Contributions/issues/1389#issuecomment-594099852" class="external-link uri">https://github.com/Bioconductor/Contributions/issues/1389#issuecomment-594099852</a>
.</li>
<li>Use <code>\linkS4class</code> as I see being done at <a href="https://github.com/drisso/SingleCellExperiment/search?q=linkS4class&amp;unscoped_q=linkS4class" class="external-link uri">https://github.com/drisso/SingleCellExperiment/search?q=linkS4class&amp;unscoped_q=linkS4class</a>.</li>
<li>Use <code><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply()</a></code> instead of <code><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply()</a></code>.</li>
<li>Fix (or attempt to) some doc links.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.11" id="spatiallibd-09911">spatialLIBD 0.99.11<a class="anchor" aria-label="anchor" href="#spatiallibd-09911"></a></h2>
<p>BUG FIXES</p>
<ul><li>Check if removing the <code>RcppAnnoy</code> line in the DESCRIPTION
actually works now based on Aaron Lun’s comment at <a href="https://github.com/eddelbuettel/rcppannoy/issues/57#issuecomment-594097241" class="external-link uri">https://github.com/eddelbuettel/rcppannoy/issues/57#issuecomment-594097241</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.10" id="spatiallibd-09910">spatialLIBD 0.99.10<a class="anchor" aria-label="anchor" href="#spatiallibd-09910"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Include AWS links to the image TIFF files (~500mb each) as requested
by Qian Zhu <a href="mailto:zqian@jimmy.harvard.edu" class="email">zqian@jimmy.harvard.edu</a> for visualizing the data on
the Giotto Viewer <a href="https://www.biorxiv.org/content/10.1101/701680v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/701680v1</a>.</li>
</ul><p>BUG FIXES</p>
<ul><li>Fix <code><a href="../reference/fetch_data.html">fetch_data()</a></code> and the vignette by specifying the
<code>mode = "wb"</code> for <code><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">utils::download.file()</a></code> in
order to resolve an issue with Windows OS reported here <a href="http://bioconductor.org/spb_reports/spatialLIBD_buildreport_20200302120158.html#tokay2_buildsrc_anchor" class="external-link uri">http://bioconductor.org/spb_reports/spatialLIBD_buildreport_20200302120158.html#tokay2_buildsrc_anchor</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.9" id="spatiallibd-0999">spatialLIBD 0.99.9<a class="anchor" aria-label="anchor" href="#spatiallibd-0999"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Link to <a href="https://doi.org/10.1101/2020.02.28.969931" class="external-link uri">https://doi.org/10.1101/2020.02.28.969931</a> now that its
public.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.8" id="spatiallibd-0998">spatialLIBD 0.99.8<a class="anchor" aria-label="anchor" href="#spatiallibd-0998"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>
<a href="https://spatial.libd.org/spatialLIBD" class="external-link uri">https://spatial.libd.org/spatialLIBD</a> is not supported
since we are using Shiny Server and not Shiny Server Pro. So all links
have now been updated to <a href="http://spatial.libd.org/spatialLIBD" class="external-link uri">http://spatial.libd.org/spatialLIBD</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.7" id="spatiallibd-0997">spatialLIBD 0.99.7<a class="anchor" aria-label="anchor" href="#spatiallibd-0997"></a></h2>
<p>BUG FIXES</p>
<ul><li>Run a test that might help with <a href="https://github.com/r-lib/pkgdown/issues/1230" class="external-link uri">https://github.com/r-lib/pkgdown/issues/1230</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.6" id="spatiallibd-0996">spatialLIBD 0.99.6<a class="anchor" aria-label="anchor" href="#spatiallibd-0996"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Add mirrors for the shiny app and change the main location.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.5" id="spatiallibd-0995">spatialLIBD 0.99.5<a class="anchor" aria-label="anchor" href="#spatiallibd-0995"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Make <code><a href="../reference/fetch_data.html">fetch_data()</a></code> more flexible. Should now work when
the data is absent.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.4" id="spatiallibd-0994">spatialLIBD 0.99.4<a class="anchor" aria-label="anchor" href="#spatiallibd-0994"></a></h2>
<p>BUG FIXES</p>
<ul><li>Fix Travis badges</li>
<li>Fix Kristen’s name on the vignette</li>
<li>Add the same welcome information to the top of the vignette, since
this will be what Bioconductor users see first. Basically, we have made
sure that users will see the same information first regardless if they
find the package README, open the shiny app, or find the package
vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.3" id="spatiallibd-0993">spatialLIBD 0.99.3<a class="anchor" aria-label="anchor" href="#spatiallibd-0993"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Further refine the READMEs (pkg and shiny). They now include the
list of links to the raw 10x Genomics files as well as a short
description of the project at the top. This was in response to feedback
by Andrew Jaffe.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.2" id="spatiallibd-0992">spatialLIBD 0.99.2<a class="anchor" aria-label="anchor" href="#spatiallibd-0992"></a></h2>
<p>SIGNIFICANT USER-VISIBLE CHANGES</p>
<ul><li>Update main package READMEs to reflect the changes to the shiny web
app README.md.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.1" id="spatiallibd-0991">spatialLIBD 0.99.1<a class="anchor" aria-label="anchor" href="#spatiallibd-0991"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added Kristen R Maynard to the DESCRIPTION file.</li>
<li>Improved the shiny app page footer.</li>
<li>Moved around the documentation and added a new main tab with an
overview in response to the feedback by Stephanie Hicks.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.0" id="spatiallibd-0990">spatialLIBD 0.99.0<a class="anchor" aria-label="anchor" href="#spatiallibd-0990"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added a <code>NEWS.md</code> file to track changes to the
package.</li>
<li>First full version of the package to be submitted to Bioconductor.
Note that the <code><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub::ExperimentHub()</a></code> functionality
won’t work until they approve the package. However, for now
<code><a href="../reference/fetch_data.html">fetch_data()</a></code> has a backup mechanism in place.</li>
<li>Submitted to Bioconductor <a href="https://github.com/Bioconductor/Contributions/issues/1389" class="external-link">here</a>.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

