<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Quality Control for Spatial Data — add_qc_metrics • spatialLIBD</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Quality Control for Spatial Data — add_qc_metrics"><meta property="og:description" content="This function identify spots in a
SpatialExperiment-class (SPE)
with outlier quality control values: low sum_umi or sum_gene, or high
expr_chrM_ratio, utilizing scuttle::isOutlier. Also identifies in-tissue
edge spots and distance to the edge for each spot."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/guide_to_spatial_registration.html">Guide to Spatial Registration</a>
    </li>
    <li>
      <a href="../articles/multi_gene_plots.html">Guide to Multi-Gene Plots</a>
    </li>
    <li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Quality Control for Spatial Data</h1>
    <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/HEAD/R/add_qc_metrics.R" class="external-link"><code>R/add_qc_metrics.R</code></a></small>
    <div class="hidden name"><code>add_qc_metrics.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function identify spots in a
<a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a> (SPE)
with outlier quality control values: low <code>sum_umi</code> or <code>sum_gene</code>, or high
<code>expr_chrM_ratio</code>, utilizing <a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">scuttle::isOutlier</a>. Also identifies in-tissue
edge spots and distance to the edge for each spot.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">add_qc_metrics</span><span class="op">(</span><span class="va">spe</span>, overwrite <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>spe</dt>
<dd><p>a <a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment</a>
object that has <code>sum_umi</code>, <code>sum_gene</code>, <code>expr_chrM_ratio</code>, and <code>in_tissue</code>
variables in the <code>colData(spe)</code>. Note that these are automatically created
when you build your <code>spe</code> object with <code><a href="read10xVisiumWrapper.html">spatialLIBD::read10xVisiumWrapper()</a></code>.</p></dd>


<dt>overwrite</dt>
<dd><p>a <code>logical(1)</code> specifying whether to overwrite the 7
<code>colData(spe)</code> columns that this function creates. If set to <code>FALSE</code> and any
of them are present, the function will return an error.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment</a></p>


<p>with added quality control information added to the <code>colData()</code>.</p><dl><dt><code>scran_low_lib_size</code></dt>
<dd><p>shows spots that have a low library size.</p></dd>

<dt><code>scran_low_n_features</code></dt>
<dd><p>spots with a low number of expressed genes.</p></dd>

<dt><code>scran_high_Mito_percent</code></dt>
<dd><p>spots with a high percent of mitochondrial gene expression.</p></dd>

<dt><code>scran_discard</code></dt>
<dd><p>spots belonging to either <code>scran_low_lib_size</code>,
<code>scran_low_n_feature</code>, or <code>scran_high_Mito_percent</code>.</p></dd>

<dt><code>edge_spot</code></dt>
<dd><p>spots that are automatically detected as the edge spots
of the <code>in_tissue</code> section.</p></dd>

<dt><code>edge_distance</code></dt>
<dd><p>closest distance in number of spots to either the
vertical or horizontal edge.</p></dd>

<dt><code>scran_low_lib_size_edge</code></dt>
<dd><p>spots that have a low library size and
are an edge spot.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The initial version of this function lives at
<a href="https://github.com/LieberInstitute/Visium_SPG_AD/blob/master/code/07_spot_qc/01_qc_metrics_and_segmentation.R" class="external-link">https://github.com/LieberInstitute/Visium_SPG_AD/blob/master/code/07_spot_qc/01_qc_metrics_and_segmentation.R</a>.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Louise A. Huuki-Myers</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Obtain the necessary data</span></span></span>
<span class="r-in"><span><span class="va">spe_pre_qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="fetch_data.html">fetch_data</a></span><span class="op">(</span><span class="st">"spatialDLPFC_Visium_example_subset"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2024-04-29</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-07-12 13:30:58.248283 loading file /Users/leocollado/Library/Caches/org.R-project.R/R/BiocFileCache/6f563a40ecf7_spatialDLPFC_spe_subset_example.rds%3Fdl%3D1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## For now, we fake out tissue spots in example data</span></span></span>
<span class="r-in"><span><span class="va">spe_qc</span> <span class="op">&lt;-</span> <span class="va">spe_pre_qc</span></span></span>
<span class="r-in"><span><span class="va">spe_qc</span><span class="op">$</span><span class="va">in_tissue</span><span class="op">[</span><span class="va">spe_qc</span><span class="op">$</span><span class="va">array_col</span> <span class="op">&lt;</span> <span class="fl">10</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## adds QC metrics to colData of the spe</span></span></span>
<span class="r-in"><span><span class="va">spe_qc</span> <span class="op">&lt;-</span> <span class="fu">add_qc_metrics</span><span class="op">(</span><span class="va">spe_qc</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe_qc</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">vars</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^(scran|edge)"</span>, <span class="va">vars</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "scran_discard"                   "scran_high_subsets_Mito_percent"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3] "scran_low_lib_size"              "scran_low_n_features"           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "scran_quick_cluster"             "scran_high_Mito_percent"        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [7] "edge_spot"                       "edge_distance"                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [9] "scran_low_lib_size_edge"        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## visualize edge spots</span></span></span>
<span class="r-in"><span><span class="fu"><a href="vis_clus.html">vis_clus</a></span><span class="op">(</span><span class="va">spe_qc</span>, sampleid <span class="op">=</span> <span class="st">"Br6432_ant"</span>, clustervar <span class="op">=</span> <span class="st">"edge_spot"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="vis_gene.html">vis_gene</a></span><span class="op">(</span><span class="va">spe_qc</span>, sampleid <span class="op">=</span> <span class="st">"Br6432_ant"</span>, geneid <span class="op">=</span> <span class="st">"edge_distance"</span>, minCount <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Visualize scran QC flags</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Check the spots with low library size as detected by scran::isOutlier()</span></span></span>
<span class="r-in"><span><span class="fu"><a href="vis_clus.html">vis_clus</a></span><span class="op">(</span><span class="va">spe_qc</span>, sample_id <span class="op">=</span> <span class="st">"Br6432_ant"</span>, clustervar <span class="op">=</span> <span class="st">"scran_low_lib_size"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Violin plot of library size with low library size highlighted in a</span></span></span>
<span class="r-in"><span><span class="co">## different color.</span></span></span>
<span class="r-in"><span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">spe_qc</span><span class="op">[</span>, <span class="va">spe_qc</span><span class="op">$</span><span class="va">in_tissue</span><span class="op">]</span>, x <span class="op">=</span> <span class="st">"sample_id"</span>, y <span class="op">=</span> <span class="st">"sum_umi"</span>, colour_by <span class="op">=</span> <span class="st">"scran_low_lib_size"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Check any spots that scran::isOutlier() flagged</span></span></span>
<span class="r-in"><span><span class="fu"><a href="vis_clus.html">vis_clus</a></span><span class="op">(</span><span class="va">spe_qc</span>, sampleid <span class="op">=</span> <span class="st">"Br6432_ant"</span>, clustervar <span class="op">=</span> <span class="st">"scran_discard"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Low library spots that are on the edge of the tissue</span></span></span>
<span class="r-in"><span><span class="fu"><a href="vis_clus.html">vis_clus</a></span><span class="op">(</span><span class="va">spe_qc</span>, sampleid <span class="op">=</span> <span class="st">"Br6432_ant"</span>, clustervar <span class="op">=</span> <span class="st">"scran_low_lib_size_edge"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Use `low_library_size` (or other variables) and `edge_distance` as you</span></span></span>
<span class="r-in"><span><span class="co">## please.</span></span></span>
<span class="r-in"><span><span class="va">spe_qc</span><span class="op">$</span><span class="va">our_low_lib_edge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">spe_qc</span><span class="op">$</span><span class="va">scran_low_lib_size</span> <span class="op">==</span> <span class="st">"TRUE"</span> <span class="op">&amp;</span></span></span>
<span class="r-in"><span>        <span class="va">spe_qc</span><span class="op">$</span><span class="va">edge_distance</span> <span class="op">&lt;</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>    levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRUE"</span>, <span class="st">"FALSE"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="vis_clus.html">vis_clus</a></span><span class="op">(</span><span class="va">spe_qc</span>, sample_id <span class="op">=</span> <span class="st">"Br6432_ant"</span>, clustervar <span class="op">=</span> <span class="st">"our_low_lib_edge"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="add_qc_metrics-7.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Clean up</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">spe_qc</span>, <span class="va">spec_pre_qc</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>object 'spec_pre_qc' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

