<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Annotated spatially-registered clusters — annotate_registered_clusters • spatialLIBD</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Annotated spatially-registered clusters — annotate_registered_clusters"><meta property="og:description" content="Once you have computed the enrichment t-statistics for your sc/snRNA-seq data
using registration_wrapper() and related functions, you can then use
layer_stat_cor() and layer_stat_cor_plot() to perform the spatial
registartion of your sc/snRNA-seq data. This function helps interpret that
matrix and assign layer labels to your clusters."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Annotated spatially-registered clusters</h1>
    <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/HEAD/R/annotate_registered_clusters.R" class="external-link"><code>R/annotate_registered_clusters.R</code></a></small>
    <div class="hidden name"><code>annotate_registered_clusters.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Once you have computed the enrichment t-statistics for your sc/snRNA-seq data
using <code><a href="registration_wrapper.html">registration_wrapper()</a></code> and related functions, you can then use
<code><a href="layer_stat_cor.html">layer_stat_cor()</a></code> and <code><a href="layer_stat_cor_plot.html">layer_stat_cor_plot()</a></code> to perform the spatial
registartion of your sc/snRNA-seq data. This function helps interpret that
matrix and assign layer labels to your clusters.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">annotate_registered_clusters</span><span class="op">(</span></span>
<span>  <span class="va">cor_stats_layer</span>,</span>
<span>  confidence_threshold <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  cutoff_merge_ratio <span class="op">=</span> <span class="fl">0.25</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>cor_stats_layer</dt>
<dd><p>The output of <code><a href="layer_stat_cor.html">layer_stat_cor()</a></code>.</p></dd>


<dt>confidence_threshold</dt>
<dd><p>A <code>numeric(1)</code> specifying the minimum correlation
that a given cluster must have against any of the layers (by default) to
be considered as having a 'good' assignment. Otherwise, the confidence will
be 'poor' and the final label will have an asterisk.</p></dd>


<dt>cutoff_merge_ratio</dt>
<dd><p>A <code>numeric(1)</code> specifying the threshold for merging
or not layer assignments (by default). This is a proportion of the difference
between the current correlation and the next highest given the units of the
next highest correlation. Defaults to a difference of 25% of the next highest
correlation: if the observed difference is lower than this threshold, then we
keep merging. Higher values will lead to more layers (by default) being
merged.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code>data.frame</code> with 3 columns. Your <code>cluster</code>s, the <code>layer_confidence</code></p>


<p>which depends on <code>confidence_threshold</code>, and the <code>layer_label</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>If you change the input <code>modeling_results</code> to <code><a href="layer_stat_cor.html">layer_stat_cor()</a></code> then the
interpretation of this function could change. For example, maybe you have
your own spatially-resolved transcriptomics data that doesn't have to be
about DLPFC layers.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other Layer correlation functions: 
<code><a href="layer_stat_cor_plot.html">layer_stat_cor_plot</a>()</code>,
<code><a href="layer_stat_cor.html">layer_stat_cor</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Obtain the necessary data</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"modeling_results"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">modeling_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="fetch_data.html">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"modeling_results"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2022-10-31</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-12-16 22:32:47 loading file /github/home/.cache/R/BiocFileCache/4f674d1888_Human_DLPFC_Visium_modeling_results.Rdata%3Fdl%3D1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Compute the correlations</span></span></span>
<span class="r-in"><span><span class="va">cor_stats_layer</span> <span class="op">&lt;-</span> <span class="fu"><a href="layer_stat_cor.html">layer_stat_cor</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>,</span></span>
<span class="r-in"><span>    <span class="va">modeling_results</span>,</span></span>
<span class="r-in"><span>    model_type <span class="op">=</span> <span class="st">"enrichment"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Obtain labels</span></span></span>
<span class="r-in"><span><span class="fu">annotate_registered_clusters</span><span class="op">(</span><span class="va">cor_stats_layer</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    cluster layer_confidence layer_label</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   22 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    3 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   23 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   17 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   21 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6    7 (4)             good          L1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7    5 (4)             good          L1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8    9 (5)             poor      L1/WM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9   26 (5)             poor         L1*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10   1 (4)             poor      L1/WM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  24 (6)             poor         L6*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  13 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  10 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  27 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  29 (2)             poor         L4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  14 (2)             good          L4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17  15 (2)             poor       L5/4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18  18 (2)             poor         L4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19   2 (1)             poor         L5*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20  31 (1)             good          L5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21   8 (1)             poor         L2*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22  16 (2)             poor         L2*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23  28 (2)             poor         L2*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24  30 (2)             poor         L2*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25  20 (2)             poor     L3/5/4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26  11 (2)             poor       L4/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  25 (2)             poor     L2/4/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28   4 (1)             good        L3/4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29  12 (1)             good          L4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30   6 (1)             good          L2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31  19 (1)             good          L3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## More relaxed merging threshold</span></span></span>
<span class="r-in"><span><span class="fu">annotate_registered_clusters</span><span class="op">(</span><span class="va">cor_stats_layer</span>, cutoff_merge_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    cluster layer_confidence layer_label</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   22 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    3 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   23 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   17 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   21 (3)             good          WM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6    7 (4)             good          L1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7    5 (4)             good          L1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8    9 (5)             poor      L1/WM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9   26 (5)             poor      L1/WM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10   1 (4)             poor      L1/WM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  24 (6)             poor    L6/WM/1*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  13 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  10 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  27 (1)             good          L6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  29 (2)             poor       L4/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  14 (2)             good      L4/5/3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17  15 (2)             poor       L5/4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18  18 (2)             poor     L4/5/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19   2 (1)             poor         L5*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20  31 (1)             good          L5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21   8 (1)             poor       L2/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22  16 (2)             poor       L2/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23  28 (2)             poor       L2/3*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24  30 (2)             poor L2/3/4/5/1*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25  20 (2)             poor     L3/5/4*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26  11 (2)             poor   L4/3/2/5*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  25 (2)             poor   L2/4/3/5*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28   4 (1)             good        L3/4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29  12 (1)             good      L4/5/3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30   6 (1)             good        L2/3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31  19 (1)             good      L3/2/4</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

