<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Layer modeling correlation of statistics — layer_stat_cor • spatialLIBD</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Layer modeling correlation of statistics — layer_stat_cor" />
<meta property="og:description" content="Layer modeling correlation of statistics" />
<meta property="og:image" content="/logo.png" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Layer modeling correlation of statistics</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/spatialLIBD/blob/master/R/layer_stat_cor.R'><code>R/layer_stat_cor.R</code></a></small>
    <div class="hidden name"><code>layer_stat_cor.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Layer modeling correlation of statistics</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>layer_stat_cor</span><span class='op'>(</span>
  <span class='va'>stats</span>,
  modeling_results <span class='op'>=</span> <span class='fu'><a href='fetch_data.html'>fetch_data</a></span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"modeling_results"</span><span class='op'>)</span>,
  model_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>modeling_results</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>,
  reverse <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>stats</th>
      <td><p>A data.frame where the row names are Ensembl gene IDs, the
column names are labels for clusters of cells or cell types, and where
each cell contains the given statistic for that gene and cell type. These
statistics should be computed similarly to the modeling results from
the data we provide. For example, like the <code>enrichment</code> t-statistics that
are derived from comparing one layer against the rest. The <code>stats</code> will be
matched and then correlated with our statistics.</p></td>
    </tr>
    <tr>
      <th>modeling_results</th>
      <td><p>Defaults to the output of
<code>fetch_data(type = 'modeling_results')</code>. This is a list of tables with the
columns <code>f_stat_*</code> or <code>t_stat_*</code> as well as <code>p_value_*</code> and <code>fdr_*</code> plus
<code>ensembl</code>. The column name is used to extract the statistic results, the
p-values, and the FDR adjusted p-values. Then the <code>ensembl</code> column is used
for matching in some cases. See <code><a href='fetch_data.html'>fetch_data()</a></code> for more details.</p></td>
    </tr>
    <tr>
      <th>model_type</th>
      <td><p>A named element of the <code>modeling_results</code> list. By default
that is either <code>enrichment</code> for the model that tests one human brain layer
against the rest (one group vs the rest), <code>pairwise</code> which compares two
layers (groups) denoted by <code>layerA-layerB</code> such that <code>layerA</code> is greater
than <code>layerB</code>, and <code>anova</code> which determines if any layer (group) is different
from the rest adjusting for the mean expression level. The statistics for
<code>enrichment</code> and <code>pairwise</code> are t-statistics while the <code>anova</code> model ones
are F-statistics.</p></td>
    </tr>
    <tr>
      <th>reverse</th>
      <td><p>A <code>logical(1)</code> indicating whether to multiply by <code>-1</code> the
input statistics and reverse the <code>layerA-layerB</code> column names (using the <code>-</code>)
into <code>layerB-layerA</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A correlation matrix between <code>stats</code> and our statistics using only
the Ensembl gene IDs present in both tables. The columns are sorted using
a hierarchical cluster.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Check
https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/Layer_Guesses/dlpfc_snRNAseq_annotation.R
for a full analysis from which this family of functions is derived from.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other Layer correlation functions: 
<code><a href='layer_stat_cor_plot.html'>layer_stat_cor_plot</a>()</code></p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Andrew E Jaffe, Leonardo Collado-Torres</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'></span>
<span class='r-in'><span class='co'>## Obtain the necessary data</span></span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/exists.html'>exists</a></span><span class='op'>(</span><span class='st'>"modeling_results"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>    <span class='va'>modeling_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='fetch_data.html'>fetch_data</a></span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"modeling_results"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-08-09</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 00:43:13 loading file /github/home/.cache/R/BiocFileCache/63056d0d992_Human_DLPFC_Visium_modeling_results.Rdata%3Fdl%3D1</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Compute the correlations</span></span>
<span class='r-in'><span class='va'>cor_stats_layer</span> <span class='op'>&lt;-</span> <span class='fu'>layer_stat_cor</span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>,</span>
<span class='r-in'>    <span class='va'>modeling_results</span>,</span>
<span class='r-in'>    <span class='st'>"enrichment"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Explore the correlation matrix</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>cor_stats_layer</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fl'>3</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>               WM       Layer6     Layer5</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 22 (3) 0.6824669 -0.009192291 -0.1934265</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3 (3)  0.7154122 -0.070042729 -0.2290574</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 23 (3) 0.6637885 -0.031467704 -0.2018306</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 17 (3) 0.6364983 -0.094216046 -0.2026147</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 21 (3) 0.6281443 -0.050336358 -0.1988774</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 7 (4)  0.1850724 -0.197283175 -0.2716890</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Leonardo Collado-Torres.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


