<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Spatial registration: wrapper function — registration_wrapper • spatialLIBD</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Spatial registration: wrapper function — registration_wrapper"><meta property="og:description" content="This function is provided for convenience. It runs all the functions
required for computing the modeling_results. This can be useful for
finding marker genes on a new spatially-resolved transcriptomics dataset
and thus using it for run_app(). The results from this function can also be
used for performing spatial registration through layer_stat_cor() and
related functions of sc/snRNA-seq datasets."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/guide_to_spatial_registration.html">Guide to Spatial Registration</a>
    </li>
    <li>
      <a href="../articles/multi_gene_plots.html">Guide to Multi-Gene Plots</a>
    </li>
    <li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Spatial registration: wrapper function</h1>
    <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/devel/R/registration_wrapper.R" class="external-link"><code>R/registration_wrapper.R</code></a></small>
    <div class="hidden name"><code>registration_wrapper.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is provided for convenience. It runs all the functions
required for computing the <code>modeling_results</code>. This can be useful for
finding marker genes on a new spatially-resolved transcriptomics dataset
and thus using it for <code><a href="run_app.html">run_app()</a></code>. The results from this function can also be
used for performing spatial registration through <code><a href="layer_stat_cor.html">layer_stat_cor()</a></code> and
related functions of sc/snRNA-seq datasets.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">registration_wrapper</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  <span class="va">var_registration</span>,</span>
<span>  <span class="va">var_sample_id</span>,</span>
<span>  covars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_ensembl <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  suffix <span class="op">=</span> <span class="st">""</span>,</span>
<span>  min_ncells <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  pseudobulk_rds_file <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sce">sce<a class="anchor" aria-label="anchor" href="#arg-sce"></a></dt>
<dd><p>A
<a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment-class</a>
object or one that inherits its properties.</p></dd>


<dt id="arg-var-registration">var_registration<a class="anchor" aria-label="anchor" href="#arg-var-registration"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>colData(sce)</code>
variable of interest against which will be used for computing the relevant
statistics.</p></dd>


<dt id="arg-var-sample-id">var_sample_id<a class="anchor" aria-label="anchor" href="#arg-var-sample-id"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>colData(sce)</code> variable
with the sample ID.</p></dd>


<dt id="arg-covars">covars<a class="anchor" aria-label="anchor" href="#arg-covars"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> with names of sample-level covariates.</p></dd>


<dt id="arg-gene-ensembl">gene_ensembl<a class="anchor" aria-label="anchor" href="#arg-gene-ensembl"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>rowData(sce_pseudo)</code>
column with the ENSEMBL gene IDs. This will be used by <code><a href="layer_stat_cor.html">layer_stat_cor()</a></code>.</p></dd>


<dt id="arg-gene-name">gene_name<a class="anchor" aria-label="anchor" href="#arg-gene-name"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>rowData(sce_pseudo)</code>
column with the gene names (symbols).</p></dd>


<dt id="arg-suffix">suffix<a class="anchor" aria-label="anchor" href="#arg-suffix"></a></dt>
<dd><p>A <code>character(1)</code> specifying the suffix to use for the
F-statistics column. This is particularly useful if you will run this
function more than once and want to be able to merge the results.</p></dd>


<dt id="arg-min-ncells">min_ncells<a class="anchor" aria-label="anchor" href="#arg-min-ncells"></a></dt>
<dd><p>An <code>integer(1)</code> greater than 0 specifying the minimum
number of cells (for scRNA-seq) or spots (for spatial) that are combined
when pseudo-bulking. Pseudo-bulked samples with less than <code>min_ncells</code> on
<code>sce_pseudo$ncells</code> will be dropped.</p></dd>


<dt id="arg-pseudobulk-rds-file">pseudobulk_rds_file<a class="anchor" aria-label="anchor" href="#arg-pseudobulk-rds-file"></a></dt>
<dd><p>A <code>character(1)</code> specifying the path for saving
an RDS file with the pseudo-bulked object. It's useful to specify this since
pseudo-bulking can take hours to run on large datasets.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code> of <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> with the statistical results. This is
similar to <code>fetch_data("modeling_results")</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>We chose a default of <code>min_ncells = 10</code> based on OSCA from section 4.3
at
<a href="http://bioconductor.org/books/3.15/OSCA.multisample/multi-sample-comparisons.html" class="external-link">http://bioconductor.org/books/3.15/OSCA.multisample/multi-sample-comparisons.html</a>.
They cite <a href="https://doi.org/10.1038/s41467-020-19894-4" class="external-link">https://doi.org/10.1038/s41467-020-19894-4</a> as the paper where
they came up with the definition of "very low" being 10. You might want
to use <code><a href="registration_pseudobulk.html">registration_pseudobulk()</a></code> and manually explore <code>sce_pseudo$ncells</code>
to choose the best cutoff.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other spatial registration and statistical modeling functions:
<code><a href="registration_block_cor.html">registration_block_cor</a>()</code>,
<code><a href="registration_model.html">registration_model</a>()</code>,
<code><a href="registration_pseudobulk.html">registration_pseudobulk</a>()</code>,
<code><a href="registration_stats_anova.html">registration_stats_anova</a>()</code>,
<code><a href="registration_stats_enrichment.html">registration_stats_enrichment</a>()</code>,
<code><a href="registration_stats_pairwise.html">registration_stats_pairwise</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Ensure reproducibility of example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20220907</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Generate example data</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/mockSCE.html" class="external-link">mockSCE</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Add some sample IDs</span></span></span>
<span class="r-in"><span><span class="va">sce</span><span class="op">$</span><span class="va">sample_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Add a sample-level covariate: age</span></span></span>
<span class="r-in"><span><span class="va">ages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">5</span>, mean <span class="op">=</span> <span class="fl">20</span>, sd <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ages</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">sce</span><span class="op">$</span><span class="va">age</span> <span class="op">&lt;-</span> <span class="va">ages</span><span class="op">[</span><span class="va">sce</span><span class="op">$</span><span class="va">sample_id</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Add gene-level information</span></span></span>
<span class="r-in"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"ENSG"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Compute all modeling results</span></span></span>
<span class="r-in"><span><span class="va">example_modeling_results</span> <span class="op">&lt;-</span> <span class="fu">registration_wrapper</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">sce</span>,</span></span>
<span class="r-in"><span>    var_registration <span class="op">=</span> <span class="st">"Cell_Cycle"</span>,</span></span>
<span class="r-in"><span>    var_sample_id <span class="op">=</span> <span class="st">"sample_id"</span>,</span></span>
<span class="r-in"><span>    covars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"age"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    gene_ensembl <span class="op">=</span> <span class="st">"ensembl"</span>,</span></span>
<span class="r-in"><span>    gene_name <span class="op">=</span> <span class="st">"gene_name"</span>,</span></span>
<span class="r-in"><span>    suffix <span class="op">=</span> <span class="st">"wrapper"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.101932 make pseudobulk object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.255897 dropping 9 pseudo-bulked samples that are below 'min_ncells'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.277534 drop lowly expressed genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.395509 normalize expression</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.452923 create model matrix</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:50:58.463573 run duplicateCorrelation()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:00.834925 The estimated correlation is: -0.0783081238514532</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:00.837409 computing enrichment statistics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:00.952989 extract and reformat enrichment results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:00.978619 running the baseline pairwise model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:00.996935 computing pairwise statistics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-10-22 13:51:01.07164 computing F-statistics</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Explore the results from registration_wrapper()</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">example_modeling_results</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "list"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">example_modeling_results</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 3</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">example_modeling_results</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "anova"      "enrichment" "pairwise"  </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">example_modeling_results</span>, <span class="va">head</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $anova</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $enrichment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        t_stat_G0  t_stat_G1 t_stat_G2M    t_stat_S p_value_G0 p_value_G1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1 -0.1547374  1.5323608 -0.2348981 -1.29289250 0.88011484 0.15652594</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2  0.5630611  1.4947773 -0.5105565 -1.76319136 0.58582739 0.16593179</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3  0.4228896 -0.8317725 -0.4890891  1.03785001 0.68134877 0.42500953</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4 -0.8090376  0.8315512  1.5976031 -1.75950259 0.43736743 0.42512872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5 -2.0758087  0.2188378  0.2130925  1.15619668 0.06472348 0.83119090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6  0.3913959 -1.8396065  1.6689318  0.07483167 0.70373612 0.09574789</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       p_value_G2M p_value_S    fdr_G0    fdr_G1   fdr_G2M     fdr_S   logFC_G0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1   0.8190364 0.2252116 0.9852255 0.9966809 0.9953748 0.9681139 -0.1236594</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2   0.6207620 0.1084407 0.9612662 0.9966809 0.9899698 0.9392089  0.5488682</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3   0.6353470 0.3238615 0.9648540 0.9966809 0.9921856 0.9681139  0.1430331</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4   0.1412712 0.1090902 0.9612662 0.9966809 0.9399839 0.9392089 -0.9237743</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5   0.8355440 0.2745611 0.8631058 0.9966809 0.9953748 0.9681139 -2.1724966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6   0.1261463 0.9418292 0.9648540 0.9966809 0.9399839 0.9906646  0.3550924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         logFC_G1  logFC_G2M     logFC_S ensembl  gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1  0.8654897 -0.1777803 -0.84406114   ENSG1 gene1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2  1.0494805 -0.4733543 -1.34394213   ENSG2 gene2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3 -0.2158331 -0.1565191  0.29666354   ENSG3 gene3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4  0.7435077  1.5938565 -1.59726925   ENSG4 gene4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5  0.2144746  0.2525529  1.19887499   ENSG5 gene5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6 -1.1400513  1.2795838  0.06030075   ENSG6 gene6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $pairwise</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       t_stat_G0-G1 t_stat_G0-G2M t_stat_G0-S t_stat_G1-G2M t_stat_G1-S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1   -0.6671464     0.2089510   0.6599089    0.91003998   1.5430022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2   -0.1147743     0.9762769   1.5297496    1.24964041   2.0208873</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3    0.5773407     0.4415661  -0.2753587   -0.06404534  -0.9687469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4   -0.9551094    -1.4267069   0.3614961   -0.70335585   1.4860622</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5   -1.6015394    -1.5242463  -2.1000859   -0.17030985  -0.8625983</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6    1.1007274    -0.8249394   0.1399109   -2.05967232  -1.0228113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       t_stat_G2M-S p_value_G0-G1 p_value_G0-G2M p_value_G0-S p_value_G1-G2M</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1    0.4932384     0.5234599      0.8397112   0.52785817     0.38940816</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2    0.5901220     0.9114537      0.3575335   0.16462386     0.24676596</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3   -0.8005264     0.5795972      0.6705001   0.79002220     0.95050622</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4    2.0036167     0.3675007      0.1915203   0.72709410     0.50179611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5   -0.6032018     0.1479422      0.1659702   0.06894394     0.86899666</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6    1.0822474     0.3030434      0.4333238   0.89219028     0.07341321</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       p_value_G1-S p_value_G2M-S fdr_G0-G1 fdr_G0-G2M  fdr_G0-S fdr_G1-G2M</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1   0.16142199     0.6351035 0.9984915  0.9764704 0.9742266  0.9854617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2   0.07796735     0.5714040 0.9984915  0.9531915 0.9683756  0.9854617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3   0.36105562     0.4465356 0.9984915  0.9764704 0.9742266  0.9977609</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4   0.17558652     0.0800825 0.9984915  0.8839550 0.9742266  0.9901259</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5   0.41347911     0.5630883 0.9984915  0.8492243 0.8695062  0.9947237</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6   0.33634080     0.3106958 0.9984915  0.9531915 0.9864940  0.9854617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        fdr_G1-S fdr_G2M-S logFC_G0-G1 logFC_G0-G2M logFC_G0-S logFC_G1-G2M</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1 0.9983017 0.9742209  -0.5605347    0.2040786  0.6323864   0.76461332</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2 0.9860485 0.9695084  -0.1085720    1.0735394  1.6504823   1.18211143</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3 0.9983017 0.9651275   0.2209283    0.1964203 -0.1201808  -0.02450793</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4 0.9983017 0.9297596  -1.0540875   -1.8303321  0.4550345  -0.77624467</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5 0.9983017 0.9695084  -1.9220357   -2.1264276 -2.8746043  -0.20439186</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6 0.9983017 0.9651275   0.9692943   -0.8444414  0.1405221  -1.81373577</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       logFC_G1-S logFC_G2M-S ensembl  gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG1  1.1929212   0.4283078   ENSG1 gene1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG2  1.7590544   0.5769429   ENSG2 gene2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG3 -0.3411090  -0.3166011   ENSG3 gene3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG4  1.5091219   2.2853666   ENSG4 gene4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG5 -0.9525685  -0.7481767   ENSG5 gene5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG6 -0.8287723   0.9849635   ENSG6 gene6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

