<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Sample spatial gene visualization grid — vis_grid_gene • spatialLIBD</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Sample spatial gene visualization grid — vis_grid_gene"><meta property="og:description" content="This function visualizes the gene expression stored in assays(spe) or any
continuous variable stored in colData(spe) for a set of samples at the
spot-level using (by default) the histology information on the background.
To visualize clusters (or any discrete variable) use vis_grid_clus()."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/guide_to_spatial_registration.html">Guide to Spatial Registration</a>
    </li>
    <li>
      <a href="../articles/multi_gene_plots.html">Guide to Multi-Gene Plots</a>
    </li>
    <li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Sample spatial gene visualization grid</h1>
    <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/devel/R/vis_grid_gene.R" class="external-link"><code>R/vis_grid_gene.R</code></a></small>
    <div class="hidden name"><code>vis_grid_gene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function visualizes the gene expression stored in <code>assays(spe)</code> or any
continuous variable stored in <code>colData(spe)</code> for a set of samples at the
spot-level using (by default) the histology information on the background.
To visualize clusters (or any discrete variable) use <code><a href="vis_grid_clus.html">vis_grid_clus()</a></code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">vis_grid_gene</span><span class="op">(</span></span>
<span>  <span class="va">spe</span>,</span>
<span>  geneid <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_search</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>  <span class="va">pdf_file</span>,</span>
<span>  assayname <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>  minCount <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  return_plots <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  spatial <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  viridis <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">24</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">36</span>,</span>
<span>  image_id <span class="op">=</span> <span class="st">"lowres"</span>,</span>
<span>  alpha <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  cont_colors <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="va">viridis</span><span class="op">)</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="kw">else</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aquamarine4"</span>,</span>
<span>    <span class="st">"springgreen"</span>, <span class="st">"goldenrod"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>  sample_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">sample_id</span><span class="op">)</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  auto_crop <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  na_color <span class="op">=</span> <span class="st">"#CCCCCC40"</span>,</span>
<span>  is_stitched <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-spe">spe<a class="anchor" aria-label="anchor" href="#arg-spe"></a></dt>
<dd><p>A
<a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a>
object. See <code><a href="fetch_data.html">fetch_data()</a></code> for how to download some example objects or
<code><a href="read10xVisiumWrapper.html">read10xVisiumWrapper()</a></code> to read in <code>spaceranger --count</code> output files and
build your own <code>spe</code> object.</p></dd>


<dt id="arg-geneid">geneid<a class="anchor" aria-label="anchor" href="#arg-geneid"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> specifying the gene ID(s) stored in
<code>rowData(spe)$gene_search</code> or a continuous variable(s) stored in
<code>colData(spe)</code> to visualize. For each ID, if <code>rowData(spe)$gene_search</code> is
missing, then <code>rownames(spe)</code> is used to search for the gene ID. When a
vector of length &gt; 1 is supplied, the continuous variables are combined
according to <code>multi_gene_method</code>, producing a single value for each spot.</p></dd>


<dt id="arg-pdf-file">pdf_file<a class="anchor" aria-label="anchor" href="#arg-pdf-file"></a></dt>
<dd><p>A <code>character(1)</code> specifying the path for the resulting PDF.</p></dd>


<dt id="arg-assayname">assayname<a class="anchor" aria-label="anchor" href="#arg-assayname"></a></dt>
<dd><p>The name of the <code>assays(spe)</code> to use for extracting the
gene expression data. Defaults to <code>logcounts</code>.</p></dd>


<dt id="arg-mincount">minCount<a class="anchor" aria-label="anchor" href="#arg-mincount"></a></dt>
<dd><p>A <code>numeric(1)</code> specifying the minimum gene expression (or
value in the continuous variable) to visualize. Values at or below this
threshold will be set to <code>NA</code>. Defaults to <code>0</code>.</p></dd>


<dt id="arg-return-plots">return_plots<a class="anchor" aria-label="anchor" href="#arg-return-plots"></a></dt>
<dd><p>A <code>logical(1)</code> indicating whether to print the plots
to a PDF or to return the list of plots that you can then print using
<a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a>.</p></dd>


<dt id="arg-spatial">spatial<a class="anchor" aria-label="anchor" href="#arg-spatial"></a></dt>
<dd><p>A <code>logical(1)</code> indicating whether to include the histology
layer from <code><a href="geom_spatial.html">geom_spatial()</a></code>. If you plan to use
<a href="https://rdrr.io/pkg/plotly/man/ggplotly.html" class="external-link">ggplotly()</a> then it's best to set this to <code>FALSE</code>.</p></dd>


<dt id="arg-viridis">viridis<a class="anchor" aria-label="anchor" href="#arg-viridis"></a></dt>
<dd><p>A <code>logical(1)</code> whether to use the color-blind friendly
palette from <a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a> or the color palette used
in the paper that was chosen for contrast when visualizing the data on
top of the histology image. One issue is being able to differentiate low
values from NA ones due to the purple-ish histology information that is
dependent on cell density.</p></dd>


<dt id="arg-height">height<a class="anchor" aria-label="anchor" href="#arg-height"></a></dt>
<dd><p>A <code>numeric(1)</code> passed to <a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a>.</p></dd>


<dt id="arg-width">width<a class="anchor" aria-label="anchor" href="#arg-width"></a></dt>
<dd><p>A <code>numeric(1)</code> passed to <a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a>.</p></dd>


<dt id="arg-image-id">image_id<a class="anchor" aria-label="anchor" href="#arg-image-id"></a></dt>
<dd><p>A <code>character(1)</code> with the name of the image ID you want to
use in the background.</p></dd>


<dt id="arg-alpha">alpha<a class="anchor" aria-label="anchor" href="#arg-alpha"></a></dt>
<dd><p>A <code>numeric(1)</code> in the <code>[0, 1]</code> range that specifies the
transparency level of the data on the spots.</p></dd>


<dt id="arg-cont-colors">cont_colors<a class="anchor" aria-label="anchor" href="#arg-cont-colors"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> vector of colors that supersedes the
<code>viridis</code> argument.</p></dd>


<dt id="arg-sample-order">sample_order<a class="anchor" aria-label="anchor" href="#arg-sample-order"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> with the names of the samples to use
and their order.</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>A <code>numeric(1)</code> specifying the size of the points. Defaults
to <code>1.25</code>. Some colors look better if you use <code>2</code> for instance.</p></dd>


<dt id="arg-auto-crop">auto_crop<a class="anchor" aria-label="anchor" href="#arg-auto-crop"></a></dt>
<dd><p>A <code>logical(1)</code> indicating whether to automatically crop
the image / plotting area, which is useful if the Visium capture area is
not centered on the image and if the image is not a square.</p></dd>


<dt id="arg-na-color">na_color<a class="anchor" aria-label="anchor" href="#arg-na-color"></a></dt>
<dd><p>A <code>character(1)</code> specifying a color for the NA values.
If you set <code>alpha = NA</code> then it's best to set <code>na_color</code> to a color that has
alpha blending already, which will make non-NA values pop up more and the NA
values will show with a lighter color. This behavior is lost when <code>alpha</code> is
set to a non-<code>NA</code> value.</p></dd>


<dt id="arg-is-stitched">is_stitched<a class="anchor" aria-label="anchor" href="#arg-is-stitched"></a></dt>
<dd><p>A <code>logical(1)</code> vector: If <code>TRUE</code>, expects a
<a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment-class</a> built
with <code>visiumStitched::build_spe()</code>.
<a href="http://research.libd.org/visiumStitched/reference/build_spe.html" class="external-link">http://research.libd.org/visiumStitched/reference/build_spe.html</a>; in
particular, expects a logical colData column <code>exclude_overlapping</code>
specifying which spots to exclude from the plot. Sets <code>auto_crop = FALSE</code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Passed to <a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0()</a> for making the title of the
plot following the <code>sampleid</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list of <a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2</a> objects.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function prepares the data and then loops through
<code><a href="vis_gene.html">vis_gene()</a></code> for computing the list of <a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2</a>
objects.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other Spatial gene visualization functions:
<code><a href="vis_gene.html">vis_gene</a>()</code>,
<code><a href="vis_gene_p.html">vis_gene_p</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="enough_ram.html">enough_ram</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="co">## Obtain the necessary data</span></span></span>
<span class="r-in"><span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"spe"</span><span class="op">)</span><span class="op">)</span> <span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="fetch_data.html">fetch_data</a></span><span class="op">(</span><span class="st">"spe"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Subset to two samples of interest and obtain the plot list</span></span></span>
<span class="r-in"><span>    <span class="va">p_list</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>        <span class="fu">vis_grid_gene</span><span class="op">(</span></span></span>
<span class="r-in"><span>            <span class="va">spe</span><span class="op">[</span>, <span class="va">spe</span><span class="op">$</span><span class="va">sample_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"151673"</span>, <span class="st">"151674"</span><span class="op">)</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>            spatial <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>            return_plots <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>        <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Visualize the spatial adjacent replicates for position = 0 micro meters</span></span></span>
<span class="r-in"><span>    <span class="co">## for subject 3</span></span></span>
<span class="r-in"><span>    <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">p_list</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-07-19 16:15:31.676291 loading file /github/home/.cache/R/BiocFileCache/d681608cce0_Human_DLPFC_Visium_processedData_sce_scran_spatialLIBD.Rdata%3Fdl%3D1</span>
<span class="r-plt img"><img src="vis_grid_gene-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

