<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Sample spatial gene visualization — vis_gene • spatialLIBD</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Sample spatial gene visualization — vis_gene" />
<meta property="og:description" content="This function visualizes the gene expression stored in assays(spe) or any
continuous variable stored in colData(spe) for one given sample at the
spot-level using (by default) the histology information on the background.
To visualize clusters (or any discrete variable) use vis_clus()." />
<meta property="og:image" content="/logo.png" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->




  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Sample spatial gene visualization</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/spatialLIBD/blob/master/R/vis_gene.R'><code>R/vis_gene.R</code></a></small>
    <div class="hidden name"><code>vis_gene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function visualizes the gene expression stored in <code>assays(spe)</code> or any
continuous variable stored in <code>colData(spe)</code> for one given sample at the
spot-level using (by default) the histology information on the background.
To visualize clusters (or any discrete variable) use <code><a href='vis_clus.html'>vis_clus()</a></code>.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>vis_gene</span><span class='op'>(</span>
  <span class='va'>spe</span>,
  <span class='va'>sampleid</span>,
  geneid <span class='op'>=</span> <span class='st'>"SCGB2A2; ENSG00000110484"</span>,
  spatial <span class='op'>=</span> <span class='cn'>TRUE</span>,
  assayname <span class='op'>=</span> <span class='st'>"logcounts"</span>,
  minCount <span class='op'>=</span> <span class='fl'>0</span>,
  viridis <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>spe</th>
      <td><p>Defaults to the output of
<code>fetch_data(type = 'spe')</code>. This is a
<a href='https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html'>SpatialExperiment-class</a>
object with the spot-level Visium data and information required for
visualizing the histology. See <code><a href='fetch_data.html'>fetch_data()</a></code> for more details.</p></td>
    </tr>
    <tr>
      <th>sampleid</th>
      <td><p>A <code>character(1)</code> specifying which sample to plot from
<code>colData(spe)$sample_name</code>.</p></td>
    </tr>
    <tr>
      <th>geneid</th>
      <td><p>A <code>character(1)</code> specifying the gene ID stored in
<code>rowData(spe)$gene_search</code> or a continuous variable stored in <code>colData(spe)</code>
to visualize.</p></td>
    </tr>
    <tr>
      <th>spatial</th>
      <td><p>A <code>logical(1)</code> indicating whether to include the histology
layer from <code><a href='geom_spatial.html'>geom_spatial()</a></code>. If you plan to use
<a href='https://docs.ropensci.org/plotly/reference/ggplotly.html'>ggplotly()</a> then it's best to set this to <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>assayname</th>
      <td><p>The name of the <code>assays(spe)</code> to use for extracting the
gene expression data. Defaults to <code>logcounts</code>.</p></td>
    </tr>
    <tr>
      <th>minCount</th>
      <td><p>A <code>numeric(1)</code> specifying the minimum gene expression (or
value in the continuous variable) to visualize. Values at or below this
threshold will be set to <code>NA</code>. Defaults to <code>0</code>.</p></td>
    </tr>
    <tr>
      <th>viridis</th>
      <td><p>A <code>logical(1)</code> whether to use the color-blind friendly
palette from <a href='https://sjmgarnier.github.io/viridisLite/reference/viridis.html'>viridis</a> or the color palette used
in the paper that was chosen for contrast when visualizing the data on
top of the histology image. One issue is being able to differentiate low
values from NA ones due to the purple-ish histology information that is
dependent on cell density.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Passed to <a href='https://rdrr.io/r/base/paste.html'>paste0()</a> for making the title of the
plot following the <code>sampleid</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot2</a> object.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function subsets <code>spe</code> to the given sample and prepares the
data and title for <code><a href='vis_gene_p.html'>vis_gene_p()</a></code>. It also adds a caption to the plot.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other Spatial gene visualization functions: 
<code><a href='vis_gene_p.html'>vis_gene_p</a>()</code>,
<code><a href='vis_grid_gene.html'>vis_grid_gene</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'></span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='enough_ram.html'>enough_ram</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>    <span class='co'>## Obtain the necessary data</span></span>
<span class='r-in'>    <span class='kw'>if</span> <span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/exists.html'>exists</a></span><span class='op'>(</span><span class='st'>"spe"</span><span class='op'>)</span><span class='op'>)</span> <span class='va'>spe</span> <span class='op'>&lt;-</span> <span class='fu'><a href='fetch_data.html'>fetch_data</a></span><span class='op'>(</span><span class='st'>"spe"</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'>    <span class='co'>## Valid `geneid` values are those in</span></span>
<span class='r-in'>    <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>spe</span><span class='op'>)</span><span class='op'>$</span><span class='va'>gene_search</span><span class='op'>)</span></span>
<span class='r-in'>    <span class='co'>## or continuous variables stored in colData(spe)</span></span>
<span class='r-in'></span>
<span class='r-in'>    <span class='co'>## Visualize a default gene on the non-viridis scale</span></span>
<span class='r-in'>    <span class='fu'>vis_gene</span><span class='op'>(</span></span>
<span class='r-in'>        spe <span class='op'>=</span> <span class='va'>spe</span>,</span>
<span class='r-in'>        sampleid <span class='op'>=</span> <span class='st'>"151507"</span>,</span>
<span class='r-in'>        viridis <span class='op'>=</span> <span class='cn'>FALSE</span></span>
<span class='r-in'>    <span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'>    <span class='co'>## Visualize a continuous variable, in this case, the ratio of chrM</span></span>
<span class='r-in'>    <span class='co'>## gene expression compared to the total expression at the spot-level</span></span>
<span class='r-in'>    <span class='fu'>vis_gene</span><span class='op'>(</span></span>
<span class='r-in'>        spe <span class='op'>=</span> <span class='va'>spe</span>,</span>
<span class='r-in'>        sampleid <span class='op'>=</span> <span class='st'>"151507"</span>,</span>
<span class='r-in'>        geneid <span class='op'>=</span> <span class='st'>"expr_chrM_ratio"</span></span>
<span class='r-in'>    <span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-05-05</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-05-14 22:57:35 loading file /github/home/.cache/R/BiocFileCache/21e220788ecc_Human_DLPFC_Visium_processedData_sce_scran_spatialLIBD.Rdata%3Fdl%3D1</span>
<span class='r-plt'><img src='vis_gene-1.png' alt='' width='700' height='433' /></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Leonardo Collado-Torres.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


