<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to spatialLIBD • spatialLIBD</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to spatialLIBD">
<meta property="og:description" content="spatialLIBD">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to spatialLIBD</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 class="author">Kristen R. Maynard</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:Kristen.Maynard@libd.org" class="email">Kristen.Maynard@libd.org</a>
      </address>
                              <h4 class="author">Andrew E. Jaffe</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:andrew.jaffe@libd.org" class="email">andrew.jaffe@libd.org</a>
      </address>
                  
            <h4 class="date">26 April 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/vignettes/spatialLIBD.Rmd"><code>vignettes/spatialLIBD.Rmd</code></a></small>
      <div class="hidden name"><code>spatialLIBD.Rmd</code></div>

    </div>

    
    
<div id="welcome" class="section level1">
<h1 class="hasAnchor">
<a href="#welcome" class="anchor"></a>Welcome</h1>
<p>Welcome to the <code>spatialLIBD</code> project! It is composed of:</p>
<ul>
<li>a <a href="https://shiny.rstudio.com/">shiny</a> web application that we are hosting at <a href="http://spatial.libd.org/spatialLIBD/">spatial.libd.org/spatialLIBD/</a> that can handle a <a href="https://github.com/LieberInstitute/spatialLIBD/issues/2">limited</a> set of concurrent users,</li>
<li>a Bioconductor package at <a href="http://bioconductor.org/packages/spatialLIBD">bioconductor.org/packages/spatialLIBD</a> (or from <a href="http://research.libd.org/spatialLIBD/">here</a>) that lets you analyze the data and run a local version of our web application (with our data or yours),</li>
<li>and a <a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">research article</a> with the scientific knowledge we drew from this dataset. The analysis code for our project is available <a href="https://github.com/LieberInstitute/HumanPilot/">here</a>.</li>
</ul>
<p>The web application allows you to browse the LIBD human dorsolateral pre-frontal cortex (DLPFC) spatial transcriptomics data generated with the 10x Genomics Visium platform. Through the <a href="https://bioconductor.org/packages/spatialLIBD">R/Bioconductor package</a> you can also download the data as well as visualize your own datasets using this web application. Please check the <a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">bioRxiv pre-print</a> for more details about this project.</p>
If you tweet about this website, the data or the R package please use the <code>#spatialLIBD</code> hashtag. You can find previous tweets that way as shown <a href="https://twitter.com/search?q=%23spatialLIBD&amp;src=typed_query">here</a>. Thank you! <a href="https://twitter.com/intent/tweet?button_hashtag=spatialLIBD&amp;ref_src=twsrc%5Etfw" class="twitter-hashtag-button" data-show-count="false">Tweet #spatialLIBD</a>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script><div id="study-design" class="section level2">
<h2 class="hasAnchor">
<a href="#study-design" class="anchor"></a>Study design</h2>
<p>As a quick overview, the data presented here is from portion of the DLPFC that spans six neuronal layers plus white matter (<strong>A</strong>) for a total of three subjects with two pairs of spatially adjacent replicates (<strong>B</strong>). Each dissection of DLPFC was designed to span all six layers plus white matter (<strong>C</strong>). Using this web application you can explore the expression of known genes such as <em>SNAP25</em> (<strong>D</strong>, a neuronal gene), <em>MOBP</em> (<strong>E</strong>, an oligodendrocyte gene), and known layer markers from mouse studies such as <em>PCP4</em> (<strong>F</strong>, a known layer 5 marker gene).</p>
<p><img src="http://research.libd.org/spatialLIBD/reference/figures/paper_figure1.jpg" align="center" width="800px"></p>
</div>
<div id="shiny-website-mirrors" class="section level2">
<h2 class="hasAnchor">
<a href="#shiny-website-mirrors" class="anchor"></a>Shiny website mirrors</h2>
<ul>
<li><a href="http://spatial.libd.org/spatialLIBD">Main shiny application website</a></li>
<li><a href="https://jhubiostatistics.shinyapps.io/spatialLIBD/">Shinyapps</a></li>
<li><a href="https://jhubiostatistics.shinyapps.io/spatialLIBD_mirror01/">Shinyapps Mirror 1</a></li>
<li><a href="https://jhubiostatistics.shinyapps.io/spatialLIBD_mirror02/">Shinyapps Mirror 2</a></li>
</ul>
</div>
</div>
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-spatiallibd" class="section level2">
<h2 class="hasAnchor">
<a href="#install-spatiallibd" class="anchor"></a>Install <code>spatialLIBD</code>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> is a <code>R</code> package available via Bioconductor. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) {
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
  }

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"spatialLIBD"</span>)

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span>()</pre></body></html></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> <a id="cite-Collado-Torres_2020"></a>(<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with single cell RNA sequencing data, visualization functions, and interactive data exploration. That is, packages like <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> that allow you to store the data, <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> and <em><a href="https://CRAN.R-project.org/package=plotly">plotly</a></em> for visualizing the data, and <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> for building an interactive interface. A <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> user who only accesses the web application is not expected to deal with those packages directly. A <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> user will need to be familiar with <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> and <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> to understand the data provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> or the graphical results <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> provides. Furthermore, it’ll be useful for the user to know about <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> and <em><a href="https://CRAN.R-project.org/package=plotly">plotly</a></em> if you wish to adapt the web application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help regarding Bioconductor. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-spatiallibd" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-spatiallibd" class="anchor"></a>Citing <code>spatialLIBD</code>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> will be useful for your research. Please use the following information to cite the package and the research article describing the data provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em>. Thank you!</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"spatialLIBD"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Collado-Torres L, Maynard KR, Jaffe AE (2020). _LIBD Visium spatial</span>
<span class="co">#&gt; transcriptomics human pilot data inspector_. doi:</span>
<span class="co">#&gt; 10.18129/B9.bioc.spatialLIBD (URL:</span>
<span class="co">#&gt; https://doi.org/10.18129/B9.bioc.spatialLIBD),</span>
<span class="co">#&gt; https://github.com/LieberInstitute/spatialLIBD - R package version</span>
<span class="co">#&gt; 0.99.15, &lt;URL: http://www.bioconductor.org/packages/spatialLIBD&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Maynard KR, Collado-Torres L, Weber LM, Uytingco C, Barry BK, Williams</span>
<span class="co">#&gt; SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y, Kleinman JE,</span>
<span class="co">#&gt; Hyde TM, Rao N, Hicks SC, Martinowich K, Jaffe AE (2020).</span>
<span class="co">#&gt; "Transcriptome-scale spatial gene expression in the human dorsolateral</span>
<span class="co">#&gt; prefrontal cortex." _bioRxiv_. doi: 10.1101/2020.02.28.969931 (URL:</span>
<span class="co">#&gt; https://doi.org/10.1101/2020.02.28.969931), &lt;URL:</span>
<span class="co">#&gt; https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span>
<span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span>
<span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></pre></body></html></div>
</div>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) package was developed for analyzing the human dorsolateral prefrontal cortex (DLPFC) spatial transcriptomics data generated with the 10x Genomics Visium technology by researchers at the Lieber Institute for Brain Development (LIBD) <a id="cite-Maynard_2020"></a>(<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>). An initial <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application was developed for interactively exploring this data and for assigning human brain layer labels to the each spot for each sample generated. While this was useful enough for our project, we made this <a href="http://bioconductor.org/">Bioconductor</a> package in case you want to:</p>
<ul>
<li>access our Visium data to get some data from this new technology and develop methods or infrastructure for other Visium datasets.</li>
<li>re-shape your data into what ours is structured as, then re-use the visualization functions and/or the shiny app itself.</li>
<li>want to explore our data in more detail. This can range from launching the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application locally to diving into the specifics of the data from our project (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>).</li>
</ul>
<p>In this vignette we’ll showcase how you can access the Human DLPFC LIBD Visium dataset (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>), the R functions provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>), and an overview of how you can re-shape your own Visium dataset to match the structure we used.</p>
<p>To get started, please load the <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> package.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"spatialLIBD"</span>)</pre></body></html></div>
</div>
<div id="human-dlpfc-visium-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#human-dlpfc-visium-dataset" class="anchor"></a>Human DLPFC Visium dataset</h1>
<p>The human DLPFC 10x Genomics Visium dataset analyzed by LIBD researchers and colleagues is described in detail by Maynard, Collado-Torres et al (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>). However, briefly, this dataset is composed of:</p>
<ul>
<li>Three brain subjects (all controls; two males, one female; ages 30-46; <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/visium_dlpfc_pilot_sample_metrics.tsv">details</a>).</li>
<li>Four images per subject: two spatially adjacent replicates at position 0, then two more spatially adjacent replicates 300 micrometers away.</li>
<li>Slices designed to cover layers 1 through 6 and the white matter (WM) of the dorsolateral prefrontal cortex (DLPFC).</li>
</ul>
<div id="data-specifics" class="section level2">
<h2 class="hasAnchor">
<a href="#data-specifics" class="anchor"></a>Data specifics</h2>
<p>We combined all the Visium data into a single <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> <a id="cite-Lun_2020"></a>(<a href="#bib-Lun_2020">Lun and Risso, 2020</a>) object that we typically refer to as <strong><code>sce</code></strong> <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. It has 33,538 genes (rows) and 47,681 spots (columns). This is the initial point for most of our analyses (<a href="https://github.com/LieberInstitute/HumanPilot">code available on GitHub</a>). Using <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) we manually assigned each spot across all 12 images to a layer (L1 through L6 or WM). We then compressed the spot-level data at the layer-level using a pseudo-bulking approach resulting in the <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> object we typically refer to as <strong><code>sce_layer</code></strong> <a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. We then computed for each gene t or F statistics assessing whether the gene had higher expression in a given layer compared to the rest (<code>enrichment</code>; t-stat), between one layer and another layer (<code>pairwise</code>; t-stat), or had any expression variability across all layers (<code>anova</code>; F-stat). The results from the models are stored in what we refer to as <strong><code>modeling_results</code></strong> <a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>In summary,</p>
<ul>
<li>
<code>sce</code> is the <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> object with all the spot-level data and the histology information for visualization of the data.</li>
<li>
<code>sce_layer</code> is the <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> object with the layer-level data.</li>
<li>
<code>modeling_results</code> contains the layer-level <code>enrichment</code>, <code>pairwise</code> and <code>anova</code> statistics.</li>
</ul>
</div>
<div id="downloading-the-data-with-spatiallibd" class="section level2">
<h2 class="hasAnchor">
<a href="#downloading-the-data-with-spatiallibd" class="anchor"></a>Downloading the data with <code>spatialLIBD</code>
</h2>
<p>Using <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) you can download all of these R objects. They are hosted by <a href="http://bioconductor.org/">Bioconductor</a>’s <em><a href="https://bioconductor.org/packages/3.11/ExperimentHub">ExperimentHub</a></em> <a id="cite-Morgan_2020"></a>(<a href="#bib-Morgan_2020">Morgan and Shepherd, 2020</a>) resource and you can download them using <code><a href="../reference/fetch_data.html">spatialLIBD::fetch_data()</a></code>. <code><a href="../reference/fetch_data.html">fetch_data()</a></code> will query <em><a href="https://bioconductor.org/packages/3.11/ExperimentHub">ExperimentHub</a></em> which in turn will download the data and cache it so you don’t have to download it again. If <em><a href="https://bioconductor.org/packages/3.11/ExperimentHub">ExperimentHub</a></em> is unavailable, then <code><a href="../reference/fetch_data.html">fetch_data()</a></code> has a backup option that does not cache the files <a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a> Below we obtain all of these objects.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">## Connect to ExperimentHub</span>
<span class="no">ehub</span> <span class="kw">&lt;-</span> <span class="kw pkg">ExperimentHub</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span>()
<span class="co">#&gt; using temporary cache /tmp/RtmpJfYYks/BiocFileCache</span>
<span class="co">#&gt; snapshotDate(): 2020-04-03</span></pre></body></html></div>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">## Downloa small example sce data</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span>(<span class="st">"sce"</span>)) <span class="no">sce</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span>(<span class="kw">type</span> <span class="kw">=</span> <span class="st">"sce_example"</span>, <span class="kw">eh</span> <span class="kw">=</span> <span class="no">ehub</span>)
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   sce_sub</span>

<span class="co">## If you want to download the full real data (about 2.1 GB in RAM) use:</span>
<span class="kw">if</span> (<span class="fl">FALSE</span>) {
    <span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span>(<span class="st">"sce"</span>)) <span class="no">sce</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span>(<span class="kw">type</span> <span class="kw">=</span> <span class="st">"sce"</span>, <span class="kw">eh</span> <span class="kw">=</span> <span class="no">ehub</span>)
}

<span class="co">## Query ExperimentHub and download the data</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span>(<span class="st">"sce_layer"</span>)) <span class="no">sce_layer</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span>(<span class="kw">type</span> <span class="kw">=</span> <span class="st">"sce_layer"</span>, <span class="kw">eh</span> <span class="kw">=</span> <span class="no">ehub</span>)
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   sce_layer</span>
<span class="no">modeling_results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span>(<span class="st">"modeling_results"</span>, <span class="kw">eh</span> <span class="kw">=</span> <span class="no">ehub</span>)
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   modeling_results</span></pre></body></html></div>
<p>Once you have downloaded the objects, we can explore them a little bit</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co">## spot-level data</span>
<span class="no">sce</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 33538 47681 </span>
<span class="co">#&gt; metadata(1): image</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">#&gt;   ENSG00000268674</span>
<span class="co">#&gt; rowData names(9): source type ... gene_search is_top_hvg</span>
<span class="co">#&gt; colnames(47681): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(73): barcode sample_name ... pseudobulk_UMAP_spatial</span>
<span class="co">#&gt;   markers_UMAP_spatial</span>
<span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity50 ... TSNE_perplexity80</span>
<span class="co">#&gt;   UMAP_neighbors15</span>
<span class="co">#&gt; altExpNames(0):</span>

<span class="co">## layer-level data</span>
<span class="no">sce_layer</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 22331 76 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(22331): ENSG00000243485 ENSG00000238009 ... ENSG00000278384</span>
<span class="co">#&gt;   ENSG00000271254</span>
<span class="co">#&gt; rowData names(10): source type ... is_top_hvg is_top_hvg_sce_layer</span>
<span class="co">#&gt; colnames(76): 151507_Layer1 151507_Layer2 ... 151676_Layer6 151676_WM</span>
<span class="co">#&gt; colData names(12): sample_name layer_guess ... layer_guess_reordered</span>
<span class="co">#&gt;   layer_guess_reordered_short</span>
<span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity5 ... UMAP_neighbors15 PCAsub</span>
<span class="co">#&gt; altExpNames(0):</span>

<span class="co">## list of modeling result tables</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">modeling_results</span>, <span class="no">class</span>)
<span class="co">#&gt;        anova   enrichment     pairwise </span>
<span class="co">#&gt; "data.frame" "data.frame" "data.frame"</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">modeling_results</span>, <span class="no">dim</span>)
<span class="co">#&gt;      anova enrichment pairwise</span>
<span class="co">#&gt; [1,] 22331      22331    22331</span>
<span class="co">#&gt; [2,]    10         23       65</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">modeling_results</span>, <span class="kw">function</span>(<span class="no">x</span>) {
      <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>))
  })
<span class="co">#&gt;      anova          enrichment      pairwise          </span>
<span class="co">#&gt; [1,] "f_stat_full"  "t_stat_WM"     "t_stat_WM-Layer1"</span>
<span class="co">#&gt; [2,] "p_value_full" "t_stat_Layer1" "t_stat_WM-Layer2"</span>
<span class="co">#&gt; [3,] "fdr_full"     "t_stat_Layer2" "t_stat_WM-Layer3"</span>
<span class="co">#&gt; [4,] "full_AveExpr" "t_stat_Layer3" "t_stat_WM-Layer4"</span>
<span class="co">#&gt; [5,] "f_stat_noWM"  "t_stat_Layer4" "t_stat_WM-Layer5"</span>
<span class="co">#&gt; [6,] "p_value_noWM" "t_stat_Layer5" "t_stat_WM-Layer6"</span></pre></body></html></div>
<p>The modeling statistics are in wide format, which can make some visualizations complicated. The function <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> provides a way to convert them into long format and add some useful information. Let’s do so below.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Convert to a long format the modeling results</span>
<span class="co">## This takes a few seconds to run</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(
    <span class="no">sig_genes</span> <span class="kw">&lt;-</span>
        <span class="fu"><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all</a></span>(
            <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">sce_layer</span>),
            <span class="kw">modeling_results</span> <span class="kw">=</span> <span class="no">modeling_results</span>,
            <span class="kw">sce_layer</span> <span class="kw">=</span> <span class="no">sce_layer</span>
        )
)
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;  10.641   0.883  31.717</span>

<span class="co">## Explore the result</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">sig_genes</span>)
<span class="co">#&gt; [1] "DFrame"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "S4Vectors"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">sig_genes</span>)
<span class="co">#&gt; [1] 1138881      12</span></pre></body></html></div>
</div>
</div>
<div id="interactively-explore-the-spatiallibd-data" class="section level1">
<h1 class="hasAnchor">
<a href="#interactively-explore-the-spatiallibd-data" class="anchor"></a>Interactively explore the <code>spatialLIBD</code> data</h1>
<p>Now that you have downloaded the data, you can interactively explore the data using a <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> <a id="cite-Chang_2020"></a>(<a href="https://CRAN.R-project.org/package=shiny">Chang, Cheng, Allaire, Xie, et al., 2020</a>) web application contained within <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>). To do so, use the <code><a href="../reference/run_app.html">run_app()</a></code> function as shown below:</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/interactive.html">interactive</a></span>()) {
    <span class="fu"><a href="../reference/run_app.html">run_app</a></span>(
        <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
        <span class="kw">sce_layer</span> <span class="kw">=</span> <span class="no">sce_layer</span>,
        <span class="kw">modeling_results</span> <span class="kw">=</span> <span class="no">modeling_results</span>,
        <span class="kw">sig_genes</span> <span class="kw">=</span> <span class="no">sig_genes</span>
    )
}</pre></body></html></div>
<p>The <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application allows you to browse the spot-level data and interactively label spots, as well as explore the layer-level results. Once you launch it, check the <em>Documentation</em> tab for each view for more details. In order to avoid duplicating the documentation, we provide all the details on the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application itself.</p>
<p>Though overall, this application allows you to export all static visualizations as PDF files or all interactive visualizations as PNG files, as well as all result table as CSV files. This is what produces the version you can access without any R use from your side at <a href="http://spatial.libd.org/spatialLIBD/">spatial.libd.org/spatialLIBD/</a>.</p>
</div>
<div id="spatiallibd-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#spatiallibd-functions" class="anchor"></a><code>spatialLIBD</code> functions</h1>
<p>We already covered <code><a href="../reference/fetch_data.html">fetch_data()</a></code> which allows you to download the Human DLPFC Visium data from LIBD researchers and colleagues (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>).</p>
<div id="spot-level-clusters-and-discrete-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#spot-level-clusters-and-discrete-variables" class="anchor"></a>Spot-level clusters and discrete variables</h2>
<p>With the <code>sce</code> object that contains the spot-level data, we can visualize any discrete variable such as the layers using <code><a href="../reference/sce_image_clus.html">sce_image_clus()</a></code> and related functions. These functions know where to extract and how to visualize the histology information.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">## View our LIBD layers for one sample</span>
<span class="fu"><a href="../reference/sce_image_clus.html">sce_image_clus</a></span>(
    <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
    <span class="kw">clustervar</span> <span class="kw">=</span> <span class="st">"layer_guess_reordered"</span>,
    <span class="kw">sampleid</span> <span class="kw">=</span> <span class="st">"151673"</span>,
    <span class="kw">colors</span> <span class="kw">=</span> <span class="no">libd_layer_colors</span>,
    <span class="kw">...</span> <span class="kw">=</span> <span class="st">" LIBD Layers"</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sce_image_clus-1.png" width="700"></p>
<p>Most of the variables stored in <code>sce</code> are discrete variables and as such, you can visualize them using <code><a href="../reference/sce_image_clus.html">sce_image_clus()</a></code> and <code><a href="../reference/sce_image_grid.html">sce_image_grid()</a></code> for one or more than one sample respectively.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## This is not fully precise, but gives you a rough idea</span>
<span class="co">## Some integer columns are actually continuous variables</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>), <span class="no">class</span>) <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"factor"</span>, <span class="st">"integer"</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;    12    61</span>

<span class="co">## This is more precise (one cluster has 28 unique values)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>), <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">x</span>))) <span class="kw">&lt;</span> <span class="fl">29</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;    10    63</span></pre></body></html></div>
<p>Notably, <code><a href="../reference/sce_image_clus.html">sce_image_clus()</a></code> has a <code>spatial</code> <code><a href="https://rdrr.io/r/base/logical.html">logical(1)</a></code> argument which is useful if you want to visualize the data without the histology information provided by <code><a href="../reference/geom_spatial.html">geom_spatial()</a></code> (a custom <code><a href="https://ggplot2.tidyverse.org/reference/layer.html">ggplot2::layer()</a></code>). In particular, this is useful if you then want to use <code><a href="https://rdrr.io/pkg/plotly/man/ggplotly.html">plotly::ggplotly()</a></code> or other similar functions on the resulting <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot2::ggplot()</a></code> object.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co">## View our LIBD layers for one sample</span>
<span class="co">## without spatial information</span>
<span class="fu"><a href="../reference/sce_image_clus.html">sce_image_clus</a></span>(
    <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
    <span class="kw">clustervar</span> <span class="kw">=</span> <span class="st">"layer_guess_reordered"</span>,
    <span class="kw">sampleid</span> <span class="kw">=</span> <span class="st">"151673"</span>,
    <span class="kw">colors</span> <span class="kw">=</span> <span class="no">libd_layer_colors</span>,
    <span class="kw">...</span> <span class="kw">=</span> <span class="st">" LIBD Layers"</span>,
    <span class="kw">spatial</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sce_image_clus_no_spatial-1.png" width="700"></p>
<p>Some helper functions include <code><a href="../reference/get_colors.html">get_colors()</a></code> and <code><a href="../reference/sort_clusters.html">sort_clusters()</a></code> as well as the <code>libd_layer_colors</code> object included in <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>).</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co">## Color palette designed by Lukas M. Weber with feedback from the team.</span>
<span class="no">libd_layer_colors</span>
<span class="co">#&gt;        Layer1        Layer2        Layer3        Layer4        Layer5 </span>
<span class="co">#&gt;     "#F0027F"     "#377EB8"     "#4DAF4A"     "#984EA3"     "#FFD700" </span>
<span class="co">#&gt;        Layer6            WM            NA           WM2 </span>
<span class="co">#&gt;     "#FF7F00"     "#1A1A1A" "transparent"     "#666666"</span></pre></body></html></div>
</div>
<div id="spot-level-genes-and-continuous-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#spot-level-genes-and-continuous-variables" class="anchor"></a>Spot-level genes and continuous variables</h2>
<p>Similar to <code><a href="../reference/sce_image_clus.html">sce_image_clus()</a></code>, the <code><a href="../reference/sce_image_gene.html">sce_image_gene()</a></code> family of functions use the <code>sce</code> spot-level object to visualize the gene expression or any continuous variable such as the number of cells per spots. That is, <code><a href="../reference/sce_image_gene.html">sce_image_gene()</a></code> can visualize any of the <code>assays(sce)</code> or any of the continuous variables stored in <code>colData(sce)</code>. If you want to visualize more than one sample at a time, use <code><a href="../reference/sce_image_grid_gene.html">sce_image_grid_gene()</a></code> instead. And just like <code><a href="../reference/sce_image_clus.html">sce_image_clus()</a></code>, <code><a href="../reference/sce_image_gene.html">sce_image_gene()</a></code> has a <code>spatial</code> <code><a href="https://rdrr.io/r/base/logical.html">logical(1)</a></code> argument to turn off the custom spatial <code><a href="https://ggplot2.tidyverse.org/reference/layer.html">ggplot2::layer()</a></code> produced by <code><a href="../reference/geom_spatial.html">geom_spatial()</a></code>.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co">## Available gene expression assays</span>
<span class="fu">assayNames</span>(<span class="no">sce</span>)
<span class="co">#&gt; [1] "counts"    "logcounts"</span>

<span class="co">## Not all of these make sense to visualize</span>
<span class="co">## In particular, barcode, row, col, imagerow, imagecol, key are not</span>
<span class="co">## useful to visualize.</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>))[<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>), <span class="kw">function</span>(<span class="no">x</span>) {
      <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">x</span>))
  }) <span class="kw">&gt;</span> <span class="fl">29</span>]
<span class="co">#&gt;  [1] "barcode"         "row"             "col"             "imagerow"       </span>
<span class="co">#&gt;  [5] "imagecol"        "sum_umi"         "sum_gene"        "key"            </span>
<span class="co">#&gt;  [9] "expr_chrM"       "expr_chrM_ratio"</span>

<span class="co">## Visualize a gene</span>
<span class="fu"><a href="../reference/sce_image_gene.html">sce_image_gene</a></span>(
    <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
    <span class="kw">sampleid</span> <span class="kw">=</span> <span class="st">"151673"</span>,
    <span class="kw">viridis</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sce_image_gene-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><html><body><pre class="r">
<span class="co">## Visualize the estimated number of cells per spot</span>
<span class="fu"><a href="../reference/sce_image_gene.html">sce_image_gene</a></span>(
    <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
    <span class="kw">sampleid</span> <span class="kw">=</span> <span class="st">"151673"</span>,
    <span class="kw">geneid</span> <span class="kw">=</span> <span class="st">"cell_count"</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sce_image_gene-2.png" width="700"></p>
<div class="sourceCode" id="cb15"><html><body><pre class="r">
<span class="co">## Visualize the fraction of chrM expression per spot</span>
<span class="co">## without the spatial layer</span>
<span class="fu"><a href="../reference/sce_image_gene.html">sce_image_gene</a></span>(
    <span class="kw">sce</span> <span class="kw">=</span> <span class="no">sce</span>,
    <span class="kw">sampleid</span> <span class="kw">=</span> <span class="st">"151673"</span>,
    <span class="kw">geneid</span> <span class="kw">=</span> <span class="st">"expr_chrM_ratio"</span>,
    <span class="kw">spatial</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sce_image_gene-3.png" width="700"></p>
<p>As for the color palette, you can either use the color blind friendly palette when <code>viridis = TRUE</code> or a custom palette we determined. Note that if you design your own palette, you have to take into account that values it can be hard to distinguish some colors from the histology set of purple tones that are noticeable when the continuous variable is below or equal to the <code>minCount</code> (in our palettes such points have a <code>'transparent'</code> color). For more details, check the internal code of <code><a href="../reference/sce_image_gene_p.html">sce_image_gene_p()</a></code>.</p>
</div>
<div id="extract-significant-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#extract-significant-genes" class="anchor"></a>Extract significant genes</h2>
<p>Earlier we also ran <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> in order to run the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> web application. However, we didn’t explain the output. If you explore it, you’ll notice that it’s a very long table with several columns.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sig_genes</span>)
<span class="co">#&gt; DataFrame with 6 rows and 12 columns</span>
<span class="co">#&gt;         top  model_type        test        gene      stat        pval</span>
<span class="co">#&gt;   &lt;integer&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; 1         1  enrichment          WM       NDRG1   16.3053 1.25896e-26</span>
<span class="co">#&gt; 2         2  enrichment          WM      PTP4A2   16.1469 2.25133e-26</span>
<span class="co">#&gt; 3         3  enrichment          WM        AQP1   15.9927 3.97849e-26</span>
<span class="co">#&gt; 4         4  enrichment          WM       PAQR6   15.1971 7.86258e-25</span>
<span class="co">#&gt; 5         5  enrichment          WM      ANP32B   14.9798 1.80183e-24</span>
<span class="co">#&gt; 6         6  enrichment          WM        JAM3   14.7413 4.50756e-24</span>
<span class="co">#&gt;           fdr gene_index         ensembl           in_rows       in_rows_top20</span>
<span class="co">#&gt;     &lt;numeric&gt;  &lt;integer&gt;     &lt;character&gt;     &lt;IntegerList&gt;       &lt;IntegerList&gt;</span>
<span class="co">#&gt; 1 2.51372e-22      10404 ENSG00000104419 1,40215,65023,... 1,156337,223320,...</span>
<span class="co">#&gt; 2 2.51372e-22        487 ENSG00000184007 2,40531,64532,... 2,223323,245661,...</span>
<span class="co">#&gt; 3 2.96145e-22       8201 ENSG00000240583 3,32241,65062,... 3,223314,245643,...</span>
<span class="co">#&gt; 4 4.38948e-21       1501 ENSG00000160781 4,34318,65863,...     4,245654,267982</span>
<span class="co">#&gt; 5 8.04735e-21      10962 ENSG00000136938 5,28964,63931,...     5,223329,267975</span>
<span class="co">#&gt; 6 1.67295e-20      12600 ENSG00000166086 6,40868,65917,... 6,156334,223324,...</span>
<span class="co">#&gt;                                       results</span>
<span class="co">#&gt;                               &lt;CharacterList&gt;</span>
<span class="co">#&gt; 1 WM_top1,WM-Layer1_top20,WM-Layer4_top10,...</span>
<span class="co">#&gt; 2 WM_top2,WM-Layer4_top13,WM-Layer5_top20,...</span>
<span class="co">#&gt; 3   WM_top3,WM-Layer4_top4,WM-Layer5_top2,...</span>
<span class="co">#&gt; 4     WM_top4,WM-Layer5_top13,WM-Layer6_top10</span>
<span class="co">#&gt; 5      WM_top5,WM-Layer4_top19,WM-Layer6_top3</span>
<span class="co">#&gt; 6 WM_top6,WM-Layer1_top17,WM-Layer4_top14,...</span></pre></body></html></div>
<p>The output of <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> contains the following columns:</p>
<ul>
<li>
<code>top</code>: the rank of the gene for the given <code>test</code>.</li>
<li>
<code>model_type</code>: either <code>enrichment</code>, <code>pairwise</code> or <code>anova</code>.</li>
<li>
<code>test</code>: the short notation for the test performed. For example, <code>WM</code> is white matter versus the other layers while <code>WM-Layer1</code> is white matter greater than layer 1.</li>
<li>
<code>gene</code>: the gene symbol.</li>
<li>
<code>stat</code>: the corresponding F or t-statistic.</li>
<li>
<code>pval</code>: the corresponding p-value (two-sided for t-stats).</li>
<li>
<code>fdr</code>: the FDR adjusted p-value.</li>
<li>
<code>gene_index</code>: the row of <code>sce_layer</code> and the original tables in <code>modeling_results</code> to join the tables if necessary.</li>
<li>
<code>ensembl</code>: Ensembl gene ID.</li>
<li>
<code>in_rows</code>: an <code>IntegerList()</code> specifying all the rows where that gene is present.</li>
<li>
<code>in_rows_top20</code>: an <code>IntegerList()</code> specifying all the rows where that gene is present and where its rank (<code>top</code>) is less than or equal to 20. This information is only included for the first occurrence of each gene if that gene is on the top 20 rank for any of the models.</li>
<li>
<code>results</code>: an <code>CharacterList()</code> specifying all the <code>test</code> results where the gene is ranked (<code>top</code>) among the top 20 genes.</li>
</ul>
<p><code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> uses <code><a href="../reference/sig_genes_extract.html">sig_genes_extract()</a></code> as it’s workhorse and as such has a very similar output.</p>
</div>
<div id="visualize-modeling-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-modeling-results" class="anchor"></a>Visualize modeling results</h2>
<p>After extracting the table of modeling results in long format with <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code>, we can then use <code><a href="../reference/layer_boxplot.html">layer_boxplot()</a></code> to visualize any gene of interest for any of the model types and tests we performed. Below we explore the first one (by default). We also show the color palettes used in the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>). The last example has the long title version that uses more information from the <code>sig_genes</code> object we created earlier.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="co">## Note that we recommend setting the random seed so the jittering of the</span>
<span class="co">## points will be reproducible. Given the requirements by BiocCheck this</span>
<span class="co">## cannot be done inside the layer_boxplot() function.</span>

<span class="co">## Create a boxplot of the first gene in `sig_genes`.</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">20200206</span>)
<span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span>(<span class="kw">sig_genes</span> <span class="kw">=</span> <span class="no">sig_genes</span>, <span class="kw">sce_layer</span> <span class="kw">=</span> <span class="no">sce_layer</span>)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-1.png" width="700"></p>
<div class="sourceCode" id="cb18"><html><body><pre class="r">
<span class="co">## Viridis colors displayed in the shiny app</span>
<span class="co">## showing the first pairwise model result</span>
<span class="co">## (which illustrates the background colors used for the layers not</span>
<span class="co">## involved in the pairwise comparison)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">20200206</span>)
<span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span>(
    <span class="kw">i</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">sig_genes</span>$<span class="no">model_type</span> <span class="kw">==</span> <span class="st">"pairwise"</span>)[<span class="fl">1</span>],
    <span class="kw">sig_genes</span> <span class="kw">=</span> <span class="no">sig_genes</span>,
    <span class="kw">sce_layer</span> <span class="kw">=</span> <span class="no">sce_layer</span>,
    <span class="kw">col_low_box</span> <span class="kw">=</span> <span class="kw pkg">viridisLite</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span>(<span class="fl">4</span>)[<span class="fl">2</span>],
    <span class="kw">col_low_point</span> <span class="kw">=</span> <span class="kw pkg">viridisLite</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span>(<span class="fl">4</span>)[<span class="fl">1</span>],
    <span class="kw">col_high_box</span> <span class="kw">=</span> <span class="kw pkg">viridisLite</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span>(<span class="fl">4</span>)[<span class="fl">3</span>],
    <span class="kw">col_high_point</span> <span class="kw">=</span> <span class="kw pkg">viridisLite</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span>(<span class="fl">4</span>)[<span class="fl">4</span>]
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-2.png" width="700"></p>
<div class="sourceCode" id="cb19"><html><body><pre class="r">
<span class="co">## Paper colors displayed in the shiny app</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">20200206</span>)
<span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span>(
    <span class="kw">sig_genes</span> <span class="kw">=</span> <span class="no">sig_genes</span>,
    <span class="kw">sce_layer</span> <span class="kw">=</span> <span class="no">sce_layer</span>,
    <span class="kw">short_title</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">col_low_box</span> <span class="kw">=</span> <span class="st">"palegreen3"</span>,
    <span class="kw">col_low_point</span> <span class="kw">=</span> <span class="st">"springgreen2"</span>,
    <span class="kw">col_high_box</span> <span class="kw">=</span> <span class="st">"darkorange2"</span>,
    <span class="kw">col_high_point</span> <span class="kw">=</span> <span class="st">"orange1"</span>
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-3.png" width="700"></p>
</div>
<div id="correlation-of-layer-level-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#correlation-of-layer-level-statistics" class="anchor"></a>Correlation of layer-level statistics</h2>
<p>Just like we compressed our <code>sce</code> object into <code>sce_layer</code> by pseudo-bulking <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>, we can do the same for other single nucleus or single cell RNA sequencing datasets (snRNA-seq, scRNA-seq) and then compute enrichment t-statistics (one group vs the rest; could be one cell type vs the rest or one cluster of cells vs the rest). In our analysis (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>), we did this for several datasets including one of our LIBD Human DLPFC snRNA-seq data generated by Matthew N Tran et al <a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>. In <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) we include a small set of these statistics for the 31 cell clusters identified by Matthew N Tran et al.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="co">## Explore the enrichment t-statistics derived from Tran et al's snRNA-seq</span>
<span class="co">## DLPFC data</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>)
<span class="co">#&gt; [1] 692  31</span>
<span class="no">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>[<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">3</span>), <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">6</span>)]
<span class="co">#&gt;                      2 (1)       4 (1)      6 (1)      8 (1)     10 (1)</span>
<span class="co">#&gt; ENSG00000104419 -0.6720726  0.30024609  0.5854130  0.1358497  0.3981996</span>
<span class="co">#&gt; ENSG00000184007  0.2535838 -0.28998049 -1.8106052 -0.2096309 -0.4521719</span>
<span class="co">#&gt; ENSG00000240583 -0.2015831 -0.04053423 -0.5120807  0.1688405 -0.3969257</span>
<span class="co">#&gt;                     12 (1)</span>
<span class="co">#&gt; ENSG00000104419 -0.9731938</span>
<span class="co">#&gt; ENSG00000184007 -0.1840286</span>
<span class="co">#&gt; ENSG00000240583 -0.2086954</span></pre></body></html></div>
<p>The function <code><a href="../reference/layer_stat_cor.html">layer_stat_cor()</a></code> will take as input one such matrix of statistics and correlate them against our layer enrichment results (or other model types) using the subset of Ensembl gene IDs that are observed in both tables.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="co">## Compute the correlation matrix of enrichment t-statistics between our data</span>
<span class="co">## and Tran et al's snRNA-seq data</span>
<span class="no">cor_stats_layer</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/layer_stat_cor.html">layer_stat_cor</a></span>(
    <span class="no">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>,
    <span class="no">modeling_results</span>,
    <span class="st">"enrichment"</span>
)

<span class="co">## Explore the correlation matrix</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">cor_stats_layer</span>[, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">3</span>)])
<span class="co">#&gt;               WM       Layer6     Layer5</span>
<span class="co">#&gt; 22 (3) 0.6824669 -0.009192291 -0.1934265</span>
<span class="co">#&gt; 3 (3)  0.7154122 -0.070042729 -0.2290574</span>
<span class="co">#&gt; 23 (3) 0.6637885 -0.031467704 -0.2018306</span>
<span class="co">#&gt; 17 (3) 0.6364983 -0.094216046 -0.2026147</span>
<span class="co">#&gt; 21 (3) 0.6281443 -0.050336358 -0.1988774</span>
<span class="co">#&gt; 7 (4)  0.1850724 -0.197283175 -0.2716890</span></pre></body></html></div>
<p>Once we have computed this correlation matrix, we can then visualize it using <code><a href="../reference/layer_stat_cor_plot.html">layer_stat_cor_plot()</a></code> as shown below.</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="co">## Visualize the correlation matrix</span>
<span class="fu"><a href="../reference/layer_stat_cor_plot.html">layer_stat_cor_plot</a></span>(<span class="no">cor_stats_layer</span>, <span class="kw">max</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">cor_stats_layer</span>))</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/layer_stat_cor_plot-1.png" width="700"></p>
<p>In order to fully interpret the resulting heatmap you need to know what each of the cell clusters labels mean. In this case, the syntax is <code>xx (Y)</code> where <code>xx</code> is the cluster number and <code>Y</code> is:</p>
<ol style="list-style-type: decimal">
<li>
<code>Excit</code>: excitatory neurons.</li>
<li>
<code>Inhib</code>: inhibitory neurons.</li>
<li>
<code>Oligo</code>: oligodendrocytes.</li>
<li>
<code>Astro</code>: astrocytes.</li>
<li>
<code>OPC</code>: oligodendrocyte progenitor cells.</li>
<li>
<code>Drop</code>: an ambiguous cluster of cells that potentially should be dropped.</li>
</ol>
<p>You can find the version with the full names <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/Layer_Guesses/pdf/snRNAseq_overlap_heatmap.pdf">here</a> if you are interested in it.</p>
<p>The <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) allows users to upload CSV file with these t-statistics, view the correlation heatmaps, download them, and download the correlation matrix. An example CSV file is provided <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/data-raw/tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer.csv">here</a>.</p>
</div>
<div id="gene-set-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-enrichment" class="anchor"></a>Gene set enrichment</h2>
<p>Many researchers have identified lists of genes that increase the risk of a given disorder or disease, are differentially expressed in a given experiment or set of conditions, have been described in a several research papers, among other collections. We can ask for any of our modeling results whether a list of genes is enriched among our significant results.</p>
<p>For illustration purposes, we included in <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) a set of genes from the Simons Foundation Autism Research Initiative <a href="https://www.sfari.org/">SFARI</a>. In our analysis (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>) we used more gene lists. Below, we read in the data and create a <code><a href="https://rdrr.io/r/base/list.html">list()</a></code> object with the Ensembl gene IDs that SFARI has provided that are related to autism.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="co">## Read in the SFARI gene sets included in the package</span>
<span class="no">asd_sfari</span> <span class="kw">&lt;-</span> <span class="kw pkg">utils</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(
        <span class="st">"extdata"</span>,
        <span class="st">"SFARI-Gene_genes_01-03-2020release_02-04-2020export.csv"</span>,
        <span class="kw">package</span> <span class="kw">=</span> <span class="st">"spatialLIBD"</span>
    ),
    <span class="kw">as.is</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)

<span class="co">## Format them appropriately</span>
<span class="no">asd_sfari_geneList</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    <span class="kw">Gene_SFARI_all</span> <span class="kw">=</span> <span class="no">asd_sfari</span>$<span class="no">ensembl.id</span>,
    <span class="kw">Gene_SFARI_high</span> <span class="kw">=</span> <span class="no">asd_sfari</span>$<span class="no">ensembl.id</span>[<span class="no">asd_sfari</span>$<span class="no">gene.score</span> <span class="kw">&lt;</span> <span class="fl">3</span>],
    <span class="kw">Gene_SFARI_syndromic</span> <span class="kw">=</span> <span class="no">asd_sfari</span>$<span class="no">ensembl.id</span>[<span class="no">asd_sfari</span>$<span class="no">syndromic</span> <span class="kw">==</span> <span class="fl">1</span>]
)</pre></body></html></div>
<p>After reading the list of genes, we can then compute the enrichment odds ratios and p-values for a given FDR threshold in our statistics.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="co">## Compute the gene set enrichment results</span>
<span class="no">asd_sfari_enrichment</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/gene_set_enrichment.html">gene_set_enrichment</a></span>(
    <span class="kw">gene_list</span> <span class="kw">=</span> <span class="no">asd_sfari_geneList</span>,
    <span class="kw">modeling_results</span> <span class="kw">=</span> <span class="no">modeling_results</span>,
    <span class="kw">model_type</span> <span class="kw">=</span> <span class="st">"enrichment"</span>
)

<span class="co">## Explore the results</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">asd_sfari_enrichment</span>)
<span class="co">#&gt;          OR       Pval   test                   ID model_type fdr_cut</span>
<span class="co">#&gt; 1 1.2659915 0.00318733     WM       Gene_SFARI_all enrichment     0.1</span>
<span class="co">#&gt; 2 1.1819109 0.17753833     WM      Gene_SFARI_high enrichment     0.1</span>
<span class="co">#&gt; 3 1.2333378 0.31971056     WM Gene_SFARI_syndromic enrichment     0.1</span>
<span class="co">#&gt; 4 0.9702022 0.85164140 Layer1       Gene_SFARI_all enrichment     0.1</span>
<span class="co">#&gt; 5 0.7192630 0.14735979 Layer1      Gene_SFARI_high enrichment     0.1</span>
<span class="co">#&gt; 6 1.1216176 0.73790069 Layer1 Gene_SFARI_syndromic enrichment     0.1</span></pre></body></html></div>
<p>Using the above enrichment table, we can then visualize the odds ratios on a heatmap as shown below. Note that we use the thresholded p-values at <code>-log10(p) = 12</code> for visualization purposes and only show the odds ratios for <code>-log10(p) &gt; 3</code> by default.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="co">## Visualize gene set enrichment results</span>
<span class="fu"><a href="../reference/gene_set_enrichment_plot.html">gene_set_enrichment_plot</a></span>(
    <span class="no">asd_sfari_enrichment</span>,
    <span class="kw">xlabs</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".*_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">asd_sfari_enrichment</span>$<span class="no">ID</span>)),
    <span class="kw">layerHeights</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">40</span>, <span class="fl">55</span>, <span class="fl">75</span>, <span class="fl">85</span>, <span class="fl">110</span>, <span class="fl">120</span>, <span class="fl">135</span>)
)</pre></body></html></div>
<p><img src="spatialLIBD_files/figure-html/sfari_enrichment_plot-1.png" width="700"></p>
<p>The <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) allows users to upload CSV file their gene lists, compute the enrichment statistics, visualize them, download the PDF, and download the enrichment table. An example CSV file is provided <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/data-raw/asd_sfari_geneList.csv">here</a>.</p>
</div>
</div>
<div id="re-shaping-your-data-to-our-structure" class="section level1">
<h1 class="hasAnchor">
<a href="#re-shaping-your-data-to-our-structure" class="anchor"></a>Re-shaping your data to our structure</h1>
<p>This section gets into the details of how we generated the data (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>) behind <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>). This could be useful if you are a Bioconductor developer or a user very familiar with packages such as <em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em>.</p>
<div id="generating-our-data" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-our-data" class="anchor"></a>Generating our data</h2>
<p>If you are interested in reshaping your data to fit our structure, we do not provide a quick function to do so. That is intentional given the active development by the Bioconductor community for determining the best way to deal with spatial transcriptomics data in general and the 10x Visium data in particular. Having said that, we do have all the steps and reproducibility information documented across several of our analysis (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>) scripts.</p>
<ul>
<li>
<code>reorganize_folder.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/reorganize_folder.R">here</a> re-organizes the raw data we were sent by 10x Genomics.</li>
<li>
<code>Layer_Notebook.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/Layer_Notebook.R">here</a> reads in the Visium data and builds a list of <code>RangeSummarizedExperiment()</code> objects from <em><a href="https://bioconductor.org/packages/3.11/SummarizedExperiment">SummarizedExperiment</a></em>, one per sample (image) that is eventually saved as <code>Human_DLPFC_Visium_processedData_rseList.rda</code>.</li>
<li>
<code>convert_sce.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/convert_sce.R">here</a> reads in <code>Human_DLPFC_Visium_processedData_rseList.rda</code> and builds an initial <code>sce</code> object with image data under <code>metadata(sce)$image</code> which is a single data.frame. Subsetting doesn’t automatically subset the image, so you have to do it yourself when plotting as is done by <code><a href="../reference/sce_image_clus_p.html">sce_image_clus_p()</a></code> and <code><a href="../reference/sce_image_gene_p.html">sce_image_gene_p()</a></code>. Having the data from all images in a single object allows you to use the spot-level data from all images to compute clusters and do other similar analyses to the ones you would do with sc/snRNA-seq data. The script creates the <code>Human_DLPFC_Visium_processedData_sce.Rdata</code> file.</li>
<li>
<code>sce_scran.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/sce_scran.R">here</a> then uses <em><a href="https://bioconductor.org/packages/3.11/scran">scran</a></em> to read in <code>Human_DLPFC_Visium_processedData_sce.Rdata</code>, compute the highly variable genes (stored in our final <code>sce</code> object at <code>rowData(sce)$is_top_hvg</code>), perform dimensionality reduction (PCA, TSNE, UMAP) and identify clusters using the data from all images. The resulting data is then stored as <code>Human_DLPFC_Visium_processedData_sce_scran.Rdata</code> and is the main object used throughout our analysis code (<a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1">Maynard, Collado-Torres, Weber, Uytingco, et al., 2020</a>).</li>
<li>
<code>make-data_spatialLIBD.R</code> available in the source version of <code>spatialLIBD</code> and <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/inst/scripts/make-data_spatialLIBD.R">online here</a> is the script that reads in <code>Human_DLPFC_Visium_processedData_sce_scran.Rdata</code> as well as some other outputs from our analysis and combines them into the final <code>sce</code> and <code>sce_layer</code> objects provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>). This script simplifies some operations in order to simplify the code behind the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em>.</li>
</ul>
<p>You don’t necessarily need to do all of this to use the functions provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em>. Note that external to the R objects, for the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application provided by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> you will need to have the <code>tissue_lowres_image.png</code> image files in a directory structure by sample as shown <a href="https://github.com/LieberInstitute/spatialLIBD/tree/master/inst/app/www/data">here</a> in order for the interactive visualizations made with <em><a href="https://CRAN.R-project.org/package=plotly">plotly</a></em> to work.</p>
</div>
<div id="expected-characteristics-of-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#expected-characteristics-of-the-data" class="anchor"></a>Expected characteristics of the data</h2>
<p>Here is a summary some of the key characteristics required by <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em>’s functions or the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> application.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="co">## Images data stored under metadata(sce)$image</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">metadata</span>(<span class="no">sce</span>)$<span class="no">image</span>)
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;   sample grob       height width</span>
<span class="co">#&gt;    &lt;int&gt; &lt;list&gt;      &lt;int&gt; &lt;int&gt;</span>
<span class="co">#&gt; 1 151507 &lt;rastrgrb&gt;    600   600</span>
<span class="co">#&gt; 2 151508 &lt;rastrgrb&gt;    600   600</span>
<span class="co">#&gt; 3 151509 &lt;rastrgrb&gt;    600   600</span>
<span class="co">#&gt; 4 151510 &lt;rastrgrb&gt;    600   600</span>
<span class="co">#&gt; 5 151669 &lt;rastrgrb&gt;    600   600</span>
<span class="co">#&gt; 6 151670 &lt;rastrgrb&gt;    600   600</span>

<span class="co">## Ensembl gene IDs are row names with the symbol (gene_name) and Ensembl</span>
<span class="co">## ID (gene_name) pasted into `gene_search`</span>
<span class="co">## gene_search &lt;- paste0(gene_name, '; ', gene_id)</span>
<span class="fu">rowData</span>(<span class="no">sce</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"gene_search"</span>)]
<span class="co">#&gt; DataFrame with 33538 rows and 3 columns</span>
<span class="co">#&gt;                         gene_id   gene_name                  gene_search</span>
<span class="co">#&gt;                     &lt;character&gt; &lt;character&gt;                  &lt;character&gt;</span>
<span class="co">#&gt; ENSG00000243485 ENSG00000243485 MIR1302-2HG MIR1302-2HG; ENSG00000243485</span>
<span class="co">#&gt; ENSG00000237613 ENSG00000237613     FAM138A     FAM138A; ENSG00000237613</span>
<span class="co">#&gt; ENSG00000186092 ENSG00000186092       OR4F5       OR4F5; ENSG00000186092</span>
<span class="co">#&gt; ENSG00000238009 ENSG00000238009  AL627309.1  AL627309.1; ENSG00000238009</span>
<span class="co">#&gt; ENSG00000239945 ENSG00000239945  AL627309.3  AL627309.3; ENSG00000239945</span>
<span class="co">#&gt; ...                         ...         ...                          ...</span>
<span class="co">#&gt; ENSG00000277856 ENSG00000277856  AC233755.2  AC233755.2; ENSG00000277856</span>
<span class="co">#&gt; ENSG00000275063 ENSG00000275063  AC233755.1  AC233755.1; ENSG00000275063</span>
<span class="co">#&gt; ENSG00000271254 ENSG00000271254  AC240274.1  AC240274.1; ENSG00000271254</span>
<span class="co">#&gt; ENSG00000277475 ENSG00000277475  AC213203.1  AC213203.1; ENSG00000277475</span>
<span class="co">#&gt; ENSG00000268674 ENSG00000268674     FAM231C     FAM231C; ENSG00000268674</span>
<span class="co">## This information also has to be present in sce_layer</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"gene_search"</span>) <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">rowData</span>(<span class="no">sce_layer</span>))
))


<span class="co">## Information about the image coordinates stored in imagerow and imagecol</span>
<span class="fu">colData</span>(<span class="no">sce</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"imagerow"</span>, <span class="st">"imagecol"</span>)]
<span class="co">#&gt; DataFrame with 47681 rows and 2 columns</span>
<span class="co">#&gt;                     imagerow  imagecol</span>
<span class="co">#&gt;                    &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; AAACAACGAATAGTTC-1   113.141   147.435</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1   383.438   413.051</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1   129.523   231.008</span>
<span class="co">#&gt; AAACACCAATAACTGC-1   431.188   155.806</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1   344.869   125.068</span>
<span class="co">#&gt; ...                      ...       ...</span>
<span class="co">#&gt; TTGTTGTGTGTCAAGA-1   287.039   357.606</span>
<span class="co">#&gt; TTGTTTCACATCCAGG-1   431.773   248.065</span>
<span class="co">#&gt; TTGTTTCATTAGTCTA-1   442.259   210.801</span>
<span class="co">#&gt; TTGTTTCCATACAACT-1   361.341   202.115</span>
<span class="co">#&gt; TTGTTTGTGTAAATTC-1   157.021   277.993</span>

<span class="co">## The sample names stored under sce$sample_name</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="st">"sample_name"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>)))

<span class="co">## A unique spot-level ID (such as barcode) stored under sce$key</span>
<span class="co">## The 'key' column is necessary for the plotly code to work.</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="st">"key"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>)))
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">sce</span>$<span class="no">key</span>)) <span class="kw">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">sce</span>))
<span class="co">## Here's how we made those unique keys</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">sce</span>$<span class="no">key</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">sce</span>$<span class="no">sample_name</span>, <span class="st">"_"</span>, <span class="no">sce</span>$<span class="no">barcode</span>))
<span class="co">#&gt; [1] TRUE</span>

<span class="co">## Some cluster variables that you can specify with run_app().</span>
<span class="co">## For example:</span>
<span class="no">cluster_variables</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"GraphBased"</span>,
    <span class="st">"Layer"</span>,
    <span class="st">"Maynard"</span>,
    <span class="st">"Martinowich"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"SNN_k50_k"</span>, <span class="fl">4</span>:<span class="fl">28</span>),
    <span class="st">"layer_guess_reordered_short"</span>,
    <span class="st">"SpatialDE_PCA"</span>,
    <span class="st">"SpatialDE_pool_PCA"</span>,
    <span class="st">"HVG_PCA"</span>,
    <span class="st">"pseudobulk_PCA"</span>,
    <span class="st">"markers_PCA"</span>,
    <span class="st">"SpatialDE_UMAP"</span>,
    <span class="st">"SpatialDE_pool_UMAP"</span>,
    <span class="st">"HVG_UMAP"</span>,
    <span class="st">"pseudobulk_UMAP"</span>,
    <span class="st">"markers_UMAP"</span>,
    <span class="st">"SpatialDE_PCA_spatial"</span>,
    <span class="st">"SpatialDE_pool_PCA_spatial"</span>,
    <span class="st">"HVG_PCA_spatial"</span>,
    <span class="st">"pseudobulk_PCA_spatial"</span>,
    <span class="st">"markers_PCA_spatial"</span>,
    <span class="st">"SpatialDE_UMAP_spatial"</span>,
    <span class="st">"SpatialDE_pool_UMAP_spatial"</span>,
    <span class="st">"HVG_UMAP_spatial"</span>,
    <span class="st">"pseudobulk_UMAP_spatial"</span>,
    <span class="st">"markers_UMAP_spatial"</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="no">cluster_variables</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>))))
<span class="co">## The last one gets renamed to spatialLIBD in spatialLIBD::app_server()</span>

<span class="co">## Some continuous variables that you can specify in run_app()</span>
<span class="co">## as well as the already mentioned rowData(sce)$gene_search</span>
<span class="co">## For example:</span>
<span class="no">continuous_variables</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"cell_count"</span>,
    <span class="st">"sum_umi"</span>,
    <span class="st">"sum_gene"</span>,
    <span class="st">"expr_chrM"</span>,
    <span class="st">"expr_chrM_ratio"</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="no">continuous_variables</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>))))

<span class="co">## None of the values of rowData(sce)$gene_search should be re-used in</span>
<span class="co">## colData(sce)</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu">rowData</span>(<span class="no">sce</span>)$<span class="no">gene_search</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>))))

<span class="co">## No column named COUNT as that's used by sce_image_gene_p() and related</span>
<span class="co">## functions.</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(!<span class="st">"COUNT"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce</span>)))

<span class="co">## The counts and logcounts assays</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"counts"</span>, <span class="st">"logcounts"</span>) <span class="kw">%in%</span> <span class="fu">assayNames</span>(<span class="no">sce</span>)))

<span class="co">## For the layer-level data we also need layer_guess_reordered</span>
<span class="co">## although this is a parameter you can change in run_app()</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="st">"layer_guess_reordered"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">colData</span>(<span class="no">sce_layer</span>)))

<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"enrichment"</span>, <span class="st">"pairwise"</span>, <span class="st">"anova"</span>) <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">modeling_results</span>)
))

<span class="co">## All the model tables contain the ensembl column</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">modeling_results</span>, <span class="kw">function</span>(<span class="no">x</span>) {
      <span class="st">"ensembl"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">x</span>)
  })))

<span class="co">## The following column prefixes are present in each of the model tables</span>
<span class="no">expected_column_name_starts</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"^[f|t]_stat_"</span>, <span class="st">"^p_value_"</span>, <span class="st">"^fdr_"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">modeling_results</span>, <span class="kw">function</span>(<span class="no">model_table</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">expected_column_name_starts</span>, <span class="kw">function</span>(<span class="no">expected_prefix</span>) {
          <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(
              <span class="no">expected_prefix</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">model_table</span>)
          ))
      }))
})))</pre></body></html></div>
<p>If you want to run these checks, you can do so using the <code>check_*</code> family of functions.</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="../reference/check_sce.html">check_sce</a></span>(<span class="no">sce</span>)
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 33538 47681 </span>
<span class="co">#&gt; metadata(1): image</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">#&gt;   ENSG00000268674</span>
<span class="co">#&gt; rowData names(9): source type ... gene_search is_top_hvg</span>
<span class="co">#&gt; colnames(47681): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span>
<span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span>
<span class="co">#&gt; colData names(73): barcode sample_name ... pseudobulk_UMAP_spatial</span>
<span class="co">#&gt;   markers_UMAP_spatial</span>
<span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity50 ... TSNE_perplexity80</span>
<span class="co">#&gt;   UMAP_neighbors15</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="fu"><a href="../reference/check_sce_layer.html">check_sce_layer</a></span>(<span class="no">sce_layer</span>)
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 22331 76 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(22331): ENSG00000243485 ENSG00000238009 ... ENSG00000278384</span>
<span class="co">#&gt;   ENSG00000271254</span>
<span class="co">#&gt; rowData names(10): source type ... is_top_hvg is_top_hvg_sce_layer</span>
<span class="co">#&gt; colnames(76): 151507_Layer1 151507_Layer2 ... 151676_Layer6 151676_WM</span>
<span class="co">#&gt; colData names(12): sample_name layer_guess ... layer_guess_reordered</span>
<span class="co">#&gt;   layer_guess_reordered_short</span>
<span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity5 ... UMAP_neighbors15 PCAsub</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">## The output here is too long to print</span>
<span class="no">xx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/check_modeling_results.html">check_modeling_results</a></span>(<span class="no">modeling_results</span>)
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">xx</span>, <span class="no">modeling_results</span>)
<span class="co">#&gt; [1] TRUE</span></pre></body></html></div>
<p>As we generate more data at LIBD, we might make <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) more general if the need arises and we have the resources to work on this.</p>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.11/spatialLIBD">spatialLIBD</a></em> package (<a href="http://www.bioconductor.org/packages/spatialLIBD">Collado-Torres, Maynard, and Jaffe, 2020</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/AnnotationHub">AnnotationHub</a></em> <a id="cite-Morgan_2020a"></a>(<a href="#bib-Morgan_2020a">Morgan and Shepherd, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=benchmarkme">benchmarkme</a></em> <a id="cite-Gillespie_2019"></a>(<a href="https://CRAN.R-project.org/package=benchmarkme">Gillespie, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=cowplot">cowplot</a></em> <a id="cite-Wilke_2019"></a>(<a href="https://CRAN.R-project.org/package=cowplot">Wilke, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=DT">DT</a></em> <a id="cite-Xie_2020"></a>(<a href="https://CRAN.R-project.org/package=DT">Xie, Cheng, and Tan, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/ExperimentHub">ExperimentHub</a></em> (<a href="#bib-Morgan_2020">Morgan and Shepherd, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=fields">fields</a></em> <a id="cite-Douglas_Nychka_2017"></a>(<a href="https://github.com/NCAR/Fields">Douglas Nychka, Reinhard Furrer, John Paige, and Stephan Sain, 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> <a id="cite-Wickham_2016"></a>(<a href="https://ggplot2.tidyverse.org">Wickham, 2016</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=golem">golem</a></em> <a id="cite-Guyader_2020"></a>(<a href="https://CRAN.R-project.org/package=golem">Guyader, Fay, Rochette, and Girard, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=plotly">plotly</a></em> <a id="cite-Sievert_2020"></a>(<a href="https://plotly-r.com">Sievert, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=Polychrome">Polychrome</a></em> <a id="cite-Coombes_2019"></a>(<a href="https://doi.org/10.18637/jss.v090.c01">Coombes, Brock, Abrams, and Abruzzo, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RColorBrewer">RColorBrewer</a></em> <a id="cite-Neuwirth_2014"></a>(<a href="https://CRAN.R-project.org/package=RColorBrewer">Neuwirth, 2014</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> <a id="cite-Allaire_2020"></a>(<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/S4Vectors">S4Vectors</a></em> <a id="cite-Pages_2020"></a>(<a href="#bib-Pages_2020">Pagès, Lawrence, and Aboyoun, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/scater">scater</a></em> <a id="cite-McCarthy_2017"></a>(<a href="https://doi.org/10.1093/bioinformatics/btw777">McCarthy, Campbell, Lun, and Willis, 2017</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/SingleCellExperiment">SingleCellExperiment</a></em> (<a href="#bib-Lun_2020">Lun and Risso, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> (<a href="https://CRAN.R-project.org/package=shiny">Chang, Cheng, Allaire, Xie, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/SummarizedExperiment">SummarizedExperiment</a></em> <a id="cite-Morgan_2020ab"></a>(<a href="#bib-Morgan_2020ab">Morgan, Obenchain, Hester, and Pagès, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=viridisLite">viridisLite</a></em> <a id="cite-Garnier_2018"></a>(<a href="https://CRAN.R-project.org/package=viridisLite">Garnier, 2018</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="co">## Create the vignette</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rmarkdown"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"spatialLIBD.Rmd"</span>))

<span class="co">## Extract the R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">"spatialLIBD.Rmd"</span>, <span class="kw">tangle</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2020-04-26 01:03:41 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 1.139 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                                             
#&gt;  version  R Under development (unstable) (2020-04-24 r78289)
#&gt;  os       Debian GNU/Linux 10 (buster)                      
#&gt;  system   x86_64, linux-gnu                                 
#&gt;  ui       X11                                               
#&gt;  language (EN)                                              
#&gt;  collate  en_US.UTF-8                                       
#&gt;  ctype    en_US.UTF-8                                       
#&gt;  tz       UTC                                               
#&gt;  date     2020-04-26                                        
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                * version  date       lib source        
#&gt;  AnnotationDbi            1.49.1   2020-01-25 [1] Bioconductor  
#&gt;  AnnotationHub            2.19.12  2020-04-21 [1] Bioconductor  
#&gt;  assertthat               0.2.1    2019-03-21 [2] CRAN (R 4.1.0)
#&gt;  attempt                  0.3.0    2019-04-08 [1] CRAN (R 4.1.0)
#&gt;  backports                1.1.6    2020-04-05 [2] CRAN (R 4.1.0)
#&gt;  beeswarm                 0.2.3    2016-04-25 [1] CRAN (R 4.1.0)
#&gt;  benchmarkme              1.0.3    2019-11-09 [1] CRAN (R 4.1.0)
#&gt;  benchmarkmeData          1.0.4    2020-04-23 [1] CRAN (R 4.1.0)
#&gt;  bibtex                   0.4.2.2  2020-01-02 [1] CRAN (R 4.1.0)
#&gt;  Biobase                * 2.47.3   2020-03-16 [1] Bioconductor  
#&gt;  BiocFileCache            1.11.6   2020-04-16 [1] Bioconductor  
#&gt;  BiocGenerics           * 0.33.3   2020-03-23 [1] Bioconductor  
#&gt;  BiocManager              1.30.10  2019-11-16 [2] CRAN (R 4.1.0)
#&gt;  BiocNeighbors            1.5.6    2020-04-21 [1] Bioconductor  
#&gt;  BiocParallel             1.21.3   2020-04-21 [1] Bioconductor  
#&gt;  BiocSingular             1.3.2    2020-02-14 [1] Bioconductor  
#&gt;  BiocStyle              * 2.15.8   2020-04-19 [1] Bioconductor  
#&gt;  BiocVersion              3.11.1   2019-11-13 [2] Bioconductor  
#&gt;  bit                      1.1-15.2 2020-02-10 [1] CRAN (R 4.1.0)
#&gt;  bit64                    0.9-7    2017-05-08 [1] CRAN (R 4.1.0)
#&gt;  bitops                   1.0-6    2013-08-17 [1] CRAN (R 4.1.0)
#&gt;  blob                     1.2.1    2020-01-20 [1] CRAN (R 4.1.0)
#&gt;  bookdown                 0.18     2020-03-05 [1] CRAN (R 4.1.0)
#&gt;  cli                      2.0.2    2020-02-28 [2] CRAN (R 4.1.0)
#&gt;  codetools                0.2-16   2018-12-24 [3] CRAN (R 4.1.0)
#&gt;  colorspace               1.4-1    2019-03-18 [1] CRAN (R 4.1.0)
#&gt;  config                   0.3      2018-03-27 [1] CRAN (R 4.1.0)
#&gt;  cowplot                  1.0.0    2019-07-11 [1] CRAN (R 4.1.0)
#&gt;  crayon                   1.3.4    2017-09-16 [2] CRAN (R 4.1.0)
#&gt;  curl                     4.3      2019-12-02 [2] CRAN (R 4.1.0)
#&gt;  data.table               1.12.8   2019-12-09 [1] CRAN (R 4.1.0)
#&gt;  DBI                      1.1.0    2019-12-15 [1] CRAN (R 4.1.0)
#&gt;  dbplyr                   1.4.3    2020-04-19 [1] CRAN (R 4.1.0)
#&gt;  DelayedArray           * 0.13.12  2020-04-10 [1] Bioconductor  
#&gt;  DelayedMatrixStats       1.9.1    2020-03-30 [1] Bioconductor  
#&gt;  desc                     1.2.0    2018-05-01 [2] CRAN (R 4.1.0)
#&gt;  digest                   0.6.25   2020-02-23 [2] CRAN (R 4.1.0)
#&gt;  dockerfiler              0.1.3    2019-03-19 [1] CRAN (R 4.1.0)
#&gt;  doParallel               1.0.15   2019-08-02 [1] CRAN (R 4.1.0)
#&gt;  dotCall64                1.0-0    2018-07-30 [1] CRAN (R 4.1.0)
#&gt;  dplyr                    0.8.5    2020-03-07 [1] CRAN (R 4.1.0)
#&gt;  DT                       0.13     2020-03-23 [2] CRAN (R 4.1.0)
#&gt;  ellipsis                 0.3.0    2019-09-20 [2] CRAN (R 4.1.0)
#&gt;  evaluate                 0.14     2019-05-28 [2] CRAN (R 4.1.0)
#&gt;  ExperimentHub            1.13.8   2020-04-21 [1] Bioconductor  
#&gt;  fansi                    0.4.1    2020-01-08 [2] CRAN (R 4.1.0)
#&gt;  farver                   2.0.3    2020-01-16 [1] CRAN (R 4.1.0)
#&gt;  fastmap                  1.0.1    2019-10-08 [1] CRAN (R 4.1.0)
#&gt;  fields                   10.3     2020-02-04 [1] CRAN (R 4.1.0)
#&gt;  foreach                  1.5.0    2020-03-30 [1] CRAN (R 4.1.0)
#&gt;  fs                       1.4.1    2020-04-04 [2] CRAN (R 4.1.0)
#&gt;  generics                 0.0.2    2018-11-29 [1] CRAN (R 4.1.0)
#&gt;  GenomeInfoDb           * 1.23.17  2020-04-13 [1] Bioconductor  
#&gt;  GenomeInfoDbData         1.2.3    2020-04-26 [1] Bioconductor  
#&gt;  GenomicRanges          * 1.39.3   2020-03-24 [1] Bioconductor  
#&gt;  ggbeeswarm               0.6.0    2017-08-07 [1] CRAN (R 4.1.0)
#&gt;  ggplot2                  3.3.0    2020-03-05 [1] CRAN (R 4.1.0)
#&gt;  glue                     1.4.0    2020-04-03 [2] CRAN (R 4.1.0)
#&gt;  golem                    0.2.1    2020-03-05 [1] CRAN (R 4.1.0)
#&gt;  gridExtra                2.3      2017-09-09 [1] CRAN (R 4.1.0)
#&gt;  gtable                   0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
#&gt;  htmltools                0.4.0    2019-10-04 [2] CRAN (R 4.1.0)
#&gt;  htmlwidgets              1.5.1    2019-10-08 [2] CRAN (R 4.1.0)
#&gt;  httpuv                   1.5.2    2019-09-11 [1] CRAN (R 4.1.0)
#&gt;  httr                     1.4.1    2019-08-05 [2] CRAN (R 4.1.0)
#&gt;  interactiveDisplayBase   1.25.0   2019-10-29 [1] Bioconductor  
#&gt;  IRanges                * 2.21.8   2020-03-25 [1] Bioconductor  
#&gt;  irlba                    2.3.3    2019-02-05 [1] CRAN (R 4.1.0)
#&gt;  iterators                1.0.12   2019-07-26 [1] CRAN (R 4.1.0)
#&gt;  jsonlite                 1.6.1    2020-02-02 [2] CRAN (R 4.1.0)
#&gt;  knitcitations          * 1.0.10   2019-09-15 [1] CRAN (R 4.1.0)
#&gt;  knitr                    1.28     2020-02-06 [2] CRAN (R 4.1.0)
#&gt;  labeling                 0.3      2014-08-23 [1] CRAN (R 4.1.0)
#&gt;  later                    1.0.0    2019-10-04 [2] CRAN (R 4.1.0)
#&gt;  lattice                  0.20-41  2020-04-02 [3] CRAN (R 4.1.0)
#&gt;  lazyeval                 0.2.2    2019-03-15 [2] CRAN (R 4.1.0)
#&gt;  lifecycle                0.2.0    2020-03-06 [2] CRAN (R 4.1.0)
#&gt;  lubridate                1.7.8    2020-04-06 [1] CRAN (R 4.1.0)
#&gt;  magrittr                 1.5      2014-11-22 [2] CRAN (R 4.1.0)
#&gt;  maps                     3.3.0    2018-04-03 [1] CRAN (R 4.1.0)
#&gt;  MASS                     7.3-51.5 2019-12-20 [3] CRAN (R 4.1.0)
#&gt;  Matrix                   1.2-18   2019-11-27 [3] CRAN (R 4.1.0)
#&gt;  matrixStats            * 0.56.0   2020-03-13 [1] CRAN (R 4.1.0)
#&gt;  memoise                  1.1.0    2017-04-21 [2] CRAN (R 4.1.0)
#&gt;  mime                     0.9      2020-02-04 [2] CRAN (R 4.1.0)
#&gt;  munsell                  0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
#&gt;  pillar                   1.4.3    2019-12-20 [2] CRAN (R 4.1.0)
#&gt;  pkgconfig                2.0.3    2019-09-22 [2] CRAN (R 4.1.0)
#&gt;  pkgdown                  1.5.1    2020-04-09 [1] CRAN (R 4.1.0)
#&gt;  pkgload                  1.0.2    2018-10-29 [2] CRAN (R 4.1.0)
#&gt;  plotly                   4.9.2.1  2020-04-04 [1] CRAN (R 4.1.0)
#&gt;  plyr                     1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
#&gt;  png                      0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
#&gt;  Polychrome               1.2.5    2020-03-29 [1] CRAN (R 4.1.0)
#&gt;  promises                 1.1.0    2019-10-04 [2] CRAN (R 4.1.0)
#&gt;  purrr                    0.3.4    2020-04-17 [2] CRAN (R 4.1.0)
#&gt;  R6                       2.4.1    2019-11-12 [2] CRAN (R 4.1.0)
#&gt;  rappdirs                 0.3.1    2016-03-28 [1] CRAN (R 4.1.0)
#&gt;  RColorBrewer             1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
#&gt;  Rcpp                     1.0.4.6  2020-04-09 [2] CRAN (R 4.1.0)
#&gt;  RCurl                    1.98-1.2 2020-04-18 [1] CRAN (R 4.1.0)
#&gt;  RefManageR               1.2.12   2019-04-03 [1] CRAN (R 4.1.0)
#&gt;  remotes                  2.1.1    2020-02-15 [1] CRAN (R 4.1.0)
#&gt;  rlang                    0.4.5    2020-03-01 [2] CRAN (R 4.1.0)
#&gt;  rmarkdown                2.1      2020-01-20 [1] CRAN (R 4.1.0)
#&gt;  roxygen2                 7.1.0    2020-03-11 [2] CRAN (R 4.1.0)
#&gt;  rprojroot                1.3-2    2018-01-03 [2] CRAN (R 4.1.0)
#&gt;  RSQLite                  2.2.0    2020-01-07 [1] CRAN (R 4.1.0)
#&gt;  rstudioapi               0.11     2020-02-07 [2] CRAN (R 4.1.0)
#&gt;  rsvd                     1.0.3    2020-02-17 [1] CRAN (R 4.1.0)
#&gt;  S4Vectors              * 0.25.15  2020-04-04 [1] Bioconductor  
#&gt;  scales                   1.1.0    2019-11-18 [1] CRAN (R 4.1.0)
#&gt;  scater                   1.15.32  2020-04-22 [1] Bioconductor  
#&gt;  scatterplot3d            0.3-41   2018-03-14 [1] CRAN (R 4.1.0)
#&gt;  sessioninfo            * 1.1.1    2018-11-05 [2] CRAN (R 4.1.0)
#&gt;  shiny                    1.4.0.2  2020-03-13 [1] CRAN (R 4.1.0)
#&gt;  shinyWidgets             0.5.1    2020-03-04 [1] CRAN (R 4.1.0)
#&gt;  SingleCellExperiment   * 1.9.3    2020-03-31 [1] Bioconductor  
#&gt;  spam                     2.5-1    2019-12-12 [1] CRAN (R 4.1.0)
#&gt;  spatialLIBD            * 0.99.15  2020-04-26 [1] Bioconductor  
#&gt;  stringi                  1.4.6    2020-02-17 [2] CRAN (R 4.1.0)
#&gt;  stringr                  1.4.0    2019-02-10 [2] CRAN (R 4.1.0)
#&gt;  SummarizedExperiment   * 1.17.5   2020-03-27 [1] Bioconductor  
#&gt;  testthat                 2.3.2    2020-03-02 [2] CRAN (R 4.1.0)
#&gt;  tibble                   3.0.1    2020-04-20 [2] CRAN (R 4.1.0)
#&gt;  tidyr                    1.0.2    2020-01-24 [1] CRAN (R 4.1.0)
#&gt;  tidyselect               1.0.0    2020-01-27 [1] CRAN (R 4.1.0)
#&gt;  usethis                  1.6.0    2020-04-09 [2] CRAN (R 4.1.0)
#&gt;  utf8                     1.1.4    2018-05-24 [2] CRAN (R 4.1.0)
#&gt;  vctrs                    0.2.4    2020-03-10 [2] CRAN (R 4.1.0)
#&gt;  vipor                    0.4.5    2017-03-22 [1] CRAN (R 4.1.0)
#&gt;  viridis                  0.5.1    2018-03-29 [1] CRAN (R 4.1.0)
#&gt;  viridisLite              0.3.0    2018-02-01 [1] CRAN (R 4.1.0)
#&gt;  withr                    2.2.0    2020-04-20 [2] CRAN (R 4.1.0)
#&gt;  xfun                     0.13     2020-04-13 [2] CRAN (R 4.1.0)
#&gt;  xml2                     1.3.2    2020-04-23 [2] CRAN (R 4.1.0)
#&gt;  xtable                   1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
#&gt;  XVector                  0.27.2   2020-03-24 [1] Bioconductor  
#&gt;  yaml                     2.2.1    2020-02-01 [2] CRAN (R 4.1.0)
#&gt;  zlibbioc                 1.33.1   2020-01-24 [1] Bioconductor  
#&gt; 
#&gt; [1] /usr/local/lib/R/host-site-library
#&gt; [2] /usr/local/lib/R/site-library
#&gt; [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.11/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>), <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations were made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.1. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[3] W. Chang, J. Cheng, J. Allaire, Y. Xie, et al. <em>shiny: Web Application Framework for R</em>. R package version 1.4.0.2. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=shiny" class="uri">https://CRAN.R-project.org/package=shiny</a>&gt;.</p>
<p>[4] L. Collado-Torres, K. R. Maynard, and A. E. Jaffe. <em>LIBD Visium spatial transcriptomics human pilot data inspector</em>. <a href="https://github.com/LieberInstitute/spatialLIBD" class="uri">https://github.com/LieberInstitute/spatialLIBD</a> - R package version 0.99.15. 2020. DOI: 10.18129/B9.bioc.spatialLIBD. &lt;URL: <a href="http://www.bioconductor.org/packages/spatialLIBD" class="uri">http://www.bioconductor.org/packages/spatialLIBD</a>&gt;.</p>
<p>[5] K. R. Coombes, G. Brock, Z. B. Abrams, and L. V. Abruzzo. “Polychrome: Creating and Assessing Qualitative Palettes with Many Colors”. In: <em>Journal of Statistical Software, Code Snippets</em> 90.1 (2019), pp. 1-23. DOI: 10.18637/jss.v090.c01.</p>
<p>[6] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[7] Douglas Nychka, Reinhard Furrer, John Paige, and Stephan Sain. <em>fields: Tools for spatial data</em>. R package version 10.3. Boulder, CO, USA: University Corporation for Atmospheric Research, 2017. DOI: 10.5065/D6W957CT. &lt;URL: <a href="https://github.com/NCAR/Fields" class="uri">https://github.com/NCAR/Fields</a>&gt;.</p>
<p>[8] S. Garnier. <em>viridisLite: Default Color Maps from ‘matplotlib’ (Lite Version)</em>. R package version 0.3.0. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=viridisLite" class="uri">https://CRAN.R-project.org/package=viridisLite</a>&gt;.</p>
<p>[9] C. Gillespie. <em>benchmarkme: Crowd Sourced System Benchmarks</em>. R package version 1.0.3. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=benchmarkme" class="uri">https://CRAN.R-project.org/package=benchmarkme</a>&gt;.</p>
<p>[10] V. Guyader, C. Fay, S. Rochette, and C. Girard. <em>golem: A Framework for Robust Shiny Applications</em>. R package version 0.2.1. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=golem" class="uri">https://CRAN.R-project.org/package=golem</a>&gt;.</p>
<p>[11] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[12] A. Lun and D. Risso. <em>SingleCellExperiment: S4 Classes for Single Cell Data</em>. R package version 1.9.3. 2020.</p>
<p>[13] K. R. Maynard, L. Collado-Torres, L. M. Weber, C. Uytingco, et al. “Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex”. In: <em>bioRxiv</em> (2020). DOI: 10.1101/2020.02.28.969931. &lt;URL: <a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1" class="uri">https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1</a>&gt;.</p>
<p>[14] D. J. McCarthy, K. R. Campbell, A. T. L. Lun, and Q. F. Willis. “Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R”. In: <em>Bioinformatics</em> 33 (8 2017), pp. 1179-1186. DOI: 10.1093/bioinformatics/btw777.</p>
<p>[15] M. Morgan, V. Obenchain, J. Hester, and H. Pagès. <em>SummarizedExperiment: SummarizedExperiment container</em>. R package version 1.17.5. 2020.</p>
<p>[16] M. Morgan and L. Shepherd. <em>AnnotationHub: Client to access AnnotationHub resources</em>. R package version 2.19.12. 2020.</p>
<p>[17] M. Morgan and L. Shepherd. <em>ExperimentHub: Client to access ExperimentHub resources</em>. R package version 1.13.8. 2020.</p>
<p>[18] E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package version 1.1-2. 2014. &lt;URL: <a href="https://CRAN.R-project.org/package=RColorBrewer" class="uri">https://CRAN.R-project.org/package=RColorBrewer</a>&gt;.</p>
<p>[19] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.15.8. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[20] H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: Foundation of vector-like and list-like containers in Bioconductor</em>. R package version 0.25.15. 2020.</p>
<p>[21] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[22] C. Sievert. <em>Interactive Web-Based Data Visualization with R, plotly, and shiny</em>. Chapman and Hall/CRC, 2020. ISBN: 9781138331457. &lt;URL: <a href="https://plotly-r.com" class="uri">https://plotly-r.com</a>&gt;.</p>
<p>[23] H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. &lt;URL: <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>&gt;.</p>
<p>[24] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[25] C. Wilke. <em>cowplot: Streamlined Plot Theme and Plot Annotations for ‘ggplot2’</em>. R package version 1.0.0. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=cowplot" class="uri">https://CRAN.R-project.org/package=cowplot</a>&gt;.</p>
<p>[26] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[27] Y. Xie, J. Cheng, and X. Tan. <em>DT: A Wrapper of the JavaScript Library ‘DataTables’</em>. R package version 0.13. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=DT" class="uri">https://CRAN.R-project.org/package=DT</a>&gt;.</p>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis">this code</a> for details on how we built the <code>sce</code> object. In particular check <code>convert_sce.R</code> and <code>sce_scran.R</code>.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis/Layer_Guesses">this code</a> for details on how we built the <code>sce_layer</code> object. In particular check <code>spots_per_layer.R</code> and <code>layer_enrichment.R</code>.<a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis/Layer_Guesses">this code</a> for details on how we built the <code>modeling_results</code> object. In particular check <code>layer_specificity_fstats.R</code>, <code>layer_specificity.R</code>, and <code>misc_numbers.R</code>.<a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>You can change the <code>destdir</code> argument and specific a specific location that you will use and re-use. However the default value of <code>destdir</code> is a temporary directory that will be wiped out once you close your R session.<a href="#fnref4" class="footnote-back">↩</a></p></li>
<li id="fn5"><p>For more details, check <a href="https://github.com/LieberInstitute/HumanPilot/blob/ab52559957624538f8dc08b0533e4106fab4066b/Analysis/Layer_Guesses/layer_specificity.R">this script</a>.<a href="#fnref5" class="footnote-back">↩</a></p></li>
<li id="fn6"><p>For more details, check <a href="https://github.com/LieberInstitute/HumanPilot/blob/ab52559957624538f8dc08b0533e4106fab4066b/Analysis/Layer_Guesses/dlpfc_snRNAseq_annotation.R">this script</a>.<a href="#fnref6" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
