<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to spatialLIBD • spatialLIBD</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to spatialLIBD">
<meta property="og:description" content="spatialLIBD">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to spatialLIBD</h1>
                        <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusCenter for Computational Biology, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Kristen
R. Maynard</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:Kristen.Maynard@libd.org" class="email">Kristen.Maynard@libd.org</a>
      </address>
                              <h4 data-toc-skip class="author">Andrew E.
Jaffe</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:andrew.jaffe@libd.org" class="email">andrew.jaffe@libd.org</a>
      </address>
                  
            <h4 data-toc-skip class="date">15 September 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/HEAD/vignettes/spatialLIBD.Rmd" class="external-link"><code>vignettes/spatialLIBD.Rmd</code></a></small>
      <div class="hidden name"><code>spatialLIBD.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="welcome">Welcome<a class="anchor" aria-label="anchor" href="#welcome"></a>
</h2>
<p>Welcome to the <code>spatialLIBD</code> project! It is composed
of:</p>
<ul>
<li>a <a href="https://shiny.rstudio.com/" class="external-link">shiny</a> web application
that we are hosting at <a href="http://spatial.libd.org/spatialLIBD/" class="external-link">spatial.libd.org/spatialLIBD/</a>
that can handle a <a href="https://github.com/LieberInstitute/spatialLIBD/issues/2" class="external-link">limited</a>
set of concurrent users,</li>
<li>a Bioconductor package at <a href="http://bioconductor.org/packages/spatialLIBD" class="external-link">bioconductor.org/packages/spatialLIBD</a>
(or from <a href="http://research.libd.org/spatialLIBD/" class="external-link">here</a>) that
lets you analyze the data and run a local version of our web application
(with our data or yours),</li>
<li>and a <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">research
article</a> with the scientific knowledge we drew from this dataset. The
analysis code for our project is available <a href="https://github.com/LieberInstitute/HumanPilot/" class="external-link">here</a> and the
high quality figures for the manuscript are available through <a href="https://doi.org/10.6084/m9.figshare.13623902.v1" class="external-link">Figshare</a>.</li>
</ul>
<p>The web application allows you to browse the LIBD human dorsolateral
pre-frontal cortex (DLPFC) spatial transcriptomics data generated with
the 10x Genomics Visium platform. Through the <a href="https://bioconductor.org/packages/spatialLIBD" class="external-link">R/Bioconductor
package</a> you can also download the data as well as visualize your own
datasets using this web application. Please check the <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">manuscript</a> or <a href="https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1" class="external-link">bioRxiv
pre-print</a> for more details about this project.</p>
<p>If you tweet about this website, the data or the R package please use
the <code>#spatialLIBD</code> hashtag. You can find previous tweets that
way as shown
<a href="https://twitter.com/search?q=%23spatialLIBD&amp;src=typed_query" class="external-link">here</a>.
Thank you!</p>
<div class="section level3">
<h3 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h3>
<p>As a quick overview, the data presented here is from portion of the
DLPFC that spans six neuronal layers plus white matter
(<strong>A</strong>) for a total of three subjects with two pairs of
spatially adjacent replicates (<strong>B</strong>). Each dissection of
DLPFC was designed to span all six layers plus white matter
(<strong>C</strong>). Using this web application you can explore the
expression of known genes such as <em>SNAP25</em> (<strong>D</strong>, a
neuronal gene), <em>MOBP</em> (<strong>E</strong>, an oligodendrocyte
gene), and known layer markers from mouse studies such as <em>PCP4</em>
(<strong>F</strong>, a known layer 5 marker gene).</p>
<p><img src="http://research.libd.org/spatialLIBD/reference/figures/paper_figure1.jpg" align="center" width="800px"></p>
</div>
<div class="section level3">
<h3 id="shiny-website-mirrors">Shiny website mirrors<a class="anchor" aria-label="anchor" href="#shiny-website-mirrors"></a>
</h3>
<ul>
<li><a href="http://spatial.libd.org/spatialLIBD" class="external-link">Main shiny application
website</a></li>
<li><a href="https://libd.shinyapps.io/spatialLIBD/" class="external-link">Shinyapps</a></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-spatiallibd">Install <code>spatialLIBD</code><a class="anchor" aria-label="anchor" href="#install-spatiallibd"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
is a <code>R</code> package available via Bioconductor. <code>R</code>
can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To run all the code in this vignette, you might need to install other
R/Bioconductor packages, which you can do with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>If you want to use the development version of
<code>spatialLIBD</code>, you will need to use the R version
corresponding to the current Bioconductor-devel branch as described in
more detail on the <a href="http://bioconductor.org/developers/how-to/useDevel/" class="external-link">Bioconductor
website</a>. Then you can install <code>spatialLIBD</code> from GitHub
using the following command.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/spatialLIBD"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, Hicks, Jaffe, Martinowich, Maynard, and
Collado-Torres, 2022) is based on many other packages and in particular
in those that have implemented the infrastructure needed for dealing
with single cell RNA sequencing data, visualization functions, and
interactive data exploration. That is, packages like <em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>
that allow you to store the data, <em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em> and
<em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em>
for visualizing the data, and <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> for
building an interactive interface. A <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
user who only accesses the web application is not expected to deal with
those packages directly. A <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
user will need to be familiar with <em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>
and <em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em> to
understand the data provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
or the graphical results <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
provides. Furthermore, it’ll be useful for the user to know about <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> and
<em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em>
if you wish to adapt the web application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help regarding Bioconductor. Other
alternatives are available such as creating GitHub issues and tweeting.
However, please note that if you want to receive help you should adhere
to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-spatiallibd">Citing <code>spatialLIBD</code><a class="anchor" aria-label="anchor" href="#citing-spatiallibd"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
will be useful for your research. Please use the following information
to cite the package and the research article describing the data
provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>.
Thank you!</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite package 'spatialLIBD' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Pardo B, Spangler A, Weber LM, Hicks SC, Jaffe AE, Martinowich K,</span></span>
<span><span class="co">#&gt;   Maynard KR, Collado-Torres L (2022). "spatialLIBD: an R/Bioconductor</span></span>
<span><span class="co">#&gt;   package to visualize spatially-resolved transcriptomics data." _BMC</span></span>
<span><span class="co">#&gt;   Genomics_. doi:10.1186/s12864-022-08601-w</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1186/s12864-022-08601-w&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1186/s12864-022-08601-w&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Maynard KR, Collado-Torres L, Weber LM, Uytingco C, Barry BK,</span></span>
<span><span class="co">#&gt;   Williams SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y,</span></span>
<span><span class="co">#&gt;   Kleinman JE, Hyde TM, Rao N, Hicks SC, Martinowich K, Jaffe AE</span></span>
<span><span class="co">#&gt;   (2021). "Transcriptome-scale spatial gene expression in the human</span></span>
<span><span class="co">#&gt;   dorsolateral prefrontal cortex." _Nature Neuroscience_.</span></span>
<span><span class="co">#&gt;   doi:10.1038/s41593-020-00787-0</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1038/s41593-020-00787-0&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.nature.com/articles/s41593-020-00787-0&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) package was developed for
analyzing the human dorsolateral prefrontal cortex (DLPFC) spatial
transcriptomics data generated with the 10x Genomics Visium technology
by researchers at the Lieber Institute for Brain Development (LIBD)
(Maynard, Collado-Torres, Weber, Uytingco, Barry, Williams, II, Tran,
Besich, Tippani, Chew, Yin, Kleinman, Hyde, Rao, Hicks, Martinowich, and
Jaffe, 2021). An initial <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application was developed for interactively exploring this data and for
assigning human brain layer labels to the each spot for each sample
generated. While this was useful enough for our project, we made this <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> package in case you
want to:</p>
<ul>
<li>access our Visium data to get some data from this new technology and
develop methods or infrastructure for other Visium datasets.</li>
<li>re-shape your data into what ours is structured as, then re-use the
visualization functions and/or the shiny app itself.</li>
<li>want to explore our data in more detail. This can range from
launching the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application locally to diving into the specifics of the data from our
project (Maynard, Collado-Torres, Weber, et al., 2021).</li>
</ul>
<p>In this vignette we’ll showcase how you can access the Human DLPFC
LIBD Visium dataset (Maynard, Collado-Torres, Weber, et al., 2021), the
R functions provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022), and an overview of how you can
re-shape your own Visium dataset to match the structure we used.</p>
<p>To get started, please load the <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">"spatialLIBD"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="human-dlpfc-visium-dataset">Human DLPFC Visium dataset<a class="anchor" aria-label="anchor" href="#human-dlpfc-visium-dataset"></a>
</h2>
<p>The human DLPFC 10x Genomics Visium dataset analyzed by LIBD
researchers and colleagues is described in detail by Maynard,
Collado-Torres et al (Maynard, Collado-Torres, Weber, et al., 2021).
However, briefly, this dataset is composed of:</p>
<ul>
<li>Three brain subjects (all controls; two males, one female; ages
30-46; <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/visium_dlpfc_pilot_sample_metrics.tsv" class="external-link">details</a>).</li>
<li>Four images per subject: two spatially adjacent replicates at
position 0, then two more spatially adjacent replicates 300 micrometers
away.</li>
<li>Slices designed to cover layers 1 through 6 and the white matter
(WM) of the dorsolateral prefrontal cortex (DLPFC).</li>
</ul>
<div class="section level3">
<h3 id="data-specifics">Data specifics<a class="anchor" aria-label="anchor" href="#data-specifics"></a>
</h3>
<p>We combined all the Visium data into a single <em><a href="https://bioconductor.org/packages/3.15/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
(Righelli, Weber, Crowell, Pardo, Collado-Torres, Ghazanfar, Lun, Hicks,
and Risso, 2022) object that we typically refer to as
<strong><code>spe</code></strong> <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. It has 33,538 genes (rows) and 47,681
spots (columns). This is the initial point for most of our analyses (<a href="https://github.com/LieberInstitute/HumanPilot" class="external-link">code available on
GitHub</a>). Using <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) we manually assigned each spot
across all 12 images to a layer (L1 through L6 or WM). We then
compressed the spot-level data at the layer-level using a pseudo-bulking
approach resulting in the <em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>
object we typically refer to as <strong><code>sce_layer</code></strong>
<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. We
then computed for each gene t or F statistics assessing whether the gene
had higher expression in a given layer compared to the rest
(<code>enrichment</code>; t-stat), between one layer and another layer
(<code>pairwise</code>; t-stat), or had any expression variability
across all layers (<code>anova</code>; F-stat). The results from the
models are stored in what we refer to as
<strong><code>modeling_results</code></strong> <a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>In summary,</p>
<ul>
<li>
<code>spe</code> is the <em><a href="https://bioconductor.org/packages/3.15/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
object with all the spot-level data and the histology information for
visualization of the data.</li>
<li>
<code>sce_layer</code> is the <em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>
object with the layer-level data.</li>
<li>
<code>modeling_results</code> contains the layer-level
<code>enrichment</code>, <code>pairwise</code> and <code>anova</code>
statistics.</li>
</ul>
</div>
<div class="section level3">
<h3 id="downloading-the-data-with-spatiallibd">Downloading the data with <code>spatialLIBD</code><a class="anchor" aria-label="anchor" href="#downloading-the-data-with-spatiallibd"></a>
</h3>
<p>Using <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) you can download all of these R
objects. They are hosted by <a href="http://bioconductor.org/" class="external-link">Bioconductor</a>’s <em><a href="https://bioconductor.org/packages/3.15/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2022) resource and you can download them using
<code><a href="../reference/fetch_data.html">spatialLIBD::fetch_data()</a></code>. <code><a href="../reference/fetch_data.html">fetch_data()</a></code> will
query <em><a href="https://bioconductor.org/packages/3.15/ExperimentHub" class="external-link">ExperimentHub</a></em>
which in turn will download the data and cache it so you don’t have to
download it again. If <em><a href="https://bioconductor.org/packages/3.15/ExperimentHub" class="external-link">ExperimentHub</a></em>
is unavailable, then <code><a href="../reference/fetch_data.html">fetch_data()</a></code> has a backup option that
does not cache the files <a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a> Below we obtain all of these objects.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Connect to ExperimentHub</span></span>
<span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2022-04-26</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the small example sce data</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"sce_example"</span>, eh <span class="op">=</span> <span class="va">ehub</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2022-09-15 16:45:37 loading file /Users/leocollado/Library/Caches/org.R-project.R/R/BiocFileCache/4aa21b4d4942_sce_sub_for_vignette.Rdata%3Fdl%3D1</span></span>
<span></span>
<span><span class="co">## Convert to a SpatialExperiment object</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sce_to_spe.html">sce_to_spe</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## If you want to download the full real data (about 2.1 GB in RAM) use:</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"spe"</span><span class="op">)</span><span class="op">)</span> <span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"spe"</span>, eh <span class="op">=</span> <span class="va">ehub</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## Query ExperimentHub and download the data</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"sce_layer"</span><span class="op">)</span><span class="op">)</span> <span class="va">sce_layer</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"sce_layer"</span>, eh <span class="op">=</span> <span class="va">ehub</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2022-09-15 16:45:41 loading file /Users/leocollado/Library/Caches/org.R-project.R/R/BiocFileCache/1752a2a7d59e5_Human_DLPFC_Visium_processedData_sce_scran_sce_layer_spatialLIBD.Rdata%3Fdl%3D1</span></span>
<span><span class="va">modeling_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_data.html">fetch_data</a></span><span class="op">(</span><span class="st">"modeling_results"</span>, eh <span class="op">=</span> <span class="va">ehub</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2022-09-15 16:45:42 loading file /Users/leocollado/Library/Caches/org.R-project.R/R/BiocFileCache/425b2b3c4015_Human_DLPFC_Visium_modeling_results.Rdata%3Fdl%3D1</span></span></code></pre></div>
<p>Once you have downloaded the objects, we can explore them a little
bit</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## spot-level data</span></span>
<span><span class="va">spe</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 33538 47681 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(9): source type ... gene_search is_top_hvg</span></span>
<span><span class="co">#&gt; colnames(47681): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span></span>
<span><span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">#&gt; colData names(69): sample_id Cluster ... array_row array_col</span></span>
<span><span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity50 ... TSNE_perplexity80</span></span>
<span><span class="co">#&gt;   UMAP_neighbors15</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span>
<span></span>
<span><span class="co">## layer-level data</span></span>
<span><span class="va">sce_layer</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 22331 76 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(22331): ENSG00000243485 ENSG00000238009 ... ENSG00000278384</span></span>
<span><span class="co">#&gt;   ENSG00000271254</span></span>
<span><span class="co">#&gt; rowData names(10): source type ... is_top_hvg is_top_hvg_sce_layer</span></span>
<span><span class="co">#&gt; colnames(76): 151507_Layer1 151507_Layer2 ... 151676_Layer6 151676_WM</span></span>
<span><span class="co">#&gt; colData names(13): sample_name layer_guess ...</span></span>
<span><span class="co">#&gt;   layer_guess_reordered_short spatialLIBD</span></span>
<span><span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity5 ... UMAP_neighbors15 PCAsub</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span></span>
<span><span class="co">## list of modeling result tables</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">modeling_results</span>, <span class="va">class</span><span class="op">)</span></span>
<span><span class="co">#&gt;        anova   enrichment     pairwise </span></span>
<span><span class="co">#&gt; "data.frame" "data.frame" "data.frame"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">modeling_results</span>, <span class="va">dim</span><span class="op">)</span></span>
<span><span class="co">#&gt;      anova enrichment pairwise</span></span>
<span><span class="co">#&gt; [1,] 22331      22331    22331</span></span>
<span><span class="co">#&gt; [2,]    10         23       65</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">modeling_results</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co">#&gt;      anova          enrichment      pairwise          </span></span>
<span><span class="co">#&gt; [1,] "f_stat_full"  "t_stat_WM"     "t_stat_WM-Layer1"</span></span>
<span><span class="co">#&gt; [2,] "p_value_full" "t_stat_Layer1" "t_stat_WM-Layer2"</span></span>
<span><span class="co">#&gt; [3,] "fdr_full"     "t_stat_Layer2" "t_stat_WM-Layer3"</span></span>
<span><span class="co">#&gt; [4,] "full_AveExpr" "t_stat_Layer3" "t_stat_WM-Layer4"</span></span>
<span><span class="co">#&gt; [5,] "f_stat_noWM"  "t_stat_Layer4" "t_stat_WM-Layer5"</span></span>
<span><span class="co">#&gt; [6,] "p_value_noWM" "t_stat_Layer5" "t_stat_WM-Layer6"</span></span></code></pre></div>
<p>The modeling statistics are in wide format, which can make some
visualizations complicated. The function
<code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> provides a way to convert them into
long format and add some useful information. Let’s do so below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Convert to a long format the modeling results</span></span>
<span><span class="co">## This takes a few seconds to run</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span></span>
<span>    <span class="va">sig_genes</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu"><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all</a></span><span class="op">(</span></span>
<span>            n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce_layer</span><span class="op">)</span>,</span>
<span>            modeling_results <span class="op">=</span> <span class="va">modeling_results</span>,</span>
<span>            sce_layer <span class="op">=</span> <span class="va">sce_layer</span></span>
<span>        <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   4.870   0.218   5.090</span></span>
<span></span>
<span><span class="co">## Explore the result</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">sig_genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DFrame"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "S4Vectors"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sig_genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1138881      12</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="interactively-explore-the-spatiallibd-data">Interactively explore the <code>spatialLIBD</code> data<a class="anchor" aria-label="anchor" href="#interactively-explore-the-spatiallibd-data"></a>
</h2>
<p>Now that you have downloaded the data, you can interactively explore
the data using a <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> (Chang,
Cheng, Allaire, Sievert, Schloerke, Xie, Allen, McPherson, Dipert, and
Borges, 2022) web application contained within <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022). To do so, use the
<code><a href="../reference/run_app.html">run_app()</a></code> function as shown below:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span></span>
<span>        spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>        sce_layer <span class="op">=</span> <span class="va">sce_layer</span>,</span>
<span>        modeling_results <span class="op">=</span> <span class="va">modeling_results</span>,</span>
<span>        sig_genes <span class="op">=</span> <span class="va">sig_genes</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>The <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
<em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application allows you to browse the spot-level data and interactively
label spots, as well as explore the layer-level results. Once you launch
it, check the <em>Documentation</em> tab for each view for more details.
In order to avoid duplicating the documentation, we provide all the
details on the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application itself.</p>
<p>Though overall, this application allows you to export all static
visualizations as PDF files or all interactive visualizations as PNG
files, as well as all result table as CSV files. This is what produces
the version you can access without any R use from your side at <a href="http://spatial.libd.org/spatialLIBD/" class="external-link">spatial.libd.org/spatialLIBD/</a>.</p>
</div>
<div class="section level2">
<h2 id="spatiallibd-functions">
<code>spatialLIBD</code> functions<a class="anchor" aria-label="anchor" href="#spatiallibd-functions"></a>
</h2>
<p>We already covered <code><a href="../reference/fetch_data.html">fetch_data()</a></code> which allows you to
download the Human DLPFC Visium data from LIBD researchers and
colleagues (Maynard, Collado-Torres, Weber, et al., 2021).</p>
<div class="section level3">
<h3 id="spot-level-clusters-and-discrete-variables">Spot-level clusters and discrete variables<a class="anchor" aria-label="anchor" href="#spot-level-clusters-and-discrete-variables"></a>
</h3>
<p>With the <code>spe</code> object that contains the spot-level data,
we can visualize any discrete variable such as the layers using
<code><a href="../reference/vis_clus.html">vis_clus()</a></code> and related functions. These functions know
where to extract and how to visualize the histology information.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## View our LIBD layers for one sample</span></span>
<span><span class="fu"><a href="../reference/vis_clus.html">vis_clus</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    clustervar <span class="op">=</span> <span class="st">"layer_guess_reordered"</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"151673"</span>,</span>
<span>    colors <span class="op">=</span> <span class="va">libd_layer_colors</span>,</span>
<span>    ... <span class="op">=</span> <span class="st">" LIBD Layers"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/vis_clus-1.png" width="700"></p>
<p>Most of the variables stored in <code>spe</code> are discrete
variables and as such, you can visualize them using
<code><a href="../reference/vis_clus.html">vis_clus()</a></code> and <code><a href="../reference/vis_grid_clus.html">vis_grid_clus()</a></code> for one or more
than one sample respectively.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This is not fully precise, but gives you a rough idea</span></span>
<span><span class="co">## Some integer columns are actually continuous variables</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="va">class</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"factor"</span>, <span class="st">"integer"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;    12    57</span></span>
<span></span>
<span><span class="co">## This is more precise (one cluster has 28 unique values)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">29</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;     7    62</span></span></code></pre></div>
<p>Notably, <code><a href="../reference/vis_clus.html">vis_clus()</a></code> has a <code>spatial</code>
<code>logical(1)</code> argument which is useful if you want to
visualize the data without the histology information provided by
<code><a href="../reference/geom_spatial.html">geom_spatial()</a></code> (a custom <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">ggplot2::layer()</a></code>). In
particular, this is useful if you then want to use
<code><a href="https://rdrr.io/pkg/plotly/man/ggplotly.html" class="external-link">plotly::ggplotly()</a></code> or other similar functions on the
resulting <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2::ggplot()</a></code> object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## View our LIBD layers for one sample</span></span>
<span><span class="co">## without spatial information</span></span>
<span><span class="fu"><a href="../reference/vis_clus.html">vis_clus</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    clustervar <span class="op">=</span> <span class="st">"layer_guess_reordered"</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"151673"</span>,</span>
<span>    colors <span class="op">=</span> <span class="va">libd_layer_colors</span>,</span>
<span>    ... <span class="op">=</span> <span class="st">" LIBD Layers"</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/vis_clus_no_spatial-1.png" width="700"></p>
<p>Some helper functions include <code><a href="../reference/get_colors.html">get_colors()</a></code> and
<code><a href="../reference/sort_clusters.html">sort_clusters()</a></code> as well as the
<code>libd_layer_colors</code> object included in <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Color palette designed by Lukas M. Weber with feedback from the team.</span></span>
<span><span class="va">libd_layer_colors</span></span>
<span><span class="co">#&gt;        Layer1        Layer2        Layer3        Layer4        Layer5 </span></span>
<span><span class="co">#&gt;     "#F0027F"     "#377EB8"     "#4DAF4A"     "#984EA3"     "#FFD700" </span></span>
<span><span class="co">#&gt;        Layer6            WM            NA           WM2 </span></span>
<span><span class="co">#&gt;     "#FF7F00"     "#1A1A1A" "transparent"     "#666666"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="spot-level-genes-and-continuous-variables">Spot-level genes and continuous variables<a class="anchor" aria-label="anchor" href="#spot-level-genes-and-continuous-variables"></a>
</h3>
<p>Similar to <code><a href="../reference/vis_clus.html">vis_clus()</a></code>, the <code><a href="../reference/vis_gene.html">vis_gene()</a></code>
family of functions use the <code>spe</code> spot-level object to
visualize the gene expression or any continuous variable such as the
number of cells per spots. That is, <code><a href="../reference/vis_gene.html">vis_gene()</a></code> can
visualize any of the <code>assays(spe)</code> or any of the continuous
variables stored in <code>colData(spe)</code>. If you want to visualize
more than one sample at a time, use <code><a href="../reference/vis_grid_gene.html">vis_grid_gene()</a></code>
instead. And just like <code><a href="../reference/vis_clus.html">vis_clus()</a></code>, <code><a href="../reference/vis_gene.html">vis_gene()</a></code>
has a <code>spatial</code> <code>logical(1)</code> argument to turn off
the custom spatial <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">ggplot2::layer()</a></code> produced by
<code><a href="../reference/geom_spatial.html">geom_spatial()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Available gene expression assays</span></span>
<span><span class="fu">assayNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "counts"    "logcounts"</span></span>
<span></span>
<span><span class="co">## Not all of these make sense to visualize</span></span>
<span><span class="co">## In particular, the key is not useful to visualize.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "sample_id"                   "Cluster"                    </span></span>
<span><span class="co">#&gt;  [3] "sum_umi"                     "sum_gene"                   </span></span>
<span><span class="co">#&gt;  [5] "subject"                     "position"                   </span></span>
<span><span class="co">#&gt;  [7] "replicate"                   "subject_position"           </span></span>
<span><span class="co">#&gt;  [9] "discard"                     "key"                        </span></span>
<span><span class="co">#&gt; [11] "cell_count"                  "SNN_k50_k4"                 </span></span>
<span><span class="co">#&gt; [13] "SNN_k50_k5"                  "SNN_k50_k6"                 </span></span>
<span><span class="co">#&gt; [15] "SNN_k50_k7"                  "SNN_k50_k8"                 </span></span>
<span><span class="co">#&gt; [17] "SNN_k50_k9"                  "SNN_k50_k10"                </span></span>
<span><span class="co">#&gt; [19] "SNN_k50_k11"                 "SNN_k50_k12"                </span></span>
<span><span class="co">#&gt; [21] "SNN_k50_k13"                 "SNN_k50_k14"                </span></span>
<span><span class="co">#&gt; [23] "SNN_k50_k15"                 "SNN_k50_k16"                </span></span>
<span><span class="co">#&gt; [25] "SNN_k50_k17"                 "SNN_k50_k18"                </span></span>
<span><span class="co">#&gt; [27] "SNN_k50_k19"                 "SNN_k50_k20"                </span></span>
<span><span class="co">#&gt; [29] "SNN_k50_k21"                 "SNN_k50_k22"                </span></span>
<span><span class="co">#&gt; [31] "SNN_k50_k23"                 "SNN_k50_k24"                </span></span>
<span><span class="co">#&gt; [33] "SNN_k50_k25"                 "SNN_k50_k26"                </span></span>
<span><span class="co">#&gt; [35] "SNN_k50_k27"                 "SNN_k50_k28"                </span></span>
<span><span class="co">#&gt; [37] "GraphBased"                  "Maynard"                    </span></span>
<span><span class="co">#&gt; [39] "Martinowich"                 "layer_guess"                </span></span>
<span><span class="co">#&gt; [41] "layer_guess_reordered"       "layer_guess_reordered_short"</span></span>
<span><span class="co">#&gt; [43] "expr_chrM"                   "expr_chrM_ratio"            </span></span>
<span><span class="co">#&gt; [45] "SpatialDE_PCA"               "SpatialDE_pool_PCA"         </span></span>
<span><span class="co">#&gt; [47] "HVG_PCA"                     "pseudobulk_PCA"             </span></span>
<span><span class="co">#&gt; [49] "markers_PCA"                 "SpatialDE_UMAP"             </span></span>
<span><span class="co">#&gt; [51] "SpatialDE_pool_UMAP"         "HVG_UMAP"                   </span></span>
<span><span class="co">#&gt; [53] "pseudobulk_UMAP"             "markers_UMAP"               </span></span>
<span><span class="co">#&gt; [55] "SpatialDE_PCA_spatial"       "SpatialDE_pool_PCA_spatial" </span></span>
<span><span class="co">#&gt; [57] "HVG_PCA_spatial"             "pseudobulk_PCA_spatial"     </span></span>
<span><span class="co">#&gt; [59] "markers_PCA_spatial"         "SpatialDE_UMAP_spatial"     </span></span>
<span><span class="co">#&gt; [61] "SpatialDE_pool_UMAP_spatial" "HVG_UMAP_spatial"           </span></span>
<span><span class="co">#&gt; [63] "pseudobulk_UMAP_spatial"     "markers_UMAP_spatial"       </span></span>
<span><span class="co">#&gt; [65] "spatialLIBD"                 "ManualAnnotation"           </span></span>
<span><span class="co">#&gt; [67] "in_tissue"                   "array_row"                  </span></span>
<span><span class="co">#&gt; [69] "array_col"</span></span>
<span></span>
<span><span class="co">## Visualize a gene</span></span>
<span><span class="fu"><a href="../reference/vis_gene.html">vis_gene</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"151673"</span>,</span>
<span>    viridis <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/vis_gene-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Visualize the estimated number of cells per spot</span></span>
<span><span class="fu"><a href="../reference/vis_gene.html">vis_gene</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"151673"</span>,</span>
<span>    geneid <span class="op">=</span> <span class="st">"cell_count"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/vis_gene-2.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Visualize the fraction of chrM expression per spot</span></span>
<span><span class="co">## without the spatial layer</span></span>
<span><span class="fu"><a href="../reference/vis_gene.html">vis_gene</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"151673"</span>,</span>
<span>    geneid <span class="op">=</span> <span class="st">"expr_chrM_ratio"</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/vis_gene-3.png" width="700"></p>
<p>As for the color palette, you can either use the color blind friendly
palette when <code>viridis = TRUE</code> or a custom palette we
determined. Note that if you design your own palette, you have to take
into account that values it can be hard to distinguish some colors from
the histology set of purple tones that are noticeable when the
continuous variable is below or equal to the <code>minCount</code> (in
our palettes such points have a <code>'transparent'</code> color). For
more details, check the internal code of <code><a href="../reference/vis_gene_p.html">vis_gene_p()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="extract-significant-genes">Extract significant genes<a class="anchor" aria-label="anchor" href="#extract-significant-genes"></a>
</h3>
<p>Earlier we also ran <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> in order to
run the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> web
application. However, we didn’t explain the output. If you explore it,
you’ll notice that it’s a very long table with several columns.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sig_genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 12 columns</span></span>
<span><span class="co">#&gt;         top  model_type        test        gene      stat        pval</span></span>
<span><span class="co">#&gt;   &lt;integer&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1         1  enrichment          WM       NDRG1   16.3053 1.25896e-26</span></span>
<span><span class="co">#&gt; 2         2  enrichment          WM      PTP4A2   16.1469 2.25133e-26</span></span>
<span><span class="co">#&gt; 3         3  enrichment          WM        AQP1   15.9927 3.97849e-26</span></span>
<span><span class="co">#&gt; 4         4  enrichment          WM       PAQR6   15.1971 7.86258e-25</span></span>
<span><span class="co">#&gt; 5         5  enrichment          WM      ANP32B   14.9798 1.80183e-24</span></span>
<span><span class="co">#&gt; 6         6  enrichment          WM        JAM3   14.7413 4.50756e-24</span></span>
<span><span class="co">#&gt;           fdr gene_index         ensembl           in_rows       in_rows_top20</span></span>
<span><span class="co">#&gt;     &lt;numeric&gt;  &lt;integer&gt;     &lt;character&gt;     &lt;IntegerList&gt;       &lt;IntegerList&gt;</span></span>
<span><span class="co">#&gt; 1 2.51372e-22      10404 ENSG00000104419 1,40215,65023,... 1,156337,223320,...</span></span>
<span><span class="co">#&gt; 2 2.51372e-22        487 ENSG00000184007 2,40531,64532,... 2,223323,245661,...</span></span>
<span><span class="co">#&gt; 3 2.96145e-22       8201 ENSG00000240583 3,32241,65062,... 3,223314,245643,...</span></span>
<span><span class="co">#&gt; 4 4.38948e-21       1501 ENSG00000160781 4,34318,65863,...     4,245654,267982</span></span>
<span><span class="co">#&gt; 5 8.04735e-21      10962 ENSG00000136938 5,28964,63931,...     5,223329,267975</span></span>
<span><span class="co">#&gt; 6 1.67295e-20      12600 ENSG00000166086 6,40868,65917,... 6,156334,223324,...</span></span>
<span><span class="co">#&gt;                                       results</span></span>
<span><span class="co">#&gt;                               &lt;CharacterList&gt;</span></span>
<span><span class="co">#&gt; 1 WM_top1,WM-Layer1_top20,WM-Layer4_top10,...</span></span>
<span><span class="co">#&gt; 2 WM_top2,WM-Layer4_top13,WM-Layer5_top20,...</span></span>
<span><span class="co">#&gt; 3   WM_top3,WM-Layer4_top4,WM-Layer5_top2,...</span></span>
<span><span class="co">#&gt; 4     WM_top4,WM-Layer5_top13,WM-Layer6_top10</span></span>
<span><span class="co">#&gt; 5      WM_top5,WM-Layer4_top19,WM-Layer6_top3</span></span>
<span><span class="co">#&gt; 6 WM_top6,WM-Layer1_top17,WM-Layer4_top14,...</span></span></code></pre></div>
<p>The output of <code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> contains the
following columns:</p>
<ul>
<li>
<code>top</code>: the rank of the gene for the given
<code>test</code>.</li>
<li>
<code>model_type</code>: either <code>enrichment</code>,
<code>pairwise</code> or <code>anova</code>.</li>
<li>
<code>test</code>: the short notation for the test performed. For
example, <code>WM</code> is white matter versus the other layers while
<code>WM-Layer1</code> is white matter greater than layer 1.</li>
<li>
<code>gene</code>: the gene symbol.</li>
<li>
<code>stat</code>: the corresponding F or t-statistic.</li>
<li>
<code>pval</code>: the corresponding p-value (two-sided for
t-stats).</li>
<li>
<code>fdr</code>: the FDR adjusted p-value.</li>
<li>
<code>gene_index</code>: the row of <code>sce_layer</code> and the
original tables in <code>modeling_results</code> to join the tables if
necessary.</li>
<li>
<code>ensembl</code>: Ensembl gene ID.</li>
<li>
<code>in_rows</code>: an <code>IntegerList()</code> specifying all
the rows where that gene is present.</li>
<li>
<code>in_rows_top20</code>: an <code>IntegerList()</code> specifying
all the rows where that gene is present and where its rank
(<code>top</code>) is less than or equal to 20. This information is only
included for the first occurrence of each gene if that gene is on the
top 20 rank for any of the models.</li>
<li>
<code>results</code>: an <code>CharacterList()</code> specifying all
the <code>test</code> results where the gene is ranked
(<code>top</code>) among the top 20 genes.</li>
</ul>
<p><code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code> uses
<code><a href="../reference/sig_genes_extract.html">sig_genes_extract()</a></code> as it’s workhorse and as such has a
very similar output.</p>
</div>
<div class="section level3">
<h3 id="visualize-modeling-results">Visualize modeling results<a class="anchor" aria-label="anchor" href="#visualize-modeling-results"></a>
</h3>
<p>After extracting the table of modeling results in long format with
<code><a href="../reference/sig_genes_extract_all.html">sig_genes_extract_all()</a></code>, we can then use
<code><a href="../reference/layer_boxplot.html">layer_boxplot()</a></code> to visualize any gene of interest for any
of the model types and tests we performed. Below we explore the first
one (by default). We also show the color palettes used in the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022). The last example has the long
title version that uses more information from the <code>sig_genes</code>
object we created earlier.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Note that we recommend setting the random seed so the jittering of the</span></span>
<span><span class="co">## points will be reproducible. Given the requirements by BiocCheck this</span></span>
<span><span class="co">## cannot be done inside the layer_boxplot() function.</span></span>
<span></span>
<span><span class="co">## Create a boxplot of the first gene in `sig_genes`.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20200206</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span><span class="op">(</span>sig_genes <span class="op">=</span> <span class="va">sig_genes</span>, sce_layer <span class="op">=</span> <span class="va">sce_layer</span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Viridis colors displayed in the shiny app</span></span>
<span><span class="co">## showing the first pairwise model result</span></span>
<span><span class="co">## (which illustrates the background colors used for the layers not</span></span>
<span><span class="co">## involved in the pairwise comparison)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20200206</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span><span class="op">(</span></span>
<span>    i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sig_genes</span><span class="op">$</span><span class="va">model_type</span> <span class="op">==</span> <span class="st">"pairwise"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>    sig_genes <span class="op">=</span> <span class="va">sig_genes</span>,</span>
<span>    sce_layer <span class="op">=</span> <span class="va">sce_layer</span>,</span>
<span>    col_low_box <span class="op">=</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>    col_low_point <span class="op">=</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>    col_high_box <span class="op">=</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>,</span>
<span>    col_high_point <span class="op">=</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-2.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Paper colors displayed in the shiny app</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20200206</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/layer_boxplot.html">layer_boxplot</a></span><span class="op">(</span></span>
<span>    sig_genes <span class="op">=</span> <span class="va">sig_genes</span>,</span>
<span>    sce_layer <span class="op">=</span> <span class="va">sce_layer</span>,</span>
<span>    short_title <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    col_low_box <span class="op">=</span> <span class="st">"palegreen3"</span>,</span>
<span>    col_low_point <span class="op">=</span> <span class="st">"springgreen2"</span>,</span>
<span>    col_high_box <span class="op">=</span> <span class="st">"darkorange2"</span>,</span>
<span>    col_high_point <span class="op">=</span> <span class="st">"orange1"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/layer_boxplot-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="correlation-of-layer-level-statistics">Correlation of layer-level statistics<a class="anchor" aria-label="anchor" href="#correlation-of-layer-level-statistics"></a>
</h3>
<p>Just like we compressed our <code>spe</code> object into
<code>sce_layer</code> by pseudo-bulking <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>, we can do the same
for other single nucleus or single cell RNA sequencing datasets
(snRNA-seq, scRNA-seq) and then compute enrichment t-statistics (one
group vs the rest; could be one cell type vs the rest or one cluster of
cells vs the rest). In our analysis (Maynard, Collado-Torres, Weber, et
al., 2021), we did this for several datasets including one of our LIBD
Human DLPFC snRNA-seq data generated by Matthew N Tran et al <a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>. In <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) we include a small set of these
statistics for the 31 cell clusters identified by Matthew N Tran et
al.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the enrichment t-statistics derived from Tran et al's snRNA-seq</span></span>
<span><span class="co">## DLPFC data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 692  31</span></span>
<span><span class="va">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt;                      2 (1)       4 (1)      6 (1)      8 (1)     10 (1)</span></span>
<span><span class="co">#&gt; ENSG00000104419 -0.6720726  0.30024609  0.5854130  0.1358497  0.3981996</span></span>
<span><span class="co">#&gt; ENSG00000184007  0.2535838 -0.28998049 -1.8106052 -0.2096309 -0.4521719</span></span>
<span><span class="co">#&gt; ENSG00000240583 -0.2015831 -0.04053423 -0.5120807  0.1688405 -0.3969257</span></span>
<span><span class="co">#&gt;                     12 (1)</span></span>
<span><span class="co">#&gt; ENSG00000104419 -0.9731938</span></span>
<span><span class="co">#&gt; ENSG00000184007 -0.1840286</span></span>
<span><span class="co">#&gt; ENSG00000240583 -0.2086954</span></span></code></pre></div>
<p>The function <code><a href="../reference/layer_stat_cor.html">layer_stat_cor()</a></code> will take as input one
such matrix of statistics and correlate them against our layer
enrichment results (or other model types) using the subset of Ensembl
gene IDs that are observed in both tables.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute the correlation matrix of enrichment t-statistics between our data</span></span>
<span><span class="co">## and Tran et al's snRNA-seq data</span></span>
<span><span class="va">cor_stats_layer</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/layer_stat_cor.html">layer_stat_cor</a></span><span class="op">(</span></span>
<span>    <span class="va">tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer</span>,</span>
<span>    <span class="va">modeling_results</span>,</span>
<span>    <span class="st">"enrichment"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the correlation matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cor_stats_layer</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;               WM       Layer6     Layer5</span></span>
<span><span class="co">#&gt; 22 (3) 0.6824669 -0.009192291 -0.1934265</span></span>
<span><span class="co">#&gt; 3 (3)  0.7154122 -0.070042729 -0.2290574</span></span>
<span><span class="co">#&gt; 23 (3) 0.6637885 -0.031467704 -0.2018306</span></span>
<span><span class="co">#&gt; 17 (3) 0.6364983 -0.094216046 -0.2026147</span></span>
<span><span class="co">#&gt; 21 (3) 0.6281443 -0.050336358 -0.1988774</span></span>
<span><span class="co">#&gt; 7 (4)  0.1850724 -0.197283175 -0.2716890</span></span></code></pre></div>
<p>Once we have computed this correlation matrix, we can then visualize
it using <code><a href="../reference/layer_stat_cor_plot.html">layer_stat_cor_plot()</a></code> as shown below.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visualize the correlation matrix</span></span>
<span><span class="fu"><a href="../reference/layer_stat_cor_plot.html">layer_stat_cor_plot</a></span><span class="op">(</span><span class="va">cor_stats_layer</span>, max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">cor_stats_layer</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/layer_stat_cor_plot-1.png" width="700"></p>
<p>In order to fully interpret the resulting heatmap you need to know
what each of the cell clusters labels mean. In this case, the syntax is
<code>xx (Y)</code> where <code>xx</code> is the cluster number and
<code>Y</code> is:</p>
<ol style="list-style-type: decimal">
<li>
<code>Excit</code>: excitatory neurons.</li>
<li>
<code>Inhib</code>: inhibitory neurons.</li>
<li>
<code>Oligo</code>: oligodendrocytes.</li>
<li>
<code>Astro</code>: astrocytes.</li>
<li>
<code>OPC</code>: oligodendrocyte progenitor cells.</li>
<li>
<code>Drop</code>: an ambiguous cluster of cells that potentially
should be dropped.</li>
</ol>
<p>You can find the version with the full names <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/Layer_Guesses/pdf/snRNAseq_overlap_heatmap.pdf" class="external-link">here</a>
if you are interested in it.</p>
<p>The <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) allows users to upload CSV file
with these t-statistics, view the correlation heatmaps, download them,
and download the correlation matrix. An example CSV file is provided <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/data-raw/tstats_Human_DLPFC_snRNAseq_Nguyen_topLayer.csv" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="gene-set-enrichment">Gene set enrichment<a class="anchor" aria-label="anchor" href="#gene-set-enrichment"></a>
</h3>
<p>Many researchers have identified lists of genes that increase the
risk of a given disorder or disease, are differentially expressed in a
given experiment or set of conditions, have been described in a several
research papers, among other collections. We can ask for any of our
modeling results whether a list of genes is enriched among our
significant results.</p>
<p>For illustration purposes, we included in <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) a set of genes from the Simons
Foundation Autism Research Initiative <a href="https://www.sfari.org/" class="external-link">SFARI</a>. In our analysis (Maynard,
Collado-Torres, Weber, et al., 2021) we used more gene lists. Below, we
read in the data and create a <code><a href="https://rdrr.io/r/base/list.html" class="external-link">list()</a></code> object with the
Ensembl gene IDs that SFARI has provided that are related to autism.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Read in the SFARI gene sets included in the package</span></span>
<span><span class="va">asd_sfari</span> <span class="op">&lt;-</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>        <span class="st">"extdata"</span>,</span>
<span>        <span class="st">"SFARI-Gene_genes_01-03-2020release_02-04-2020export.csv"</span>,</span>
<span>        package <span class="op">=</span> <span class="st">"spatialLIBD"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    as.is <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Format them appropriately</span></span>
<span><span class="va">asd_sfari_geneList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Gene_SFARI_all <span class="op">=</span> <span class="va">asd_sfari</span><span class="op">$</span><span class="va">ensembl.id</span>,</span>
<span>    Gene_SFARI_high <span class="op">=</span> <span class="va">asd_sfari</span><span class="op">$</span><span class="va">ensembl.id</span><span class="op">[</span><span class="va">asd_sfari</span><span class="op">$</span><span class="va">gene.score</span> <span class="op">&lt;</span> <span class="fl">3</span><span class="op">]</span>,</span>
<span>    Gene_SFARI_syndromic <span class="op">=</span> <span class="va">asd_sfari</span><span class="op">$</span><span class="va">ensembl.id</span><span class="op">[</span><span class="va">asd_sfari</span><span class="op">$</span><span class="va">syndromic</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>After reading the list of genes, we can then compute the enrichment
odds ratios and p-values for a given FDR threshold in our
statistics.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compute the gene set enrichment results</span></span>
<span><span class="va">asd_sfari_enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_set_enrichment.html">gene_set_enrichment</a></span><span class="op">(</span></span>
<span>    gene_list <span class="op">=</span> <span class="va">asd_sfari_geneList</span>,</span>
<span>    modeling_results <span class="op">=</span> <span class="va">modeling_results</span>,</span>
<span>    model_type <span class="op">=</span> <span class="st">"enrichment"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">asd_sfari_enrichment</span><span class="op">)</span></span>
<span><span class="co">#&gt;          OR       Pval   test                   ID model_type fdr_cut</span></span>
<span><span class="co">#&gt; 1 1.2659915 0.00318733     WM       Gene_SFARI_all enrichment     0.1</span></span>
<span><span class="co">#&gt; 2 1.1819109 0.17753833     WM      Gene_SFARI_high enrichment     0.1</span></span>
<span><span class="co">#&gt; 3 1.2333378 0.31971056     WM Gene_SFARI_syndromic enrichment     0.1</span></span>
<span><span class="co">#&gt; 4 0.9702022 0.85164140 Layer1       Gene_SFARI_all enrichment     0.1</span></span>
<span><span class="co">#&gt; 5 0.7192630 0.14735979 Layer1      Gene_SFARI_high enrichment     0.1</span></span>
<span><span class="co">#&gt; 6 1.1216176 0.73790069 Layer1 Gene_SFARI_syndromic enrichment     0.1</span></span></code></pre></div>
<p>Using the above enrichment table, we can then visualize the odds
ratios on a heatmap as shown below. Note that we use the thresholded
p-values at <code>-log10(p) = 12</code> for visualization purposes and
only show the odds ratios for <code>-log10(p) &gt; 3</code> by
default.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visualize gene set enrichment results</span></span>
<span><span class="fu"><a href="../reference/gene_set_enrichment_plot.html">gene_set_enrichment_plot</a></span><span class="op">(</span></span>
<span>    <span class="va">asd_sfari_enrichment</span>,</span>
<span>    xlabs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">asd_sfari_enrichment</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    layerHeights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">40</span>, <span class="fl">55</span>, <span class="fl">75</span>, <span class="fl">85</span>, <span class="fl">110</span>, <span class="fl">120</span>, <span class="fl">135</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spatialLIBD_files/figure-html/sfari_enrichment_plot-1.png" width="700"></p>
<p>The <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022) allows users to upload CSV file
their gene lists, compute the enrichment statistics, visualize them,
download the PDF, and download the enrichment table. An example CSV file
is provided <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/data-raw/asd_sfari_geneList.csv" class="external-link">here</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="re-shaping-your-data-to-our-structure">Re-shaping your data to our structure<a class="anchor" aria-label="anchor" href="#re-shaping-your-data-to-our-structure"></a>
</h2>
<p>This section gets into the details of how we generated the data
(Maynard, Collado-Torres, Weber, et al., 2021) behind <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022). This could be useful if you are
a Bioconductor developer or a user very familiar with packages such as
<em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>.</p>
<div class="section level3">
<h3 id="spatialexperiment-support">SpatialExperiment support<a class="anchor" aria-label="anchor" href="#spatialexperiment-support"></a>
</h3>
<p>As of version 1.3.3, <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
supports the <code>SpatialExperiment</code> class from <em><a href="https://bioconductor.org/packages/3.15/SpatialExperiment" class="external-link">SpatialExperiment</a></em>.
The functions <code><a href="../reference/vis_gene_p.html">vis_gene_p()</a></code>, <code><a href="../reference/vis_gene.html">vis_gene()</a></code>,
<code><a href="../reference/vis_grid_clus.html">vis_grid_clus()</a></code>, <code><a href="../reference/vis_grid_gene.html">vis_grid_gene()</a></code>,
<code><a href="../reference/vis_clus.html">vis_clus()</a></code>, <code><a href="../reference/vis_clus_p.html">vis_clus_p()</a></code>,
<code><a href="../reference/geom_spatial.html">geom_spatial()</a></code> now work with <code>SpatialExperiment</code>
objects thanks to the updates in <em><a href="https://bioconductor.org/packages/3.15/SpatialExperiment" class="external-link">SpatialExperiment</a></em>.
If you have spot-level data formatted in the older
<code>SingleCellExperiment</code> objects that were heavily modified for
<code>spatialLIBD</code>, you can use <code><a href="../reference/sce_to_spe.html">sce_to_spe()</a></code> to
convert the objects.</p>
<p>This work was done by <a href="https://twitter.com/PardoBree" class="external-link">Brenda
Pardo</a> and Leonardo.</p>
</div>
<div class="section level3">
<h3 id="using-spatiallibd-with-your-own-data">Using spatialLIBD with your own data<a class="anchor" aria-label="anchor" href="#using-spatiallibd-with-your-own-data"></a>
</h3>
<p>Please check the second vignette on how to use <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
with your own data as exemplified with a public 10x Genomics dataset or
go directly to the <code><a href="../reference/read10xVisiumWrapper.html">read10xVisiumWrapper()</a></code>
documentation.</p>
</div>
<div class="section level3">
<h3 id="expected-characteristics-of-the-data">Expected characteristics of the data<a class="anchor" aria-label="anchor" href="#expected-characteristics-of-the-data"></a>
</h3>
<p>If you want to check the key characteristics required by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>’s
functions or the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application, use the <code>check_*</code> family of functions.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_spe.html">check_spe</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 33538 47681 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(9): source type ... gene_search is_top_hvg</span></span>
<span><span class="co">#&gt; colnames(47681): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span></span>
<span><span class="co">#&gt;   TTGTTTCCATACAACT-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">#&gt; colData names(69): sample_id Cluster ... array_row array_col</span></span>
<span><span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity50 ... TSNE_perplexity80</span></span>
<span><span class="co">#&gt;   UMAP_neighbors15</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span>
<span><span class="fu"><a href="../reference/check_sce_layer.html">check_sce_layer</a></span><span class="op">(</span><span class="va">sce_layer</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 22331 76 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(22331): ENSG00000243485 ENSG00000238009 ... ENSG00000278384</span></span>
<span><span class="co">#&gt;   ENSG00000271254</span></span>
<span><span class="co">#&gt; rowData names(10): source type ... is_top_hvg is_top_hvg_sce_layer</span></span>
<span><span class="co">#&gt; colnames(76): 151507_Layer1 151507_Layer2 ... 151676_Layer6 151676_WM</span></span>
<span><span class="co">#&gt; colData names(13): sample_name layer_guess ...</span></span>
<span><span class="co">#&gt;   layer_guess_reordered_short spatialLIBD</span></span>
<span><span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity5 ... UMAP_neighbors15 PCAsub</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">## The output here is too long to print</span></span>
<span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check_modeling_results.html">check_modeling_results</a></span><span class="op">(</span><span class="va">modeling_results</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">xx</span>, <span class="va">modeling_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="generating-our-data-legacy">Generating our data (legacy)<a class="anchor" aria-label="anchor" href="#generating-our-data-legacy"></a>
</h3>
<p>If you are interested in reshaping your data to fit our structure, we
do not provide a quick function to do so. That is intentional given the
active development by the Bioconductor community for determining the
best way to deal with spatial transcriptomics data in general and the
10x Visium data in particular. Having said that, we do have all the
steps and reproducibility information documented across several of our
analysis (Maynard, Collado-Torres, Weber, et al., 2021) scripts.</p>
<ul>
<li>
<code>reorganize_folder.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/reorganize_folder.R" class="external-link">here</a>
re-organizes the raw data we were sent by 10x Genomics.</li>
<li>
<code>Layer_Notebook.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/Layer_Notebook.R" class="external-link">here</a>
reads in the Visium data and builds a list of
<code>RangeSummarizedExperiment()</code> objects from <em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>,
one per sample (image) that is eventually saved as
<code>Human_DLPFC_Visium_processedData_rseList.rda</code>.</li>
<li>
<code>convert_sce.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/convert_sce.R" class="external-link">here</a>
reads in <code>Human_DLPFC_Visium_processedData_rseList.rda</code> and
builds an initial <code>sce</code> object with image data under
<code>metadata(sce)$image</code> which is a single data.frame.
Subsetting doesn’t automatically subset the image, so you have to do it
yourself when plotting as is done by <code><a href="../reference/vis_clus_p.html">vis_clus_p()</a></code> and
<code><a href="../reference/vis_gene_p.html">vis_gene_p()</a></code>. Having the data from all images in a single
object allows you to use the spot-level data from all images to compute
clusters and do other similar analyses to the ones you would do with
sc/snRNA-seq data. The script creates the
<code>Human_DLPFC_Visium_processedData_sce.Rdata</code> file.</li>
<li>
<code>sce_scran.R</code> available <a href="https://github.com/LieberInstitute/HumanPilot/blob/master/Analysis/sce_scran.R" class="external-link">here</a>
then uses <em><a href="https://bioconductor.org/packages/3.15/scran" class="external-link">scran</a></em> to
read in <code>Human_DLPFC_Visium_processedData_sce.Rdata</code>, compute
the highly variable genes (stored in our final <code>sce</code> object
at <code>rowData(sce)$is_top_hvg</code>), perform dimensionality
reduction (PCA, TSNE, UMAP) and identify clusters using the data from
all images. The resulting data is then stored as
<code>Human_DLPFC_Visium_processedData_sce_scran.Rdata</code> and is the
main object used throughout our analysis code (Maynard, Collado-Torres,
Weber, et al., 2021).</li>
<li>
<code>make-data_spatialLIBD.R</code> available in the source version
of <code>spatialLIBD</code> and <a href="https://github.com/LieberInstitute/spatialLIBD/blob/master/inst/scripts/make-data_spatialLIBD.R" class="external-link">online
here</a> is the script that reads in
<code>Human_DLPFC_Visium_processedData_sce_scran.Rdata</code> as well as
some other outputs from our analysis and combines them into the final
<code>sce</code> and <code>sce_layer</code> objects provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, et al., 2022). This script simplifies some
operations in order to simplify the code behind the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>.</li>
</ul>
<p>You don’t necessarily need to do all of this to use the functions
provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>.
Note that external to the R objects, for the <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>
application provided by <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
you will need to have the <code>tissue_lowres_image.png</code> image
files in a directory structure by sample as shown <a href="https://github.com/LieberInstitute/spatialLIBD/tree/master/inst/app/www/data" class="external-link">here</a>
in order for the interactive visualizations made with <em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em> to
work.</p>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/spatialLIBD" class="external-link">spatialLIBD</a></em>
package (Pardo, Spangler, Weber, et al., 2022) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/AnnotationHub" class="external-link">AnnotationHub</a></em>
(Morgan and Shepherd, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=benchmarkme" class="external-link">benchmarkme</a></em>
(Gillespie, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocFileCache" class="external-link">BiocFileCache</a></em>
(Shepherd and Morgan, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocGenerics" class="external-link">BiocGenerics</a></em>
(Huber, W., Carey, J., Gentleman, R., Anders, S., Carlson, M., Carvalho,
S., Bravo, C., Davis, S., Gatto, L., Girke, T., Gottardo, R., Hahne, F.,
Hansen, D., Irizarry, A., Lawrence, M., Love, I., MacDonald, J.,
Obenchain, V., {Ole’s}, K., {Pag`es}, H., Reyes, A., Shannon, P., Smyth,
K., Tenenbaum, D., Waldron, L., Morgan, and M., 2015)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=cowplot" class="external-link">cowplot</a></em>
(Wilke, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=DT" class="external-link">DT</a></em>
(Xie, Cheng, and Tan, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/edgeR" class="external-link">edgeR</a></em>
(Robinson, McCarthy, and Smyth, 2010)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=fields" class="external-link">fields</a></em>
(Douglas Nychka, Reinhard Furrer, John Paige, and Stephan Sain,
2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/GenomicRanges" class="external-link">GenomicRanges</a></em>
(Lawrence, Huber, Pagès, Aboyoun, Carlson, Gentleman, Morgan, and Carey,
2013)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em>
(Wickham, 2016)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=golem" class="external-link">golem</a></em> (Fay,
Guyader, Rochette, and Girard, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/IRanges" class="external-link">IRanges</a></em>
(Lawrence, Huber, Pagès, et al., 2013)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/limma" class="external-link">limma</a></em>
(Ritchie, Phipson, Wu, Hu, Law, Shi, and Smyth, 2015)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=magick" class="external-link">magick</a></em> (Ooms,
2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=Matrix" class="external-link">Matrix</a></em> (Bates,
Maechler, and Jagan, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=paletteer" class="external-link">paletteer</a></em>
(Hvitfeldt, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em>
(Sievert, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RColorBrewer" class="external-link">RColorBrewer</a></em>
(Neuwirth, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng,
Chang, and Iannone, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/rtracklayer" class="external-link">rtracklayer</a></em>
(Lawrence, Gentleman, and Carey, 2009)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/S4Vectors" class="external-link">S4Vectors</a></em>
(Pagès, Lawrence, and Aboyoun, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/scater" class="external-link">scater</a></em>
(McCarthy, Campbell, Lun, and Willis, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/scuttle" class="external-link">scuttle</a></em>
(McCarthy, Campbell, Lun, et al., 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>
(Amezquita, Lun, Becht, Carey, Carpp, Geistlinger, Marini, Rue-Albrecht,
Risso, Soneson, Waldron, Pages, Smith, Huber, Morgan, Gottardo, and
Hicks, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> (Chang,
Cheng, Allaire, et al., 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
(Righelli, Weber, Crowell, et al., 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=statmod" class="external-link">statmod</a></em>
(Giner and Smyth, 2016)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester, and Pagès, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=viridisLite" class="external-link">viridisLite</a></em>
(Garnier, Simon, Ross, Noam, Rudis, Robert, Camargo, Pedro, Sciaini,
Marco, Scherer, and Cédric, 2022)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"spatialLIBD.Rmd"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"spatialLIBD.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2022-09-15 16:45:52 EDT"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 22.288 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       macOS Monterey 12.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin20</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2022-09-15</span></span>
<span><span class="co">#&gt;  pandoc   2.17.1.1 @ /opt/homebrew/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi            1.58.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationHub            3.4.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  assertthat               0.2.1     <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  attempt                  0.3.1     <span style="color: #555555;">2020-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  beachmat                 2.12.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  beeswarm                 0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkme              1.0.8     <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkmeData          1.0.4     <span style="color: #555555;">2020-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                * 2.56.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache            2.4.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics           * 0.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocIO                   1.6.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager              1.30.18   <span style="color: #555555;">2022-05-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors            1.14.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocParallel             1.30.3    <span style="color: #555555;">2022-06-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocSingular             1.12.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle              * 2.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocVersion              3.15.2    <span style="color: #555555;">2022-03-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               2.64.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                      4.0.4     <span style="color: #555555;">2020-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bit64                    4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bitops                   1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  blob                     1.2.3     <span style="color: #555555;">2022-04-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bookdown                 0.28      <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib                    0.4.0     <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem                   1.0.6     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                      3.4.0     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  codetools                0.2-18    <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  colorspace               2.0-3     <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  config                   0.3.1     <span style="color: #555555;">2020-12-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cowplot                  1.1.1     <span style="color: #555555;">2020-12-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  crayon                   1.5.1     <span style="color: #555555;">2022-03-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  curl                     4.3.2     <span style="color: #555555;">2021-06-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  data.table               1.14.2    <span style="color: #555555;">2021-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DBI                      1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr                   2.2.1     <span style="color: #555555;">2022-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray             0.22.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedMatrixStats       1.18.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                     1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest                   0.6.29    <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  doParallel               1.0.17    <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dotCall64                1.0-1     <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                    1.0.10    <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  dqrng                    0.3.0     <span style="color: #555555;">2021-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DropletUtils             1.16.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DT                       0.25      <span style="color: #555555;">2022-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                    3.38.4    <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ellipsis                 0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate                 0.16      <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub            2.4.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  fansi                    1.0.3     <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  farver                   2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  fastmap                  1.1.0     <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fields                   14.1      <span style="color: #555555;">2022-08-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  filelock                 1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  foreach                  1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs                       1.5.2     <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  generics                 0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb           * 1.32.4    <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData         1.2.8     <span style="color: #555555;">2022-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments        1.32.1    <span style="color: #555555;">2022-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges          * 1.48.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm               0.6.0     <span style="color: #555555;">2017-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2                  3.3.6     <span style="color: #555555;">2022-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                  0.9.1     <span style="color: #555555;">2021-01-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  glue                     1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  golem                    0.3.3     <span style="color: #555555;">2022-07-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  gridExtra                2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gtable                   0.3.1     <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  HDF5Array                1.24.2    <span style="color: #555555;">2022-08-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  highr                    0.9       <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools                0.5.3     <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets              1.5.4     <span style="color: #555555;">2021-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                   1.6.6     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  httr                     1.4.4     <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  interactiveDisplayBase   1.34.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  IRanges                * 2.30.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  irlba                    2.3.5     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  iterators                1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib                0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite                 1.8.0     <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                 1.36.3    <span style="color: #555555;">2022-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                    1.40      <span style="color: #555555;">2022-08-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  labeling                 0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  later                    1.3.0     <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice                  0.20-45   <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval                 0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle                1.0.2     <span style="color: #555555;">2022-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  limma                    3.52.3    <span style="color: #555555;">2022-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  locfit                   1.5-9.6   <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  lubridate                1.8.0     <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magick                   2.7.3     <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr                 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  maps                     3.4.0     <span style="color: #555555;">2021-09-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                   1.4-1     <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics         * 1.8.1     <span style="color: #555555;">2022-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats            * 0.62.0    <span style="color: #555555;">2022-04-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise                  2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  mime                     0.12      <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  munsell                  0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  paletteer                1.4.1     <span style="color: #555555;">2022-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pillar                   1.8.1     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                  2.0.6     <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgload                  1.3.0     <span style="color: #555555;">2022-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plotly                   4.10.0    <span style="color: #555555;">2021-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plyr                     1.8.7     <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  png                      0.1-7     <span style="color: #555555;">2013-12-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  promises                 1.2.0.1   <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr                    0.3.4     <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R.methodsS3              1.8.2     <span style="color: #555555;">2022-06-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R.oo                     1.25.0    <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R.utils                  2.12.0    <span style="color: #555555;">2022-06-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R6                       2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg                     1.2.2     <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs                 0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer             1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                     1.0.9     <span style="color: #555555;">2022-07-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  RCurl                    1.98-1.8  <span style="color: #555555;">2022-07-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR             * 1.3.0     <span style="color: #555555;">2020-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rematch2                 2.1.2     <span style="color: #555555;">2020-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  restfulr                 0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5                    2.40.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rhdf5filters             1.8.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib                 1.18.2    <span style="color: #555555;">2022-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rjson                    0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang                    1.0.5     <span style="color: #555555;">2022-08-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                2.16      <span style="color: #555555;">2022-08-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  roxygen2                 7.2.1     <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot                2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools                2.12.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  RSQLite                  2.2.17    <span style="color: #555555;">2022-09-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rstudioapi               0.14      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                     1.0.5     <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rtracklayer              1.56.1    <span style="color: #555555;">2022-06-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Vectors              * 0.34.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                     0.4.2     <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix             1.4.1     <span style="color: #555555;">2022-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scales                   1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  scater                   1.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scuttle                  1.6.3     <span style="color: #555555;">2022-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sessioninfo            * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  shiny                    1.7.2     <span style="color: #555555;">2022-07-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets             0.7.3     <span style="color: #555555;">2022-08-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment   * 1.18.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  spam                     2.9-1     <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sparseMatrixStats        1.8.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  SpatialExperiment      * 1.6.1     <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  spatialLIBD            * 1.9.15    <span style="color: #555555;">2022-09-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  statmod                  1.4.37    <span style="color: #555555;">2022-08-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi                  1.7.8     <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  stringr                  1.4.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment   * 1.26.1    <span style="color: #555555;">2022-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts              1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping              0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tibble                   3.1.8     <span style="color: #555555;">2022-07-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  tidyr                    1.2.1     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect               1.1.2     <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  usethis                  2.1.6     <span style="color: #555555;">2022-05-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  utf8                     1.2.2     <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                    0.4.1     <span style="color: #555555;">2022-04-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vipor                    0.4.5     <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  viridis                  0.6.2     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite              0.4.1     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun                     0.33      <span style="color: #555555;">2022-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XML                      3.99-0.10 <span style="color: #555555;">2022-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xml2                     1.3.3     <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xtable                   1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector                  0.36.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                     2.3.5     <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                 1.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022), <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, McPherson, et al., 2022) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2022rmarkdown"></a><a href="#cite-allaire2022rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.16. 2022. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-amezquita2020orchestrating"></a><a href="#cite-amezquita2020orchestrating">[2]</a><cite>
R. Amezquita, A. Lun, E. Becht, et al. “Orchestrating single-cell
analysis with Bioconductor”. In: <em>Nature Methods</em> 17 (2020),
pp. 137–145. URL:
<a href="https://www.nature.com/articles/s41592-019-0654-x" class="external-link">https://www.nature.com/articles/s41592-019-0654-x</a>.</cite>
</p>
<p>
<a id="bib-bates2022matrix"></a><a href="#cite-bates2022matrix">[3]</a><cite>
D. Bates, M. Maechler, and M. Jagan. <em>Matrix: Sparse and Dense Matrix
Classes and Methods</em>. R package version 1.4-1. 2022. URL:
<a href="https://CRAN.R-project.org/package=Matrix" class="external-link">https://CRAN.R-project.org/package=Matrix</a>.</cite>
</p>
<p>
<a id="bib-chang2022shiny"></a><a href="#cite-chang2022shiny">[4]</a><cite>
W. Chang, J. Cheng, J. Allaire, et al. <em>shiny: Web Application
Framework for R</em>. R package version 1.7.2. 2022. URL:
<a href="https://CRAN.R-project.org/package=shiny" class="external-link">https://CRAN.R-project.org/package=shiny</a>.</cite>
</p>
<p>
<a id="bib-2021fields"></a><a href="#cite-2021fields">[5]</a><cite>
Douglas Nychka, Reinhard Furrer, John Paige, et al. <em>fields: Tools
for spatial data</em>. R package version 14.1. Boulder, CO, USA:
University Corporation for Atmospheric Research, 2021. URL:
<a href="https://github.com/dnychka/fieldsRPackage" class="external-link">https://github.com/dnychka/fieldsRPackage</a>.</cite>
</p>
<p>
<a id="bib-fay2022golem"></a><a href="#cite-fay2022golem">[6]</a><cite>
C. Fay, V. Guyader, S. Rochette, et al. <em>golem: A Framework for
Robust Shiny Applications</em>. R package version 0.3.3. 2022. URL:
<a href="https://CRAN.R-project.org/package=golem" class="external-link">https://CRAN.R-project.org/package=golem</a>.</cite>
</p>
<p>
<a id="bib-garnier2022viridis"></a><a href="#cite-garnier2022viridis">[7]</a><cite>
Garnier, Simon, Ross, et al. <em>viridis - Colorblind-Friendly Color
Maps for R</em>. R package version 0.4.1. 2022. DOI:
<a href="https://doi.org/10.5281/zenodo.4679424" class="external-link">10.5281/zenodo.4679424</a>.
URL:
<a href="https://sjmgarnier.github.io/viridis/" class="external-link">https://sjmgarnier.github.io/viridis/</a>.</cite>
</p>
<p>
<a id="bib-gillespie2022benchmarkme"></a><a href="#cite-gillespie2022benchmarkme">[8]</a><cite>
C. Gillespie. <em>benchmarkme: Crowd Sourced System Benchmarks</em>. R
package version 1.0.8. 2022. URL:
<a href="https://CRAN.R-project.org/package=benchmarkme" class="external-link">https://CRAN.R-project.org/package=benchmarkme</a>.</cite>
</p>
<p>
<a id="bib-giner2016statmod"></a><a href="#cite-giner2016statmod">[9]</a><cite>
G. Giner and G. K. Smyth. “statmod: probability calculations for the
inverse Gaussian distribution”. In: <em>R Journal</em> 8.1 (2016),
pp. 339-351.</cite>
</p>
<p>
<a id="bib-huber2015rchestrating"></a><a href="#cite-huber2015rchestrating">[10]</a><cite>
Huber, W., Carey, et al. “Orchestrating high-throughput genomic analysis
with Bioconductor”. In: <em>Nature Methods</em> 12.2 (2015),
pp. 115–121. URL:
<a href="http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html" class="external-link">http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html</a>.</cite>
</p>
<p>
<a id="bib-hvitfeldt2021paletteer"></a><a href="#cite-hvitfeldt2021paletteer">[11]</a><cite>
E. Hvitfeldt. <em>paletteer: Comprehensive Collection of Color
Palettes</em>. R package version 1.3.0. 2021. URL:
<a href="https://github.com/EmilHvitfeldt/paletteer" class="external-link">https://github.com/EmilHvitfeldt/paletteer</a>.</cite>
</p>
<p>
<a id="bib-lawrence2009rtracklayer"></a><a href="#cite-lawrence2009rtracklayer">[12]</a><cite>
M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for
interfacing with genome browsers”. In: <em>Bioinformatics</em> 25
(2009), pp. 1841-1842. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btp328" class="external-link">10.1093/bioinformatics/btp328</a>.
URL:
<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="external-link">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>.</cite>
</p>
<p>
<a id="bib-lawrence2013software"></a><a href="#cite-lawrence2013software">[13]</a><cite>
M. Lawrence, W. Huber, H. Pagès, et al. “Software for Computing and
Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8
2013). DOI:
<a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">10.1371/journal.pcbi.1003118</a>.
URL:
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118%7D." class="external-link"></a><a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="external-link uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.
</cite></p>


<p>
<a id="bib-maynard2021transcriptome"></a><a href="#cite-maynard2021transcriptome">[14]</a><cite>
K. R. Maynard, L. Collado-Torres, L. M. Weber, et al.
“Transcriptome-scale spatial gene expression in the human dorsolateral
prefrontal cortex”. In: <em>Nature Neuroscience</em> (2021). DOI:
<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">10.1038/s41593-020-00787-0</a>.
URL:
<a href="https://www.nature.com/articles/s41593-020-00787-0" class="external-link">https://www.nature.com/articles/s41593-020-00787-0</a>.</cite>
</p>
<p>
<a id="bib-mccarthy2017scater"></a><a href="#cite-mccarthy2017scater">[15]</a><cite>
D. J. McCarthy, K. R. Campbell, A. T. L. Lun, et al. “Scater:
pre-processing, quality control, normalisation and visualisation of
single-cell RNA-seq data in R”. In: <em>Bioinformatics</em> 33 (8 2017),
pp. 1179-1186. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btw777" class="external-link">10.1093/bioinformatics/btw777</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[16]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2022summarizedexperiment"></a><a href="#cite-morgan2022summarizedexperiment">[17]</a><cite>
M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment:
SummarizedExperiment container</em>. R package version 1.26.1. 2022.
URL:
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-morgan2022annotationhub"></a><a href="#cite-morgan2022annotationhub">[18]</a><cite>
M. Morgan and L. Shepherd. <em>AnnotationHub: Client to access
AnnotationHub resources</em>. R package version 3.4.0. 2022.</cite>
</p>
<p>
<a id="bib-morgan2022experimenthub"></a><a href="#cite-morgan2022experimenthub">[19]</a><cite>
M. Morgan and L. Shepherd. <em>ExperimentHub: Client to access
ExperimentHub resources</em>. R package version 2.4.0. 2022.</cite>
</p>
<p>
<a id="bib-neuwirth2022rcolorbrewer"></a><a href="#cite-neuwirth2022rcolorbrewer">[20]</a><cite>
E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package
version 1.1-3. 2022. URL:
<a href="https://CRAN.R-project.org/package=RColorBrewer" class="external-link">https://CRAN.R-project.org/package=RColorBrewer</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[21]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.24.0. 2022. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-ooms2021magick"></a><a href="#cite-ooms2021magick">[22]</a><cite>
J. Ooms. <em>magick: Advanced Graphics and Image-Processing in R</em>. R
package version 2.7.3. 2021. URL:
<a href="https://CRAN.R-project.org/package=magick" class="external-link">https://CRAN.R-project.org/package=magick</a>.</cite>
</p>
<p>
<a id="bib-pags2022s4vectors"></a><a href="#cite-pags2022s4vectors">[23]</a><cite>
H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: Foundation of
vector-like and list-like containers in Bioconductor</em>. R package
version 0.34.0. 2022. URL:
<a href="https://bioconductor.org/packages/S4Vectors" class="external-link">https://bioconductor.org/packages/S4Vectors</a>.</cite>
</p>
<p>
<a id="bib-pardo2022spatiallibd"></a><a href="#cite-pardo2022spatiallibd">[24]</a><cite>
B. Pardo, A. Spangler, L. M. Weber, et al. “spatialLIBD: an
R/Bioconductor package to visualize spatially-resolved transcriptomics
data”. In: <em>BMC Genomics</em> (2022). DOI:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">10.1186/s12864-022-08601-w</a>.
URL:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">https://doi.org/10.1186/s12864-022-08601-w</a>.</cite>
</p>
<p>
<a id="bib-2022language"></a><a href="#cite-2022language">[25]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-righelli2022spatialexperiment"></a><a href="#cite-righelli2022spatialexperiment">[26]</a><cite>
D. Righelli, L. M. Weber, H. L. Crowell, et al. “SpatialExperiment:
infrastructure for spatially-resolved transcriptomics data in R using
Bioconductor”. In: <em>Bioinformatics</em> 38.11 (2022), pp. -3. DOI:
<a href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac299" class="external-link">https://doi.org/10.1093/bioinformatics/btac299</a>.</cite>
</p>
<p>
<a id="bib-ritchie2015limma"></a><a href="#cite-ritchie2015limma">[27]</a><cite>
M. E. Ritchie, B. Phipson, D. Wu, et al. “limma powers differential
expression analyses for RNA-sequencing and microarray studies”. In:
<em>Nucleic Acids Research</em> 43.7 (2015), p. e47. DOI:
<a href="https://doi.org/10.1093/nar/gkv007" class="external-link">10.1093/nar/gkv007</a>.</cite>
</p>
<p>
<a id="bib-robinson2010edger"></a><a href="#cite-robinson2010edger">[28]</a><cite>
M. D. Robinson, D. J. McCarthy, and G. K. Smyth. “edgeR: a Bioconductor
package for differential expression analysis of digital gene expression
data”. In: <em>Bioinformatics</em> 26.1 (2010), pp. 139-140. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btp616" class="external-link">10.1093/bioinformatics/btp616</a>.</cite>
</p>
<p>
<a id="bib-shepherd2022biocfilecache"></a><a href="#cite-shepherd2022biocfilecache">[29]</a><cite>
L. Shepherd and M. Morgan. <em>BiocFileCache: Manage Files Across
Sessions</em>. R package version 2.4.0. 2022.</cite>
</p>
<p>
<a id="bib-sievert2020interactive"></a><a href="#cite-sievert2020interactive">[30]</a><cite>
C. Sievert. <em>Interactive Web-Based Data Visualization with R, plotly,
and shiny</em>. Chapman and Hall/CRC, 2020. ISBN: 9781138331457. URL:
<a href="https://plotly-r.com" class="external-link">https://plotly-r.com</a>.</cite>
</p>
<p>
<a id="bib-wickham2016ggplot2"></a><a href="#cite-wickham2016ggplot2">[31]</a><cite>
H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>.
Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. URL:
<a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[32]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[33]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2. 2021. URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-wilke2020cowplot"></a><a href="#cite-wilke2020cowplot">[34]</a><cite>
C. Wilke. <em>cowplot: Streamlined Plot Theme and Plot Annotations for
‘ggplot2’</em>. R package version 1.1.1. 2020. URL:
<a href="https://CRAN.R-project.org/package=cowplot" class="external-link">https://CRAN.R-project.org/package=cowplot</a>.</cite>
</p>
<p>
<a id="bib-xie2014knitr"></a><a href="#cite-xie2014knitr">[35]</a><cite>
Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”.
In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="external-link">http://www.crcpress.com/product/isbn/9781466561595</a>.</cite>
</p>
<p>
<a id="bib-xie2022wrapper"></a><a href="#cite-xie2022wrapper">[36]</a><cite>
Y. Xie, J. Cheng, and X. Tan. <em>DT: A Wrapper of the JavaScript
Library ‘DataTables’</em>. R package version 0.25. 2022. URL:
<a href="https://CRAN.R-project.org/package=DT" class="external-link">https://CRAN.R-project.org/package=DT</a>.</cite>
</p>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis" class="external-link">this
code</a> for details on how we built the <code>spe</code> object. In
particular check <code>convert_sce.R</code> and
<code>sce_scran.R</code>.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis/Layer_Guesses" class="external-link">this
code</a> for details on how we built the <code>sce_layer</code> object.
In particular check <code>spots_per_layer.R</code> and
<code>layer_enrichment.R</code>.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Check <a href="https://github.com/LieberInstitute/HumanPilot/tree/master/Analysis/Layer_Guesses" class="external-link">this
code</a> for details on how we built the <code>modeling_results</code>
object. In particular check <code>layer_specificity_fstats.R</code>,
<code>layer_specificity.R</code>, and <code>misc_numbers.R</code>.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>You can change the <code>destdir</code> argument and
specific a specific location that you will use and re-use. However the
default value of <code>destdir</code> is a temporary directory that will
be wiped out once you close your R session.<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>For more details, check <a href="https://github.com/LieberInstitute/HumanPilot/blob/ab52559957624538f8dc08b0533e4106fab4066b/Analysis/Layer_Guesses/layer_specificity.R" class="external-link">this
script</a>.<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>For more details, check <a href="https://github.com/LieberInstitute/HumanPilot/blob/ab52559957624538f8dc08b0533e4106fab4066b/Analysis/Layer_Guesses/dlpfc_snRNAseq_annotation.R" class="external-link">this
script</a>.<a href="#fnref6" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
