<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using spatialLIBD with 10x Genomics public datasets • spatialLIBD</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Using spatialLIBD with 10x Genomics public datasets">
<meta property="og:description" content="spatialLIBD">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spatialLIBD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.13</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/spatialLIBD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/TenX_data_download.html">Using spatialLIBD with 10x Genomics public datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/spatialLIBD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using spatialLIBD with 10x Genomics public
datasets</h1>
                        <h4 data-toc-skip class="author">Abby
Spangler</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:aspangle@gmail.com" class="email">aspangle@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Brenda
Pardo</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusLicenciatura de Ciencias Genómicas, Escuela Nacional de Estudios
Superiores Unidad Juriquilla, Universidad Nacional Autónoma de
México<br><a class="author_email" href="mailto:#"></a><a href="mailto:bpardo@lcgej.unam.mx" class="email">bpardo@lcgej.unam.mx</a>
      </address>
                              <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusCenter for Computational Biology, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">8 April 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/spatialLIBD/blob/HEAD/vignettes/TenX_data_download.Rmd" class="external-link"><code>vignettes/TenX_data_download.Rmd</code></a></small>
      <div class="hidden name"><code>TenX_data_download.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-spatiallibd">Install <code>spatialLIBD</code><a class="anchor" aria-label="anchor" href="#install-spatiallibd"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber, Hicks, Jaffe, Martinowich, Maynard, and
Collado-Torres, 2022) is an <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To run all the code in this vignette, you might need to install other
R/Bioconductor packages, which you can do with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>If you want to use the development version of
<code>spatialLIBD</code>, you will need to use the R version
corresponding to the current Bioconductor-devel branch as described in
more detail on the <a href="http://bioconductor.org/developers/how-to/useDevel/" class="external-link">Bioconductor
website</a>. Then you can install <code>spatialLIBD</code> from GitHub
using the following command.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/spatialLIBD"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p>Please first check the <em>Introduction to spatialLIBD</em> vignette
available through <a href="http://research.libd.org/spatialLIBD/articles/spatialLIBD.html" class="external-link">GitHub</a>
or <a href="https://bioconductor.org/packages/spatialLIBD" class="external-link">Bioconductor</a>.</p>
</div>
<div class="section level3">
<h3 id="citing-spatiallibd">Citing <code>spatialLIBD</code><a class="anchor" aria-label="anchor" href="#citing-spatiallibd"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"spatialLIBD"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite package 'spatialLIBD' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Pardo B, Spangler A, Weber LM, Hicks SC, Jaffe AE, Martinowich K,</span></span>
<span><span class="co">#&gt;   Maynard KR, Collado-Torres L (2022). "spatialLIBD: an R/Bioconductor</span></span>
<span><span class="co">#&gt;   package to visualize spatially-resolved transcriptomics data." _BMC</span></span>
<span><span class="co">#&gt;   Genomics_. doi:10.1186/s12864-022-08601-w</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1186/s12864-022-08601-w&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1186/s12864-022-08601-w&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Maynard KR, Collado-Torres L, Weber LM, Uytingco C, Barry BK,</span></span>
<span><span class="co">#&gt;   Williams SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y,</span></span>
<span><span class="co">#&gt;   Kleinman JE, Hyde TM, Rao N, Hicks SC, Martinowich K, Jaffe AE</span></span>
<span><span class="co">#&gt;   (2021). "Transcriptome-scale spatial gene expression in the human</span></span>
<span><span class="co">#&gt;   dorsolateral prefrontal cortex." _Nature Neuroscience_.</span></span>
<span><span class="co">#&gt;   doi:10.1038/s41593-020-00787-0</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1038/s41593-020-00787-0&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.nature.com/articles/s41593-020-00787-0&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Huuki-Myers LA, Spangler A, Eagles NJ, Montgomergy KD, Kwon SH, Guo</span></span>
<span><span class="co">#&gt;   B, Grant-Peters M, Divecha HR, Tippani M, Sriworarat C, Nguyen AB,</span></span>
<span><span class="co">#&gt;   Ravichandran P, Tran MN, Seyedian A, Consortium P, Hyde TM, Kleinman</span></span>
<span><span class="co">#&gt;   JE, Battle A, Page SC, Ryten M, Hicks SC, Martinowich K,</span></span>
<span><span class="co">#&gt;   Collado-Torres L, Maynard KR (2023). "Integrated single cell and</span></span>
<span><span class="co">#&gt;   unsupervised spatial transcriptomic analysis defines molecular</span></span>
<span><span class="co">#&gt;   anatomy of the human dorsolateral prefrontal cortex." _bioRxiv_.</span></span>
<span><span class="co">#&gt;   doi:10.1101/2023.02.15.528722</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2023.02.15.528722&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.biorxiv.org/content/10.1101/2023.02.15.528722v1&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Kwon SH, Parthiban S, Tippani M, Divecha HR, Eagles NJ, Lobana JS,</span></span>
<span><span class="co">#&gt;   Williams S, Mak M, Yu G, Avalos-Gracia J, Bharadwaj RA, Kleinman JE,</span></span>
<span><span class="co">#&gt;   Hyde TM, Page SC, Hicks SC, Martinowich K, Maynard KR, Collado-Torres</span></span>
<span><span class="co">#&gt;   L (2023). "Influence of Alzheimer’s disease related neuropathology on</span></span>
<span><span class="co">#&gt;   local microenvironment gene expression in the human inferior temporal</span></span>
<span><span class="co">#&gt;   cortex." _bioRxiv_. doi:10.1101/2023.04.TODO</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2023.04.TODO&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.biorxiv.org/content/10.1101/2023.04.TODO&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="download-data-from-10x-genomics">Download data from 10x Genomics<a class="anchor" aria-label="anchor" href="#download-data-from-10x-genomics"></a>
</h2>
<p>In this vignette we’ll show you how you can use <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
(Pardo, Spangler, Weber et al., 2022) for exploring spatially resolved
transcriptomics data from the <a href="https://www.10xgenomics.com/products/spatial-gene-expression" class="external-link">Visium
platform by 10x Genomics</a>. That is, you will learn how to use <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
for data beyond the one it was initially developed for (Maynard,
Collado-Torres, Weber, Uytingco, Barry, Williams, II, Tran, Besich,
Tippani, Chew, Yin, Kleinman, Hyde, Rao, Hicks, Martinowich, and Jaffe,
2021). To illustrate these steps, we will use data that 10x Genomics
made publicly available at <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets" class="external-link uri">https://support.10xgenomics.com/spatial-gene-expression/datasets</a>.
We will use files from the human lymph node example publicly available
at <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Human_Lymph_Node" class="external-link uri">https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Human_Lymph_Node</a>.</p>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<p>To get started, lets load the different packages we’ll need for this
vignette. Here’s a brief summary of why we need these packages:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocFileCache" class="external-link">BiocFileCache</a></em>:
for downloading and saving a local cache of the data</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/SpatialExperiment" class="external-link">SpatialExperiment</a></em>:
for reading the <code>spaceranger</code> files provided by 10x
Genomics</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/rtracklayer" class="external-link">rtracklayer</a></em>:
for importing a gene annotation GTF file</li>
<li>
<em><a href="https://CRAN.R-project.org/package=lobstr" class="external-link">lobstr</a></em>: for
checking how much memory our object is using</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>:
for launching an interactive website to explore the data</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load packages in the order we'll use them next</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BiocFileCache"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">"SpatialExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"rtracklayer"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lobstr.r-lib.org/" class="external-link">"lobstr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">"spatialLIBD"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-spaceranger-output-files">Download spaceranger output files<a class="anchor" aria-label="anchor" href="#download-spaceranger-output-files"></a>
</h3>
<p>Next we download data from 10x Genomics available from the human
lymph node example, available at <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Human_Lymph_Node" class="external-link uri">https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Human_Lymph_Node</a>.
We don’t need to download all the files listed there since <em><a href="https://bioconductor.org/packages/3.16/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
doesn’t need all of them for importing the data into R. These files are
part of the output that gets generated by <code>spaceranger</code> which
is the processing pipeline provided by 10x Genomics for Visium data.</p>
<p>We’ll use <em><a href="https://bioconductor.org/packages/3.16/BiocFileCache" class="external-link">BiocFileCache</a></em>
to keep the data in a local cache in case we want to run this example
again and don’t want to re-download the data from the web.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download and save a local cache of the data provided by 10x Genomics</span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">lymph.url</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"https://cf.10xgenomics.com/samples/spatial-exp/"</span>,</span>
<span>        <span class="st">"1.1.0/V1_Human_Lymph_Node/"</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"V1_Human_Lymph_Node_filtered_feature_bc_matrix.tar.gz"</span>,</span>
<span>            <span class="st">"V1_Human_Lymph_Node_spatial.tar.gz"</span>,</span>
<span>            <span class="st">"V1_Human_Lymph_Node_analysis.tar.gz"</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">lymph.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lymph.url</span>, <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span>, x <span class="op">=</span> <span class="va">bfc</span><span class="op">)</span></span></code></pre></div>
<p>10x Genomics provides the files in compressed tarballs
(<code>.tar.gz</code> file extension). Which is why we’ll need to use
<code><a href="https://rdrr.io/r/utils/untar.html" class="external-link">utils::untar()</a></code> to decompress the files. This will create
new directories and we will use <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files()</a></code> to see what
files these directories contain.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the files to a temporary location</span></span>
<span><span class="co">## (they'll be deleted once you close your R session)</span></span>
<span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">lymph.data</span>, <span class="fu">utils</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"outs"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## The names are the URLs, which are long and thus too wide to be shown here,</span></span>
<span><span class="co">## so we shorten them to only show the file name prior to displaying the</span></span>
<span><span class="co">## utils::untar() output status</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">xx</span></span>
<span><span class="co">#&gt; V1_Human_Lymph_Node_filtered_feature_bc_matrix.tar.gz.BFC1 </span></span>
<span><span class="co">#&gt;                                                          0 </span></span>
<span><span class="co">#&gt;                    V1_Human_Lymph_Node_spatial.tar.gz.BFC2 </span></span>
<span><span class="co">#&gt;                                                          0 </span></span>
<span><span class="co">#&gt;                   V1_Human_Lymph_Node_analysis.tar.gz.BFC3 </span></span>
<span><span class="co">#&gt;                                                          0</span></span>
<span></span>
<span><span class="co">## List the files we downloaded and extracted</span></span>
<span><span class="co">## These files are typically SpaceRanger outputs</span></span>
<span><span class="va">lymph.dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"outs"</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"filtered_feature_bc_matrix"</span>, <span class="st">"spatial"</span>, <span class="st">"raw_feature_bc_matrix"</span>, <span class="st">"analysis"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">lymph.dirs</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "aligned_fiducials.jpg"     "barcodes.tsv.gz"          </span></span>
<span><span class="co">#&gt;  [3] "clustering"                "detected_tissue_image.jpg"</span></span>
<span><span class="co">#&gt;  [5] "diffexp"                   "features.tsv.gz"          </span></span>
<span><span class="co">#&gt;  [7] "matrix.mtx.gz"             "pca"                      </span></span>
<span><span class="co">#&gt;  [9] "scalefactors_json.json"    "tissue_hires_image.png"   </span></span>
<span><span class="co">#&gt; [11] "tissue_lowres_image.png"   "tissue_positions_list.csv"</span></span>
<span><span class="co">#&gt; [13] "tsne"                      "umap"</span></span></code></pre></div>
<p>Now that we have the files that we need, we can import the data into
R using <code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">read10xVisium()</a></code> from <em><a href="https://bioconductor.org/packages/3.16/SpatialExperiment" class="external-link">SpatialExperiment</a></em>.
We’ll import the low resolution histology images produced by
<code>spaceranger</code> using the <code>images = "lowres"</code> and
<code>load = TRUE</code> arguments. We’ll also load the filtered gene
expression data using the <code>data = "filtered"</code> argument. The
count matrix can still be quite large, which is why we’ll use the
<code>type = "sparse"</code> argument to load the data into an R object
that is memory-efficient for sparse data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Import the data as a SpatialExperiment object</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">read10xVisium</a></span><span class="op">(</span></span>
<span>    samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    sample_id <span class="op">=</span> <span class="st">"lymph"</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"sparse"</span>, data <span class="op">=</span> <span class="st">"filtered"</span>,</span>
<span>    images <span class="op">=</span> <span class="st">"lowres"</span>, load <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'as(&lt;dgTMatrix&gt;, "dgCMatrix")' is deprecated.</span></span>
<span><span class="co">#&gt; Use 'as(., "CsparseMatrix")' instead.</span></span>
<span><span class="co">#&gt; See help("Deprecated") and help("Matrix-deprecated").</span></span>
<span><span class="co">## Inspect the R object we just created: class, memory, and how it looks in</span></span>
<span><span class="co">## general</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "SpatialExperiment"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "SpatialExperiment"</span></span>
<span><span class="fu">lobstr</span><span class="fu">::</span><span class="fu"><a href="https://lobstr.r-lib.org/reference/obj_size.html" class="external-link">obj_size</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1024</span><span class="op">^</span><span class="fl">2</span> <span class="co">## Convert to MB</span></span>
<span><span class="co">#&gt; 281.90 B</span></span>
<span><span class="va">spe</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 36601 4035 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817</span></span>
<span><span class="co">#&gt;   ENSG00000277196</span></span>
<span><span class="co">#&gt; rowData names(1): symbol</span></span>
<span><span class="co">#&gt; colnames(4035): AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 ...</span></span>
<span><span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">#&gt; colData names(4): in_tissue array_row array_col sample_id</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span>
<span></span>
<span><span class="co">## The counts are saved in a sparse matrix R object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "dgCMatrix"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "Matrix"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="modify-spe-for-spatiallibd">Modify spe for spatialLIBD<a class="anchor" aria-label="anchor" href="#modify-spe-for-spatiallibd"></a>
</h2>
<p>Now that we have an <code>SpatialExperiment</code> R object
(<code>spe</code>) with the data from 10x Genomics for the human lymph
node example, we need to add a few features to the R object in order to
create the interactive website using
<code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code>. These additional elements power
features in the interactive website that you might be interested in.</p>
<p>First we start with adding a few variables to the sample information
table (<code>colData()</code>) of our <code>spe</code> object. We
add:</p>
<ul>
<li>
<code>key</code>: this labels each spot with a unique identifier. We
combine the sample ID with the spot barcode ID to create this unique
identifier.</li>
<li>
<code>sum_umi</code>: this continuous variable contains the total
number of counts for each sample prior to filtering any genes.</li>
<li>
<code>sum_gene</code>: this continuous variable contains the number
of genes that have at least 1 count.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add some information used by spatialLIBD</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_key.html">add_key</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">sum_umi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">sum_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="add-gene-annotation-information">Add gene annotation information<a class="anchor" aria-label="anchor" href="#add-gene-annotation-information"></a>
</h3>
<p>The files <code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">SpatialExperiment::read10xVisium()</a></code> uses to
read in the <code>spaceranger</code> outputs into R do not include much
information about the genes, such as their chromosomes, coordinates, and
other gene annotation information. We thus recommend that you read in
this information from a gene annotation file: typically a
<code>gtf</code> file. For a real case scenario, you’ll mostly likely
have access to the GTF file provided by 10x Genomics. However, we cannot
download that file without downloading other files for this example.
Thus we’ll show you the code you would use if you had access to the GTF
file from 10x Genomics and also show a second approach that works for
this vignette.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Initially we don't have much information about the genes</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRangesList object of length 36601:</span></span>
<span><span class="co">#&gt; $ENSG00000243485</span></span>
<span><span class="co">#&gt; GRanges object with 0 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;    seqnames    ranges strand</span></span>
<span><span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: no sequences</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSG00000237613</span></span>
<span><span class="co">#&gt; GRanges object with 0 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;    seqnames    ranges strand</span></span>
<span><span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: no sequences</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSG00000186092</span></span>
<span><span class="co">#&gt; GRanges object with 0 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;    seqnames    ranges strand</span></span>
<span><span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: no sequences</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ...</span></span>
<span><span class="co">#&gt; &lt;36598 more elements&gt;</span></span></code></pre></div>
<div class="section level4">
<h4 id="from-10x">From 10x<a class="anchor" aria-label="anchor" href="#from-10x"></a>
</h4>
<p>Depending on the version of <code>spaceranger</code> you used, you
might have used different GTF files 10x Genomics has made available at
<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest" class="external-link uri">https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest</a>
and described at <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build" class="external-link uri">https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build</a>.
These files are too big though and we won’t download them in this
example. For instance, <em>References - 2020-A (July 7, 2020)</em> for
<em>Human reference (GRCh38)</em> is 11 GB in size and contains files we
do not need for this vignette. If you did have the file locally, you
could use the following code to read in the GTF file prepared by 10x
Genomics and add the information into your <code>spe</code> object that
<code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">SpatialExperiment::read10xVisium()</a></code> does not include.</p>
<p>For example, in our computing cluster this GTF file is located at the
following path and is 1.4 GB in size:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> cd /dcs04/lieber/lcolladotor/annotationFiles_LIBD001/10x/refdata-gex-GRCh38-2020-A</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> du <span class="at">-sh</span> <span class="at">--apparent-size</span> genes/genes.gtf</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="ex">1.4G</span>    genes/genes.gtf</span></code></pre></div>
<p>If you have the GTF file from 10x Genomics, we show next how you can
read the information into R, match it appropriately with the information
in the <code>spe</code> object and add it back into the <code>spe</code>
object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## You could:</span></span>
<span><span class="co">## * download the 11 GB file from</span></span>
<span><span class="co">## https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2020-A.tar.gz</span></span>
<span><span class="co">## * decompress it</span></span>
<span></span>
<span><span class="co">## Read in the gene information from the annotation GTF file provided by 10x</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span></span>
<span>        <span class="st">"/path/to/refdata-gex-GRCh38-2020-A/genes/genes.gtf"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">## Subject to genes only</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">[</span><span class="va">gtf</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Set the names to be the gene IDs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="co">## Match the genes</span></span>
<span><span class="va">match_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## They should all be present if you are using the correct GTF file from 10x</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">match_genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Keep only some columns from the gtf (you could keep all of them if you want)</span></span>
<span><span class="fu">mcols</span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">mcols</span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"source"</span>,</span>
<span>        <span class="st">"type"</span>,</span>
<span>        <span class="st">"gene_id"</span>,</span>
<span>        <span class="st">"gene_version"</span>,</span>
<span>        <span class="st">"gene_name"</span>,</span>
<span>        <span class="st">"gene_type"</span></span>
<span>    <span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Add the gene info to our SPE object</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">[</span><span class="va">match_genes</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Inspect the gene annotation data we added</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="from-gencode">From Gencode<a class="anchor" aria-label="anchor" href="#from-gencode"></a>
</h4>
<p>In this vignette, we’ll use the GTF file from Gencode v32. That’s
because the build notes from <em>References - 2020-A (July 7, 2020)</em>
and <em>Human reference, GRCh38 (GENCODE v32/Ensembl 98)</em> at <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A" class="external-link uri">https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A</a>
show that 10x Genomics used Gencode v32. They also used Ensembl version
98 which is why a few genes we have in our object are going to be
missing. We show next how you can read the information into R, match it
appropriately with the information in the <code>spe</code> object and
add it back into the <code>spe</code> object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Download the Gencode v32 GTF file and cache it</span></span>
<span><span class="va">gtf_cache</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span></span>
<span>    <span class="va">bfc</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/"</span>,</span>
<span>        <span class="st">"release_32/gencode.v32.annotation.gtf.gz"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Show the GTF cache location</span></span>
<span><span class="va">gtf_cache</span></span>
<span><span class="co">#&gt;                                                                            BFC4 </span></span>
<span><span class="co">#&gt; "/github/home/.cache/R/BiocFileCache/36754084bc5_gencode.v32.annotation.gtf.gz"</span></span>
<span></span>
<span><span class="co">## Import into R (takes ~1 min)</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">gtf_cache</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Subset to genes only</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">[</span><span class="va">gtf</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Remove the .x part of the gene IDs</span></span>
<span><span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set the names to be the gene IDs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="co">## Match the genes</span></span>
<span><span class="va">match_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">match_genes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 36572    29</span></span>
<span></span>
<span><span class="co">## Drop the few genes for which we don't have information</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">match_genes</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">match_genes</span> <span class="op">&lt;-</span> <span class="va">match_genes</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">match_genes</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Keep only some columns from the gtf</span></span>
<span><span class="fu">mcols</span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source"</span>, <span class="st">"type"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"gene_type"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Add the gene info to our SPE object</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gtf</span><span class="op">[</span><span class="va">match_genes</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Inspect the gene annotation data we added</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 36572 ranges and 5 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames       ranges strand |   source     type</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;    &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000243485     chr1  29554-31109      + |   HAVANA     gene</span></span>
<span><span class="co">#&gt;   ENSG00000237613     chr1  34554-36081      - |   HAVANA     gene</span></span>
<span><span class="co">#&gt;   ENSG00000186092     chr1  65419-71585      + |   HAVANA     gene</span></span>
<span><span class="co">#&gt;   ENSG00000238009     chr1 89295-133723      - |   HAVANA     gene</span></span>
<span><span class="co">#&gt;   ENSG00000239945     chr1  89551-91105      - |   HAVANA     gene</span></span>
<span><span class="co">#&gt;               ...      ...          ...    ... .      ...      ...</span></span>
<span><span class="co">#&gt;   ENSG00000212907     chrM  10470-10766      + |  ENSEMBL     gene</span></span>
<span><span class="co">#&gt;   ENSG00000198886     chrM  10760-12137      + |  ENSEMBL     gene</span></span>
<span><span class="co">#&gt;   ENSG00000198786     chrM  12337-14148      + |  ENSEMBL     gene</span></span>
<span><span class="co">#&gt;   ENSG00000198695     chrM  14149-14673      - |  ENSEMBL     gene</span></span>
<span><span class="co">#&gt;   ENSG00000198727     chrM  14747-15887      + |  ENSEMBL     gene</span></span>
<span><span class="co">#&gt;                           gene_id   gene_name      gene_type</span></span>
<span><span class="co">#&gt;                       &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000243485 ENSG00000243485 MIR1302-2HG         lncRNA</span></span>
<span><span class="co">#&gt;   ENSG00000237613 ENSG00000237613     FAM138A         lncRNA</span></span>
<span><span class="co">#&gt;   ENSG00000186092 ENSG00000186092       OR4F5 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000238009 ENSG00000238009  AL627309.1         lncRNA</span></span>
<span><span class="co">#&gt;   ENSG00000239945 ENSG00000239945  AL627309.3         lncRNA</span></span>
<span><span class="co">#&gt;               ...             ...         ...            ...</span></span>
<span><span class="co">#&gt;   ENSG00000212907 ENSG00000212907     MT-ND4L protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000198886 ENSG00000198886      MT-ND4 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000198786 ENSG00000198786      MT-ND5 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000198695 ENSG00000198695      MT-ND6 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000198727 ENSG00000198727      MT-CYB protein_coding</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="enable-a-friendlier-gene-search">Enable a friendlier gene search<a class="anchor" aria-label="anchor" href="#enable-a-friendlier-gene-search"></a>
</h4>
<p>Regardless of which method you used to obtain the gene annotation
information, we can now proceed by adding the gene symbol and gene ID
information that helps users search for genes in the shiny app produced
by <code>spatialLIBD</code>. This will enable users to search genes by
gene symbol or gene ID. If you didn’t do this, users would only be able
to search genes by gene ID which makes the web application harder to
use.</p>
<p>We also compute the total expression for the mitochondrial chromosome
(chrM) as well as the ratio of chrM expression. Both of these continuous
variables are interesting to explore and in some situations could be
useful for biological interpretations. For instance, in our pilot data
(Maynard, Collado-Torres, Weber et al., 2021), we noticed that the
<code>expr_chrM_ratio</code> was associated to DLPFC layers. That is,
spots with high <code>expr_chrM_ratio</code> were not randomly located
in our Visium slides.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add information used by spatialLIBD</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_search</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>    <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>, <span class="st">"; "</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Compute chrM expression and chrM expression ratio</span></span>
<span><span class="va">is_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">==</span> <span class="st">"chrM"</span><span class="op">)</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">expr_chrM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="va">is_mito</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">expr_chrM_ratio</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">$</span><span class="va">expr_chrM</span> <span class="op">/</span> <span class="va">spe</span><span class="op">$</span><span class="va">sum_umi</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="filter-the-spe-object">Filter the spe object<a class="anchor" aria-label="anchor" href="#filter-the-spe-object"></a>
</h3>
<p>We can now continue with some filtering steps since this can help
reduce the object size in memory as well as make it ready to use for
downstream processing tools such as those from the <em><a href="https://bioconductor.org/packages/3.16/scran" class="external-link">scran</a></em> and
<em><a href="https://bioconductor.org/packages/3.16/scuttle" class="external-link">scuttle</a></em>
packages. Though these steps are not absolutely necessary.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Remove genes with no data</span></span>
<span><span class="va">no_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Number of genes with no counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">no_expr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 11397</span></span>
<span></span>
<span><span class="co">## Compute the percent of genes with no counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">no_expr</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span></span>
<span><span class="co">#&gt; [1] 31.16318</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span><span class="op">-</span><span class="va">no_expr</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Remove spots without counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">sum_umi</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;      23   15917   20239   20738   25252   54931</span></span>
<span></span>
<span><span class="co">## If we had spots with no counts, we would remove them</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">sum_umi</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">spots_no_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">sum_umi</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span>    <span class="co">## Number of spots with no counts</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">spots_no_counts</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co">## Percent of spots with no counts</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">spots_no_counts</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span>    <span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="op">-</span><span class="va">spots_no_counts</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="check-object">Check object<a class="anchor" aria-label="anchor" href="#check-object"></a>
</h3>
<p>Next, we add the <code>ManualAnnotation</code> variable to the sample
information table (<code>colData()</code>) with <code>"NA"</code>. That
variable is used by the interactive website to store any manual
annotations.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add a variable for saving the manual annotations</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">ManualAnnotation</span> <span class="op">&lt;-</span> <span class="st">"NA"</span></span></code></pre></div>
<p>Finally, we can now check the final object using
<code><a href="../reference/check_spe.html">spatialLIBD::check_spe()</a></code>. This is a helper function that
will warn us if some important element is missing in <code>spe</code>
that we use later for the interactive website. If it all goes well, it
will return the original <code>spe</code> object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Check the final dimensions and object size</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 25175  4035</span></span>
<span><span class="fu">lobstr</span><span class="fu">::</span><span class="fu"><a href="https://lobstr.r-lib.org/reference/obj_size.html" class="external-link">obj_size</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1024</span><span class="op">^</span><span class="fl">2</span> <span class="co">## Convert to MB</span></span>
<span><span class="co">#&gt; 283.86 B</span></span>
<span></span>
<span><span class="co">## Run check_spe() function</span></span>
<span><span class="fu"><a href="../reference/check_spe.html">check_spe</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 25175 4035 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(25175): ENSG00000238009 ENSG00000241860 ... ENSG00000198695</span></span>
<span><span class="co">#&gt;   ENSG00000198727</span></span>
<span><span class="co">#&gt; rowData names(6): source type ... gene_type gene_search</span></span>
<span><span class="co">#&gt; colnames(4035): AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 ...</span></span>
<span><span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">#&gt; colData names(10): in_tissue array_row ... expr_chrM_ratio</span></span>
<span><span class="co">#&gt;   ManualAnnotation</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="explore-the-data">Explore the data<a class="anchor" aria-label="anchor" href="#explore-the-data"></a>
</h2>
<p>With our complete <code>spe</code> object, we can now use <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
for visualizing our data. We can do so using functions such as
<code><a href="../reference/vis_gene.html">vis_gene()</a></code> and <code><a href="../reference/vis_clus.html">vis_clus()</a></code> that are described
in more detail at the <em>Introduction to spatialLIBD</em> vignette
available through <a href="http://research.libd.org/spatialLIBD/articles/spatialLIBD.html" class="external-link">GitHub</a>
or <a href="https://bioconductor.org/packages/spatialLIBD" class="external-link">Bioconductor</a>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example visualizations. Let's start with a continuous variable.</span></span>
<span><span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="../reference/vis_gene.html">vis_gene</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"lymph"</span>,</span>
<span>    geneid <span class="op">=</span> <span class="st">"sum_umi"</span>,</span>
<span>    assayname <span class="op">=</span> <span class="st">"counts"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="TenX_data_download_files/figure-html/test_visualizations-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## We next create a random cluster label to visualize</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20210428</span><span class="op">)</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">random_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Next we visualize that random cluster</span></span>
<span><span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="../reference/vis_clus.html">vis_clus</a></span><span class="op">(</span></span>
<span>    spe <span class="op">=</span> <span class="va">spe</span>,</span>
<span>    sampleid <span class="op">=</span> <span class="st">"lymph"</span>,</span>
<span>    clustervar <span class="op">=</span> <span class="st">"random_cluster"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="TenX_data_download_files/figure-html/test_visualizations-2.png" width="700"></p>
<div class="section level3">
<h3 id="run-the-interactive-website">Run the interactive website<a class="anchor" aria-label="anchor" href="#run-the-interactive-website"></a>
</h3>
<p>We are now ready to create our interactive website for the human
lymph node data. The interactive website is a <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> web
application that uses <em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em> to
power several of the interactive features. We can create the interactive
website using the <code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code> function. The
default arguments of that function are customized for the data from our
initial study (Maynard, Collado-Torres, Weber et al., 2021), so we will
need to make some adjustments:</p>
<ul>
<li>
<code>sce_layer</code>, <code>modeling_results</code> and
<code>sig_genes</code> will be set to <code>NULL</code> since do not
have any pseudo-bulk results for this example data.</li>
<li>
<code>title</code>: we will use a custom title that reflect our
data</li>
<li>
<code>spe_discreate_vars</code>: we don’t have really any discrete
variables to show beyond <code>ManualAnnotation</code> which is used for
the manual annotations and <code>random_cluster</code> that we created
in the previous section.</li>
<li>
<code>spe_continous_vars</code>: we have computed several continuous
variables while adapting our <code>spe</code> object for <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>,
so we’ll list these variables below in order to visually inspect
them.</li>
<li>
<code>default_cluster</code>: this is used for indicating the
default discrete variable and for now we’ll set it to our
<code>random_cluster</code>.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run our shiny app</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span></span>
<span>        <span class="va">spe</span>,</span>
<span>        sce_layer <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        modeling_results <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        sig_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        title <span class="op">=</span> <span class="st">"spatialLIBD: human lymph node by 10x Genomics"</span>,</span>
<span>        spe_discrete_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"random_cluster"</span>, <span class="st">"ManualAnnotation"</span><span class="op">)</span>,</span>
<span>        spe_continuous_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum_umi"</span>, <span class="st">"sum_gene"</span>, <span class="st">"expr_chrM"</span>, <span class="st">"expr_chrM_ratio"</span><span class="op">)</span>,</span>
<span>        default_cluster <span class="op">=</span> <span class="st">"random_cluster"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>We also recommend creating custom website documentation files as
described in the documentation of <code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code>.
Those documentation files will help you describe your project to your
users in a more personalized way. The easiest way to start is to copy
our documentation files to a new location and adapt them. You can locate
them at the following path.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Locate our documentation files</span></span>
<span><span class="va">docs_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"app"</span>, <span class="st">"www"</span>, package <span class="op">=</span> <span class="st">"spatialLIBD"</span><span class="op">)</span></span>
<span><span class="va">docs_path</span></span>
<span><span class="co">#&gt; [1] "/__w/_temp/Library/spatialLIBD/app/www"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">docs_path</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "documentation_sce_layer.md" "documentation_spe.md"      </span></span>
<span><span class="co">#&gt; [3] "favicon.ico"                "footer.html"               </span></span>
<span><span class="co">#&gt; [5] "README.md"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="wrapper-functions">Wrapper functions<a class="anchor" aria-label="anchor" href="#wrapper-functions"></a>
</h2>
<p>To facilitate reading in the data and preparing it to visualize it
interactively using <code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code>, we implemented
<code><a href="../reference/read10xVisiumWrapper.html">read10xVisiumWrapper()</a></code> which expands
<code><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">SpatialExperiment::read10xVisium()</a></code> and performs the steps
described in this vignette. In this example, we’ll load all four images
created by SpaceRanger: lowres, hires, detected, and aligned. That way
we can toggle between them on the web application.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Import the data as a SpatialExperiment object</span></span>
<span><span class="va">spe_wrapper</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read10xVisiumWrapper.html">read10xVisiumWrapper</a></span><span class="op">(</span></span>
<span>    samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"outs"</span><span class="op">)</span>,</span>
<span>    sample_id <span class="op">=</span> <span class="st">"lymph"</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"sparse"</span>, data <span class="op">=</span> <span class="st">"filtered"</span>,</span>
<span>    images <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lowres"</span>, <span class="st">"hires"</span>, <span class="st">"detected"</span>, <span class="st">"aligned"</span><span class="op">)</span>, load <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    reference_gtf <span class="op">=</span> <span class="va">gtf_cache</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 2023-04-08 19:20:07 SpatialExperiment::read10xVisium: reading basic data from SpaceRanger</span></span>
<span><span class="co">#&gt; 2023-04-08 19:20:26 read10xVisiumAnalysis: reading analysis output from SpaceRanger</span></span>
<span><span class="co">#&gt; 2023-04-08 19:20:26 add10xVisiumAnalysis: adding analysis output from SpaceRanger</span></span>
<span><span class="co">#&gt; 2023-04-08 19:20:26 rtracklayer::import: reading the reference GTF file</span></span>
<span><span class="co">#&gt; 2023-04-08 19:21:01 adding gene information to the SPE object</span></span>
<span><span class="co">#&gt; Warning: Gene IDs did not match. This typically happens when you are not using</span></span>
<span><span class="co">#&gt; the same GTF file as the one that was used by SpaceRanger. For example, one</span></span>
<span><span class="co">#&gt; file uses GENCODE IDs and the other one ENSEMBL IDs. read10xVisiumWrapper()</span></span>
<span><span class="co">#&gt; will try to convert them to ENSEMBL IDs.</span></span>
<span><span class="co">#&gt; Warning: Dropping 29 out of 36601 genes for which we don't have information on</span></span>
<span><span class="co">#&gt; the reference GTF file. This typically happens when you are not using the same</span></span>
<span><span class="co">#&gt; GTF file as the one that was used by SpaceRanger.</span></span>
<span><span class="co">#&gt; 2023-04-08 19:21:03 adding information used by spatialLIBD</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run our shiny app</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe_wrapper</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span></span>
<span>        <span class="va">spe_wrapper</span>,</span>
<span>        sce_layer <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        modeling_results <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        sig_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        title <span class="op">=</span> <span class="st">"spatialLIBD: human lymph node by 10x Genomics (made with wrapper)"</span>,</span>
<span>        spe_discrete_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">vars</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"10x_"</span>, <span class="va">vars</span><span class="op">)</span><span class="op">]</span>, <span class="st">"ManualAnnotation"</span><span class="op">)</span>,</span>
<span>        spe_continuous_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum_umi"</span>, <span class="st">"sum_gene"</span>, <span class="st">"expr_chrM"</span>, <span class="st">"expr_chrM_ratio"</span><span class="op">)</span>,</span>
<span>        default_cluster <span class="op">=</span> <span class="st">"10x_graphclust"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="publishing-your-web-application">Publishing your web application<a class="anchor" aria-label="anchor" href="#publishing-your-web-application"></a>
</h2>
<p>Now we have have our <code>spe_wrapper</code> object and have
verified that the web application works, we can share it with others
through <a href="https://www.shinyapps.io/" class="external-link">shinyapps.io</a> by RStudio
if our data is small enough. To do, you will need to:</p>
<ul>
<li>create a directory <code>app_dir</code> which will contain the
script for loading the data and running the web application</li>
<li>save the <code>spe_wrapper</code> object into that directory such as
<code>app_dir/spe_wrapper.rds</code>
</li>
<li>create an <code>app.R</code> script and save it at
<code>app_dir/app.R</code>
</li>
<li>optionally copy the contents from
<code>system.file("app", "www", package = "spatialLIBD")</code> into
<code>app_dir/www</code>
</li>
<li>authenticate to <a href="https://www.shinyapps.io/" class="external-link">shinyapps.io</a>
and deploy the app, which we usually do with a
<code>app_dir/deploy.R</code> script</li>
</ul>
<p>Below we illustrate these steps</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Directory we created to host the data for the web application</span></span>
<span><span class="co">## Use a directory of your preference instead of copy-pasting this code</span></span>
<span><span class="va">app_dir</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html" class="external-link">here</a></span><span class="op">(</span><span class="st">"inst"</span>, <span class="st">"spe_wrapper_app"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">app_dir</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Code we used to save the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">spe_wrapper</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">app_dir</span>, <span class="st">"spe_wrapper.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Copy the contents of system.file("app", "www", package = "spatialLIBD")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"app"</span>, <span class="st">"www"</span>, package <span class="op">=</span> <span class="st">"spatialLIBD"</span><span class="op">)</span>, <span class="va">app_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## Manually edit them to your liking.</span></span></code></pre></div>
<p>Next, create an <code>app_dir/app.R</code> script with these
contents.</p>
<pre><code><span><span class="co">#&gt; library("spatialLIBD")</span></span>
<span><span class="co">#&gt;  library("markdown") ## due to a shinyapps.io bug</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## spatialLIBD uses golem</span></span>
<span><span class="co">#&gt;  options("golem.app.prod" = TRUE)</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## You need this to enable shinyapps to install Bioconductor packages</span></span>
<span><span class="co">#&gt;  options(repos = BiocManager::repositories())</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Load the data (all paths are relative to this script's location)</span></span>
<span><span class="co">#&gt;  spe_wrapper &lt;- readRDS("spe_wrapper.rds")</span></span>
<span><span class="co">#&gt;  vars &lt;- colnames(colData(spe_wrapper))</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Deploy the website</span></span>
<span><span class="co">#&gt;  run_app(</span></span>
<span><span class="co">#&gt;      spe_wrapper,</span></span>
<span><span class="co">#&gt;      sce_layer = NULL,</span></span>
<span><span class="co">#&gt;      modeling_results = NULL,</span></span>
<span><span class="co">#&gt;      sig_genes = NULL,</span></span>
<span><span class="co">#&gt;      title = "spatialLIBD: human lymph node by 10x Genomics",</span></span>
<span><span class="co">#&gt;      spe_discrete_vars = c(vars[grep("10x_", vars)], "ManualAnnotation"),</span></span>
<span><span class="co">#&gt;      spe_continuous_vars = c("sum_umi", "sum_gene", "expr_chrM", "expr_chrM_ratio"),</span></span>
<span><span class="co">#&gt;      default_cluster = "10x_graphclust",</span></span>
<span><span class="co">#&gt;      docs_path = "www"</span></span>
<span><span class="co">#&gt;  )</span></span></code></pre>
<p>Next create an <code>app_dir/deploy.R</code> script with these
contents. You will need to login to your <a href="https://www.shinyapps.io/" class="external-link">shinyapps.io</a> account to get the
exact authentication details for your access token.</p>
<pre><code><span><span class="co">#&gt; library("rsconnect")</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Locate app_dir. Edit as needed</span></span>
<span><span class="co">#&gt;  app_dir &lt;- here::here("inst", "spe_wrapper_app")</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Or you can go to your shinyapps.io account and copy this</span></span>
<span><span class="co">#&gt;  ## Here we do this to keep our information hidden.</span></span>
<span><span class="co">#&gt;  load(file.path(app_dir, ".deploy_info.Rdata"), verbose = TRUE)</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Authenticate to shinyapps.io</span></span>
<span><span class="co">#&gt;  rsconnect::setAccountInfo(</span></span>
<span><span class="co">#&gt;      name = deploy_info$name,</span></span>
<span><span class="co">#&gt;      token = deploy_info$token,</span></span>
<span><span class="co">#&gt;      secret = deploy_info$secret</span></span>
<span><span class="co">#&gt;  )</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## You need this to enable shinyapps to install Bioconductor packages</span></span>
<span><span class="co">#&gt;  options(repos = BiocManager::repositories())</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;  ## Deploy the app, that is, upload it to shinyapps.io</span></span>
<span><span class="co">#&gt;  ## Note that appFiles has to be relative to app_dir.</span></span>
<span><span class="co">#&gt;  ## Drop the www directory if you didn't customize the documentation files and</span></span>
<span><span class="co">#&gt;  ## edit app.R accordingly.</span></span>
<span><span class="co">#&gt;  rsconnect::deployApp(</span></span>
<span><span class="co">#&gt;      appDir = app_dir,</span></span>
<span><span class="co">#&gt;      appFiles = c(</span></span>
<span><span class="co">#&gt;          "app.R",</span></span>
<span><span class="co">#&gt;          "spe_wrapper.rds",</span></span>
<span><span class="co">#&gt;          gsub(file.path(app_dir, "www"), "www", dir(file.path(app_dir, "www"), full.names = TRUE))</span></span>
<span><span class="co">#&gt;      ),</span></span>
<span><span class="co">#&gt;      appName = "spatialLIBD_Human_Lymph_Node_10x",</span></span>
<span><span class="co">#&gt;      account = "libd",</span></span>
<span><span class="co">#&gt;      server = "shinyapps.io"</span></span>
<span><span class="co">#&gt;  )</span></span></code></pre>
<p>Next, you should test that <code>app_dir/app.R</code> works locally
before running <code>app_dir/deploy.R</code>.</p>
<p>You’ll end up with a website <a href="https://libd.shinyapps.io/spatialLIBD_Human_Lymph_Node_10x/" class="external-link">just
like this one</a>. In our case, we further configured our website
through the shinyapps.io dashboard. We selected the following
options:</p>
<ul>
<li>
<em>General</em> <code>Instance Size</code>: 3X-Large (8GB)</li>
<li>
<em>Advanced</em> <code>Max Worker Processes</code>: 1. Setting this
to one helps reduce the memory burden.</li>
<li>
<em>Advanced</em> <code>Max Connections</code>: 15. You don’t want
<code>Max Worker Processes</code> multiplied by
<code>Max Connections</code> and the object memory to exceed the
<code>Instance Size</code>. So this this value to
<code>8 / (lobstr::obj_size(spe_wrapper) / 1024^3)</code> or lower.</li>
<li>
<em>Advanced</em> <code>Startup Timeout</code>: 300. This one is
important otherwise your app might error out before even loading since
the default is 5 seconds, which is typically too short.</li>
<li>
<em>Advanced</em> <code>Idle Timeout</code>: 60. This is useful
since you might want to give users as much time for their manual
annotations as possible.</li>
</ul>
</div>
<div class="section level2">
<h2 id="limitations">Limitations<a class="anchor" aria-label="anchor" href="#limitations"></a>
</h2>
<p><code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code> has limitations that are inherent
to the methods used to implement it, such as:</p>
<ol style="list-style-type: decimal">
<li>the memory per user required by a server for hosting the web
application,</li>
<li>response speeds for the interactive views due to the number of
spots,</li>
<li>the resolution of the images displayed limiting the usefulness to
magnify specific spots,</li>
<li>and customization of the web application by the end user.</li>
</ol>
<div class="section level3">
<h3 id="memory">Memory<a class="anchor" aria-label="anchor" href="#memory"></a>
</h3>
<p>Regarding the memory limitation, you can estimate how much memory you
need per user by considering the memory required for the
<code>spe</code> and <code>sce_layer</code> objects.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">lobstr</span><span class="fu">::</span><span class="fu"><a href="https://lobstr.r-lib.org/reference/obj_size.html" class="external-link">obj_size</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1024</span><span class="op">^</span><span class="fl">2</span> <span class="co">## Convert to MB</span></span>
<span><span class="co">#&gt; 283.88 B</span></span></code></pre></div>
<p>In our pilot data (Maynard, Collado-Torres, Weber et al., 2021) our
object uses about 2.1 GB of RAM since it contains the data for 12 Visium
slides and we considered using about 3 GB of RAM per user. You could
filter the genes more aggressively to drop lowly expressed genes or if
you have many Visium slides, you could consider making multiple websites
for different sets of slides. You could also have multiple mirrors to
support several users, though in that case, we recommend linking users
to a stable website instead of one that might not be available if you
have too many users: for us our stable website is <a href="http://research.libd.org/spatialLIBD/" class="external-link uri">http://research.libd.org/spatialLIBD/</a> which includes the
links to all the mirrors.</p>
<p>Given these memory limitations, we chose to deploy our main web
application at <a href="http://spatial.libd.org/spatialLIBD/" class="external-link uri">http://spatial.libd.org/spatialLIBD/</a> using an Amazon EC2
instance: an ‘r5.4xlarge’ EC2 instance with 16 vCPUs, 128 GB DRAM, 10 Gb
max network, 1.008 USD/Hour. We also have deployed mirrors at <a href="https://www.shinyapps.io/" class="external-link uri">https://www.shinyapps.io/</a> using the “3X-Large (8 GB)”
instances they provide.</p>
</div>
<div class="section level3">
<h3 id="response-speeds">Response speeds<a class="anchor" aria-label="anchor" href="#response-speeds"></a>
</h3>
<p>This limitation is mostly due to the number of spots shown under the
“clusters (interactive)” section of the interactive website powered by
<em><a href="https://CRAN.R-project.org/package=plotly" class="external-link">plotly</a></em>.
Each spot is shown four times which is about 16 thousand spots for one
Visium slide (depending on any filter steps you applied). The response
time will depend on your own computer RAM memory, that is, the
<em>client</em> side. This limitation might be more noticeable if you
have a computer with 8GB of RAM or lower, as well as if you have other
high-memory software open. Furthermore, if you are running web
application locally through <code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code> then you
also need to consider the required memory for the R objects. That is,
the <em>server</em> side memory use.</p>
<p>Thanks to <a href="http://orcid.org/0000-0001-5128-3838" class="external-link">Jesús Vélez
Santiago</a>, the app is more responsive as of version 1.3.15 by using
<code><a href="https://rdrr.io/pkg/plotly/man/toWebGL.html" class="external-link">plotly::toWebGL()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="image-resolution">Image resolution<a class="anchor" aria-label="anchor" href="#image-resolution"></a>
</h3>
<p>When you construct the SpatialExperiment <code>spe</code> object with
<em><a href="https://bioconductor.org/packages/3.16/SpatialExperiment" class="external-link">SpatialExperiment</a></em>,
you can read in higher resolution images. However, the benefit of
loading the raw histology images (500 MB to 20 GB per image) is likely
non-existent in this web application. The memory required would likely
become prohibitive. Other solutions load these raw histology images in
chunks and display the chunks necessary for a given visualization area.
We thus recommend using other software if you want to zoom in at the
spot and/or cell resolution.</p>
</div>
<div class="section level3">
<h3 id="customization">Customization<a class="anchor" aria-label="anchor" href="#customization"></a>
</h3>
<p>While the documentation, title, icon and HTML footer are all
customizable at <code><a href="../reference/run_app.html">spatialLIBD::run_app()</a></code>, ultimately the
panels shown are not unless you fork and adapt the internal code of this
package. Thus, the interactive web applications powered by <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
are not as easy to customize as say <em><a href="https://bioconductor.org/packages/3.16/iSEE" class="external-link">iSEE</a></em> web
applications are. We think of our web application as a good enough
prototype that can be useful for initial explorations of 10x Genomics
Visium data. We welcome additions to our code, though we recognize that
you might want to build your own production-level solution.</p>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.16/spatialLIBD" class="external-link">spatialLIBD</a></em>
package (Pardo, Spangler, Weber et al., 2022) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocFileCache" class="external-link">BiocFileCache</a></em>
(Shepherd and Morgan, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=lobstr" class="external-link">lobstr</a></em>
(Wickham, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/rtracklayer" class="external-link">rtracklayer</a></em>
(Lawrence, Gentleman, and Carey, 2009)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
(Righelli, Weber, Crowell, Pardo, Collado-Torres, Ghazanfar, Lun, Hicks,
and Risso, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.16/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"TenX_data_download.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"TenX_data_download.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-04-08 19:21:05 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 2.488 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-04-08</span></span>
<span><span class="co">#&gt;  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi            1.60.2    <span style="color: #555555;">2023-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationHub            3.6.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  attempt                  0.3.1     <span style="color: #555555;">2020-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  backports                1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  beachmat                 2.14.1    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  beeswarm                 0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  benchmarkme              1.0.8     <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  benchmarkmeData          1.0.4     <span style="color: #555555;">2020-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bibtex                   0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                * 2.58.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache          * 2.6.1     <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics           * 0.44.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocIO                   1.8.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager              1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors            1.16.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocParallel             1.32.6    <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocSingular             1.14.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle              * 2.26.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocVersion              3.16.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               2.66.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                      4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bit64                    4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  bitops                   1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  blob                     1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bookdown                 0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib                    0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem                   1.0.7     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                      3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  codetools                0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;  colorspace               2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  config                   0.3.1     <span style="color: #555555;">2020-12-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  cowplot                  1.1.1     <span style="color: #555555;">2020-12-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  crayon                   1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  curl                     5.0.0     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  data.table               1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DBI                      1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dbplyr                 * 2.3.2     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray             0.24.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedMatrixStats       1.20.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                     1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest                   0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  doParallel               1.0.17    <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dotCall64                1.0-2     <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  dplyr                    1.1.1     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dqrng                    0.3.0     <span style="color: #555555;">2021-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  DropletUtils             1.18.1    <span style="color: #555555;">2022-11-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DT                       0.27      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                    3.40.2    <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ellipsis                 0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate                 0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub            2.6.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  fansi                    1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  farver                   2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  fastmap                  1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fields                   14.1      <span style="color: #555555;">2022-08-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  filelock                 1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  foreach                  1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  fs                       1.6.1     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  generics                 0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb           * 1.34.9    <span style="color: #555555;">2023-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData         1.2.9     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments        1.34.1    <span style="color: #555555;">2023-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges          * 1.50.2    <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm               0.7.1     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2                  3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                  0.9.3     <span style="color: #555555;">2023-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  glue                     1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  golem                    0.4.0     <span style="color: #555555;">2023-03-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gridExtra                2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  gtable                   0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  HDF5Array                1.26.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  highr                    0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools                0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets              1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                   1.6.9     <span style="color: #555555;">2023-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  httr                     1.4.5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  interactiveDisplayBase   1.36.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  IRanges                * 2.32.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  irlba                    2.3.5.1   <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  iterators                1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  jquerylib                0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite                 1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                 1.38.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                    1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  labeling                 0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  later                    1.3.0     <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice                  0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lazyeval                 0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  lifecycle                1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  limma                    3.54.2    <span style="color: #555555;">2023-02-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  lobstr                 * 1.1.2     <span style="color: #555555;">2022-06-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  locfit                   1.5-9.7   <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lubridate                1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magick                   2.7.4     <span style="color: #555555;">2023-03-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr                 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  maps                     3.4.1     <span style="color: #555555;">2022-10-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Matrix                   1.5-4     <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics         * 1.10.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats            * 0.63.0    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  memoise                  2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  mime                     0.12      <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  munsell                  0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  paletteer                1.5.0     <span style="color: #555555;">2022-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pillar                   1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                  2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plotly                   4.10.1    <span style="color: #555555;">2022-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  plyr                     1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  png                      0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  prettyunits              1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  promises                 1.2.0.1   <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr                    1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R.methodsS3              1.8.2     <span style="color: #555555;">2022-06-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  R.oo                     1.25.0    <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  R.utils                  2.12.2    <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  R6                       2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg                     1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs                 0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer             1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  Rcpp                     1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                    1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR             * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rematch2                 2.1.2     <span style="color: #555555;">2020-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  restfulr                 0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rhdf5                    2.42.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rhdf5filters             1.10.1    <span style="color: #555555;">2023-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib                 1.20.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rjson                    0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rlang                    1.1.0     <span style="color: #555555;">2023-03-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                2.21      <span style="color: #555555;">2023-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot                2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools                2.14.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  RSQLite                  2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                     1.0.5     <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  rtracklayer            * 1.58.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Vectors              * 0.36.2    <span style="color: #555555;">2023-02-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                     0.4.5     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix             1.6.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scales                   1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  scater                   1.26.1    <span style="color: #555555;">2022-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scuttle                  1.8.4     <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sessioninfo            * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  shiny                    1.7.4     <span style="color: #555555;">2022-12-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets             0.7.6     <span style="color: #555555;">2023-01-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment   * 1.20.1    <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  spam                     2.9-1     <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  sparseMatrixStats        1.10.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  SpatialExperiment      * 1.8.1     <span style="color: #555555;">2023-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  spatialLIBD            * 1.11.13   <span style="color: #555555;">2023-04-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  statmod                  1.5.0     <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi                  1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr                  1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment   * 1.28.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts              1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping              0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tibble                   3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                    1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect               1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  timechange               0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  utf8                     1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                    0.6.1     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vipor                    0.4.5     <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  viridis                  0.6.2     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  viridisLite              0.4.1     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  withr                    2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun                     0.38      <span style="color: #555555;">2023-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XML                      3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xml2                     1.3.3     <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xtable                   1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector                  0.38.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                     2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                 1.44.0    <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.21. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-lawrence2009rtracklayer"></a><a href="#cite-lawrence2009rtracklayer">[2]</a><cite>
M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for
interfacing with genome browsers”. In: <em>Bioinformatics</em> 25
(2009), pp. 1841-1842. DOI:
<a href="https://doi.org/10.1093/bioinformatics/btp328" class="external-link">10.1093/bioinformatics/btp328</a>.
URL:
<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="external-link">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>.</cite>
</p>
<p>
<a id="bib-maynard2021transcriptome"></a><a href="#cite-maynard2021transcriptome">[3]</a><cite>
K. R. Maynard, L. Collado-Torres, L. M. Weber, et al.
“Transcriptome-scale spatial gene expression in the human dorsolateral
prefrontal cortex”. In: <em>Nature Neuroscience</em> (2021). DOI:
<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">10.1038/s41593-020-00787-0</a>.
URL:
<a href="https://www.nature.com/articles/s41593-020-00787-0" class="external-link">https://www.nature.com/articles/s41593-020-00787-0</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[4]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[5]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.26.0. 2022. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-pardo2022spatiallibd"></a><a href="#cite-pardo2022spatiallibd">[6]</a><cite>
B. Pardo, A. Spangler, L. M. Weber, et al. “spatialLIBD: an
R/Bioconductor package to visualize spatially-resolved transcriptomics
data”. In: <em>BMC Genomics</em> (2022). DOI:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">10.1186/s12864-022-08601-w</a>.
URL:
<a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">https://doi.org/10.1186/s12864-022-08601-w</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[7]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-righelli2022spatialexperiment"></a><a href="#cite-righelli2022spatialexperiment">[8]</a><cite>
D. Righelli, L. M. Weber, H. L. Crowell, et al. “SpatialExperiment:
infrastructure for spatially-resolved transcriptomics data in R using
Bioconductor”. In: <em>Bioinformatics</em> 38.11 (2022), pp. -3. DOI:
<a href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac299" class="external-link">https://doi.org/10.1093/bioinformatics/btac299</a>.</cite>
</p>
<p>
<a id="bib-shepherd2023biocfilecache"></a><a href="#cite-shepherd2023biocfilecache">[9]</a><cite>
L. Shepherd and M. Morgan. <em>BiocFileCache: Manage Files Across
Sessions</em>. R package version 2.6.1. 2023.</cite>
</p>
<p>
<a id="bib-wickham2022lobstr"></a><a href="#cite-wickham2022lobstr">[10]</a><cite>
H. Wickham. <em>lobstr: Visualize R Data Structures with Trees</em>. R
package version 1.1.2. 2022. URL:
<a href="https://CRAN.R-project.org/package=lobstr" class="external-link">https://CRAN.R-project.org/package=lobstr</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[11]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[12]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-xie2023knitr"></a><a href="#cite-xie2023knitr">[13]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.42. 2023. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
